# Genomic profiling of response to in vivo immune perturbations

Repo for PhD thesis, submitted for the degree of Doctor of Philosophy at the Wellcome Sanger Institute and the University of Cambridge.

## Author

Benjamin Yu Hang Bai

## Supervisor

Carl A Anderson

## Citation

Bai, B. Y. H. (2020). Genomic profiling of response to in vivo immune perturbations (Doctoral thesis). https://doi.org/10.17863/CAM.66356

## Abstract

The human immune system plays a central role in defense against infection, but its dysregulation is implicated in immune-mediated diseases. The past decade has seen increasing application of high-throughput technologies to profile, predict, and understand immune response to perturbation. The ability to measure immune gene expression at scale has led to the identification of transcriptomic signatures that predict clinical phenotypes such as antibody response to vaccines. It has also been recognised that both expression and phenotypic responses are traits with complex genetic architectures. This thesis examines the longitudinal transcriptomic response to immune perturbations, and its association with clinical response phenotypes and common genetic variation.

Chapter 2 explores transcriptomic response to pandemic influenza vaccine in a multi-ethnic cohort of healthy adults: the Human Immune Response Dynamics (HIRD) cohort. The success of vaccination in controlling influenza is indisputable, but it is incompletely understood why some individuals fail to mount protective antibody responses. I meta-analysed blood microarray and RNA sequencing (RNA-seq) datasets, identifying a distinct transition from innate immune response at day 1 after vaccination to adaptive immune response at day 7. Heterogeneity between measurement platforms made it difficult to identify single-gene transcriptomic associations with antibody response. Using a gene set approach, I found expression modules related to the inflammatory response, the cell cycle, CD4+ T cells, and plasma cells to be associated with vaccine-induced antibody response.

In Chapter 3, I map response expression quantitative trait loci (reQTLs) in the HIRD cohort to investigate regulation of transcriptomic response by common genetic variants. Rather than driving differential expression post-vaccination, the strongest reQTLs appeared to be explained by changes in cell composition revealing cell type-specific expression quantitative trait locus (eQTL) effects. For example, a reQTL identified for ADCY3 specific to day 1 may be explained largely by high monocyte proportions at day 1 compared to other timepoints. Changes in cell composition present a significant challenge to interpreting reQTLs found through bulk sequencing of heterogeneous tissues.

Finally, Chapter 4 applies an analogous longitudinal study design to explore drug response in the Personalised Anti-TNF Therapy in Crohn’s Disease (PANTS) cohort: a cohort of Crohn’s disease (CD) patients treated with the anti-tumour necrosis factor (TNF) drugs, infliximab and adalimumab. Anti-TNF treatment has revolutionised patient care for CD, but 20–40% of patients show primary non-response soon after starting treatment. I identified baseline expression modules associated with primary non-response, but also found significant heterogeneity of associations between the two drugs. Expression changes post-treatment in non-responders were largely magnified in responders, suggesting there may be a continuum of response. Distinct expression trajectories identified for responders and non-responders revealed sustained expression differences during the first year of treatment. Sets of interferon-related genes were regulated in opposing directions in responders and non-responders, presenting an attractive target for future studies of the biological mechanisms underlying non-response.
