%
% Chapter 5
% Discussion
%

\chapter{Discussion}
\label{chap:discussion}

\begin{outline}

\1 In this thesis, ... <a summary of the thesis projects>
    \2 Chapters 2 and 4 focus on describing the transcriptomic response of circulating immune cells to perturbation with uncontrolled longitudinal designs
        \3 looking at how expression changes over time, and differences in transcriptomic response associated with other phenotypic definitions of response
        \3 associations are identified at a gene and gene set level
    \2 Chapter 3 (and the last part of chapter 4) focus on finding common genetic variants associated with transcriptomic response to perturbation
        \3 reQTLs are identified at lead eQTL variants for each gene
        \3 this is a first foray into defining causal molecular mechanisms that explain individual variation in response to these perturbations
    \2 Each chapter discusses the results and limitations of each study.
    \2 Being very similar studies in design and analysis, they also suffer from a core set of limitations.
    \2 This chapter discusses these, and the design and analysis of similar longitudinal studies in the future to better understand the biology of immune response to drugs and vaccines.

\subsection{Increasing the power to detect associations}

\1 It has been challenging to find robust single gene-level associations between expression and response, especially at baseline.
    \2 The biological signal may be small compared to other sources of variation: measurement platform (ch2); drug, response definition (ch4)
    \2 The obvious thing to do is increase sample size.
        \3 Under a fixed resource constraints, longitudinal studies tradeoff more samples over time for fewer individuals
        \3 Multiomic designs further tradeoff against richness/number of layers sampled
    \2 there are distinct advantages to these designs
        \3 e.g. repeated measures allows modelling of within patient covariance between timepoints with mixed model DGE (ch2 and ch4)
        \3 e.g. spline analysis and clustering in ch4 using multiple timepoints to separate responders and nonresponders
        \3 e.g. ability to test for association between layers (eQTLs), to explore molecular mechanisms
    \2 especially if considering the small effect of genetic variants, a lesson learnt in the field of complex diseases from years of underpowered candidate gene studies, is that even a single complex phenotype requires large samples
    \2 systems vaccinology studies have always had these rich datasets, but rarely considered genetics, limited by sample size
    \2 is it also challenging to collect sufficient sample size in a pharmacogenomic context \url{https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3003940/} 

\1 with rich datasets, be careful about subgroup analyses, whether done stratified or as interactions; they have lower power
    \2 e.g. looking for timepoint, responder, cell type interactions
    \2 be wary of rules of thumb for sample size, subgrouping quickly attenuates your effective sample size
        \3 e.g. in ch3, the intention was to map eQTLs at each timepoint, look for timepoint x G interaction, but it became apparent that cell proportion interactions were necessary for interpretation
    \2 prespecify interactions of interest at the power analysis stage in design of new experiments
    \2 for existing datasets, a workaround like the two stage strategy in ch3, assuming that interactions are only interesting at significant main effects

\1 Rather than increasing the sample size, can change the unit of analysis from single gene to multiple genes
    \2 use information from coordinated changes in multiple genes to detect subtle effects
    \2 use prior biological knowledge to define set of genes
    \2 this is the basis for gene set analyses used throughout this thesis
    \2 tradeoff between better interpretability but worse resolution

\1 <reproducibility of gene set vs single gene associations> 
    \2 reports about any single gene associations e.g. SIGLEC10 in ch4 of this thesis, should be treated with caution pending replication
    \2 while the rank of a single gene may vary from sample to sample due to sampling noise, a summary measure computed from multiple genes should be more robust
    \2 it is hoped that the replicability of gene sets will be greater

\subsection{Responder analysis}

\1 <how well these models correspond to reality>
    % \2 How a responder to a perturbation is modelled is key to the analyses in this thesis.
    \2 What does it mean to be a responder?
    \2 this has been discussed on this in the field of personalised medicine in the context of RCTs, but similar issues pertain to response in observational studies. \autocite{senn2018StatisticalPitfallsPersonalized}
    \2 Is being a responder is a stable personal characteristic of an individual over time, or would the same individual vary in response status over time?
    % Anchorman: The Legend of Ron Burgundy (2004) - Paul Rudd ...
    % Brian Fantana : They've done studies, you know. 60% of the time, it works every time. [cheesy grin]. Ron Burgundy : That doesn't make sense ..
    \2 Observing between individual variation in response for a single measure is not sufficient to imply response is personal.
    \2 e.g. a hypothetical agent with a 60\% response rate in a group of individuals: \enquote{60\% of the time, it works every time}
    \2 what does this mean? working 60\% of the time on 100\% of individuals, 100\% of the time on 60\% of individuals, or anything in between would be compatible
    \2 cannot distinguish with a single measure, yet the assumption is often made at the study design stage that it is the 2nd option
    \2 1st option means no associations to detect

\1 Need to establish that personal response is likely before trying to find causes for it
    \2 to determine within-individual variation, needs replication at the level of the individual, i.e. need to measure treatment effects more than once \autocite{senn2016MasteringVariationVariance}
    \2 for genetic factors, the field of complex trait genetics has long had a solution for this: twin studies are analogous to measuring the same individual twice
    \2 For phenotypes like vaccine response in ch2 that can be, twin studies have already demonstrated heritability of certain response phenos like influenza vaccine Ab titres \autocite{brodin2015VariationHumanImmune}
    \2 In pharmocogenetics, twin studies are less common \url{https://www.sciencedirect.com/science/article/abs/pii/S0928098719300326}
        \3 and have not been done for response to anti-TNFs (the ch4 setting)
    \2 repeated crossover trials can be a feasible alternative to twin studies \url{https://www.researchgate.net/profile/Laszlo_Endrenyi/publication/13553486_Hypothesis_Comparisons_of_inter-_and_intra-individual_variations_can_substitute_for_twin_studies_in_drug_research/links/59e97c2e458515c36370e6a7/Hypothesis-Comparisons-of-inter-and-intra-individual-variations-can-substitute-for-twin-studies-in-drug-research.pdf} \autocite{senn2016MasteringVariationVariance}

\1 <statistical considerations given response is sensible>
    \2 As discussed in \autoref{subsec:hird_dge_TRI} in ch2, a binary definition of response based on dichotomisation is inefficient and assumes a sharp change in biological relevance at the threshold
    \2 In ch2, used \gls{TRI}, a continuous change score residualised on the baseline, which gets around the dichotomisation and dependence of change score on baseline
        \2 two-stage approach here, so df is wrong
    \2 In multiPANTS, \autoref{multiPANTS:PR_definition} in ch4 uses the binary definition of response, which is complex and based on many inputs
        \3 having response dichotomies based on multiple inputs can lead to large discontinuities and non-monotonicity in response probabilities under small changes in input \autocite{senn2005DichotomaniaObsessiveCompulsive} 
    \2 neither approach perfect, but the driving force was pragmatism
        \3 for DGE, the model traditionally requires the response variable to be gene expression, and a predictor variable to be response status
        \todo{might be worth subbing predictor for independent and response with dependent in all chapters, but then that gets confused with statistical independence...}
        \3 thus I needed an approach that generates a single predictor variable to represent response status
        \3 In ch2, needed to combine pre-post titre data from two assays
        \3 In ch4, the clinical algorithm already combines many measures into one.
    \2 having complicated response definitions can hinder replication by adding to interstudy variation e.g. the failure to replicate the TREM1 association in ch4
    \2 it may be better to directly with the constituent phenotypes: ch2 (log titre levels), ch4 (CRP level, HBI score), but then the fundamental modelling approach for both DGE/eQTL would need to change to not be restricted to a single response x variable 
    \2 the previous paragraph's discussion also applies to continuous measures of response, as what is being considered is that it only makes sense to talk about response if the response measure is tightly correlated within an individual over time

% \1 bias in non-randomised studies
    % \2 In RCTs, the treatment effect would be defined against the effect in the control group
    % \2 In prospective cohort studies, the treatment effect is determined between two subgroups within the cohort

\subsection{Challenges in the interpretation of bulk expression data}

rnaseq tech is cmopositional

raw abundance estimates are often called \enquote{counts}, but actually dimensionless rates

% but cant get around compositional nature of data, and sep change in prop from up/down true
% TODO: Also, \gls{RNAseq} expression estimates are inherently compositional at many levels (blood sample, sequencing) \url{https://www.ncbi.nlm.nih.gov/pubmed/29608657} \url{https://academic.oup.com/gigascience/article/8/9/giz107/5572529}.
% There is ultimately no escape from the compositional nature of the assay.
% https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6084572/
% TODO:
% https://academic.oup.com/gigascience/article/8/9/giz107/5572529#163672243
%
% the output expression levels are interpretable only in comparison to other
% do not not absolute levels

% TODO: actually By  definition,  TPM  and  RPKM  are  propo rtional .
% A  co mmon  misconception  is  tha t  RPKM  and  TPM  values  are  alr eady
% normalized ,   and   thus   should   b e   compar able   across   samples   or
% R NA-seq   pr oject s.   However,   RPKM   and  TPM  repres ent  the  rela tive
% ab undance   o f  a  transcri pt  amo ng  a  popul atio n  of  seq uenced
% transcrip ts,  a nd  ther efore  dep end  on  th e  composition  of  the  RNA
% popula tion  in  a  sample.  Q
% https://rnajournal.cshlp.org/content/early/2020/04/13/rna.074922.120.full.pdf
%

and bilk data is compositional

% TODO: farahbod2020UntanglingEffectsCellular
% bulk samples are a mix
% observed expression of a gene across samples is a mix of cell type specificness of the gene and composotition (affected by sample error and bio fx)
% and composition can explain a lot of variance
% the more cell type specific, the more affected by composition
% extreme case: perfectly cell type specific is called a marker gene, and is used in deconv methods to actually get cell estiamtes.

DGE and reQTL on bulk in vivo has huge challenges
more n won't help
what data to get from these trials?

    composition explains a huge amount of var in bulk farahbod2020UntanglingEffectsCellular

    can correct, but see 30 for age correction
        ch3, coarse
        ch4, even coarser

        rarer pops, so can always do more
        is it ever enough?

    some are implicitly accounted for in peer
        but interactions with geno are not guaranteed to be

    effect not absent in vitro
        technology itself is proportional, with norm based on ng input, then random sampling during sequencing

but effect is certainly EXARCERBATED by in vivo
    heterogeneity is confounded with recruitment!!!
    not just sample var, but active recuitment changes in cell prop e.g. recruitment
        why should this be considered a con of in vivo?

dge
    mediation

reqtl
    Even if cell type interaction/proxy gene approach
        Cannot distinguish between correlated cell types

    need to recognise 
        bulk and sorted eQTL differ in interp
            and correction using cell prop covariates does not unify them

        also, it's not the same as intervertion to set cell counts e.g. FACS

            control is not intervention
                % http://bayes.cs.ucla.edu/PRIMER/ch3-preview.pdf
                The difference between intervening on a variable and conditioning on that
                variable should,hopefully, be obvious. When we intervene on a variable in a
                model, we fix its value. Wechangethe system, and the values of other variables
                often change as a result. When we condition on avariable, we change nothing; we
                merely narrow our focus to the subset of cases in which thevariable takes the
                value we are interested in. What changes, then, is our perception about
                theworld, not the world itself.

    unclear what benefit would outweigh this
        imo, needs consideration for future study design

        non-bulk means more chance for in vivo
            allows to take advantage of benefits

    A response eqtl is not always a response eqtl
    must no longer be divorced from rigourous statistical theory

        statistical considerations from small scale studies
            Two disciplines of statistical genomics : Neither can be neglected


            % NOTE: see
            % Senn: Statistical Issues in Drug Development
            % 7.2.4 Is clinical relevance a relevant consideration when choosing whether to use change scores or raw outcomes as the efficacy measure?

            in chapter 3, i discussed in brief
                change is not ancova
                stratified is not interaction
                yet many studies out there the that use one or the other to answer the same biological reQTL questions

        beware the effects of normalisation (on top of stratification)

        beware the effects of model misspec
            missing interaction terms

        esp when considering biologically meaningful replication of reQTL
            all above must line up

    before bridging to pheno, must consider how to interpret G -> E link
    
    it ulttimaley depends on the point of the reqtl mapping

    these are issues that cannot be solved by n
        i.e. do we need to measure the wrong thing more precisely?

\subsection{Finer and finer context}

Finer and finer context
    in the intro: gwas to function pipeline
    now, the future

    sorted cells

    More timepoints
    Allele-specific expression changes dynamically during T cell activation in HLA and other autoimmune loci
    % https://www.nature.com/articles/s41588-020-0579-4

    More conditions
    e.g. 250 condition ASE % https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5131815/
    e.g. StructLMM 
        Identifies eQTLs with GxE, where the number of environments in E is large (modelled as a random effect)

    as datasets and conditions get larger, proportion of eGenes is going to be 100pc, then the question is what are the most relevant ones
    % We observed that cis-eQTLs can be detected for 88% of the studied genes in vosa2018UnravelingPolygenicArchitecture
    % In contrast, trans-eQTLs (detected for 37% of 10,317 studied trait-associated variants) were more informative.

    Era of single cell. (ultamite context)
        refined over modules
        
        1st
        Single-cell RNA sequencing identifies celltype-specific cis-eQTLs and co-expression QTLs
        https://www.nature.com/articles/s41588-018-0089-9

        "Single-cell eQTLGen Consortium: a personalized understanding of disease"
        https://arxiv.org/abs/1909.12550

        Optimal design of single-cell RNA sequencing experiments for cell-type-specific eQTL analysis
        https://www.biorxiv.org/content/biorxiv/early/2019/09/12/766972.full.pdf

        Single-cell genomic approaches for developing the next generation of immunotherapies Ido Yofe, Rony Dahan and Ido Amit

        % Lähnemann, D., Köster, J., Szczurek, E., McCarthy, D. J., Hicks, S. C.,
        % Robinson, M. D., Vallejos, C. A., Campbell, K. R., Beerenwinkel, N., Mahfouz,
        % A., Pinello, L., Skums, P., Stamatakis, A., Attolini, C. S.-O., Aparicio, S.,
        % Baaijens, J., Balvert, M., Barbanson, B. de, Cappuccio, A., … Schönhuth, A.
        % (2020). Eleven grand challenges in single-cell data science. Genome Biology,
        % 21(1). https://doi.org/10.1186/s13059-020-1926-6

        reQTL detection: bulk, sorted, sc
        current sc will only detect highly expressed genes

        reqtl followup using single cell cell type specific expression
        % https://journals.plos.org/plospathogens/article?id=10.1371/journal.ppat.1008408

        devries2020IntegratingGWASBulk
        Subsequently, scRNA-seq data was used to pinpoint the potential cell type in which the response QTL effects manifest themselves
            but really they only looked at cell type specific expression, not sc eqtl mapping

    be aware that as prev discussed, finer contet needs more n

% more n allows more phenotypes
%
%         disease specific biobanks e.g. ibd bioresource/predicct
%
%    Our GWASfor 54 functionally relevantphenotypes oftheadaptiveimmune system in 489 healthy individuals identifieseight genome-wide significant associations explain-ing 6%–20% of variance.
%         https://www.cell.com/cell-reports/pdf/S2211-1247(18)31493-1.pdf
%
%     PheWAS\autocite{verma2017CurrentScopeChallenges}
%         https://onlinelibrary.wiley.com/doi/full/10.1111/imm.12195
%         PheWAS has the advantage of identifying genetic variants with pleiotropic properties.

\subsection{Prediction is not inference}

finally translation and prediction
    a diff question althogether

    have now discussed both the indep and dependent variables
    response on right, e on left

    response always considered something intrinsic, even in cases where E is temporally prior

    Prediction is not inference (the rift between philosophies, see 2 x 2 schools papers)
        % Calculating the sample size required for developing a clinical prediction model
        two chapters of mine are merely descriptive, only form the foundation for next leap should be to causal or predictive
            have touched on suggesting to build a predictive model
        but have always treated the R as indep in DGE
        kinda strange: response is a organism pheno should be downstream of E

            Is response , the main predictor
            A property measured at w14
            Is actually A time invariant property of the individual?

            for modelling convenicnece
            cant make causal claims anyway
                without a control, we cannot observe the counterfactual of what if an individual was a non-responder

            correlates of protection

            no error
                assuming otherwise would stray into the realm of error in variables models
            
                we are effectively assuming that R is an intrinsic thing, and is determined without error

        gene signatures, rise and fall
            expression as a biomarker

            success stories \url{https://www.nature.com/articles/415530a} 
            and
            rise and fall paper: 15 years experience from cancer: list challenges to clinical implementation from Discussion

            % https://www.annalsofoncology.org/article/S0923-7534(19)36531-7/fulltext
            Table 1Evidence-based criteria for a prognostic gene signature in the path from the laboratory to clinical practice

        needed to reach the status of signature

    modules
        DGE is a ranking and selection problem, so rank based methods may reproduce better
        lots of info is in the relative ordering
        and its unlikely that any one gene should be responsible, focus should be on gene set level analyses
        by chance, e.g. SIGLEC10 has clearest signal noise ratio
        
    efron2020PredictionEstimationAttribution prediction is inherently different
        % For those of uswho have struggled to find “significant” genes in
        % a microarray study,8 the almost perfect prostate cancer predictions
        % of randomforests and gbmhave to come as a disconcerting
        % surprise.
            not clear DGE -> prediction is necessary
            prediction may still be possible even with little significant single gene diffs

            is the gene by gene approach good for pred?
        
        need to repeat DGE within folds
            no guarantee that top DGE will be top predictors

        need to convert the language e.g. logFC, to prediction e.g. CV error

        cell props not ideal for prediction?
            gaujoux2019CellcentredMetaanalysisReveals: "These results suggest that the predictive power of reported gene signatures is largely based on cell subset proportion differences, whose increase in colon biopsies of non-responders may serve for predictive purposes."

        % beyond dge
        % TODO: https://bmcgenomics.biomedcentral.com/articles/10.1186/1471-2164-13-356

        potential methods

            Sparse partial least squares regression for simultaneous dimension reduction and variable selection
            % https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2810828/

             Sparse Partial Least Squares Classification for High Dimensional Data 
            % https://pubmed.ncbi.nlm.nih.gov/20361856/

        sysvacc: need genetics to move beyond prediction

    might even be said that whole blood biological interps so limited that prediction may be the only feasible goal

    explicitly consider:
    prediction models require different sample size calcualtions

    and to Cost-effectiveness and clinical implementation
        if you can identify NRs, what are you going to do about it?

        what does it take for a gene expression pred model to be clinically useful?
            take lessons from cancer

\subsection{The missing link for causality}

invert relationship

It's great to have replicable associations with fine context at gene and network levels 
    But, what we really want to know is causality. 
    counterfactual
    intervene

    to get that from uncontrolled study data, need to bring in genetics

    Mechanism is important because of possibility for intervention. Here, causal is important
    % 'molecular pathogenesis'

    need a strong causal anchor: this is going backwards towards gwas, which requires N
        first forray is the reqtl analyses, but a link is missing
        G -> pheno

        sample size a problem here again
    % \2 take advantage of existing external resources that have very large n
    %     \2 e.g. instead of mapping eQTLs in sample, use existing eQTL maps in whole blood
    %     \2 e.g. biobanks

            complex traits and disease already moved out of candidate gene era
            vaccine and drug response traits lagging due to sample size requirements

        systems immunology/vacc still needed: generate mol phenotypes, in the right context,
        but make sure to include genotypes
        Moving out of correlation land

    need causal models

        avoid "low input, high throughput, no output"
        without causality no talking about 'mechanism'

    need statistical integration
        problems with connection between dge and reqtl, as in prev ch

        simple overlap will not work

            e.g. other papers that talk about 'integrating' recognise this
            % franco,
            % https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0180824
            % devries2020IntegratingGWASBulk

    option:
        Causal Inference Test for a Binary Outcome

        % TODO
        % https://onlinelibrary.wiley.com/doi/full/10.1002/gepi.22061
        % The CIT is specifically designed to test whether a variablemediates the association between (and is the only causal linkbetween) a genetic locus and a quantitative trait. It is moreflexible than MR because it does not assume that the geneticvariant is chosen specifically to be an instrument for the medi-ator. Due to the way the test is constructed, the CIT is alsoimmune to problems of pleiotropy and reverse confounding.
            not tried in ch2 due to low power, power not evaluated for ch4

            Causal Inference Methods to Integrate Omics and Complex Traits
            \url{http://m.perspectivesinmedicine.cshlp.org/content/early/2020/08/17/cshperspect.a040493.abstract?s=09}

            % TODO: mediation can exist in the absence of a 'significant' total effect G -> pheno
            % http://davidakenny.net/cm/mediate.htm
            % In the opinion of most though not all analysts, Step 1 is not required. (See the Power section below why the test of c can be low power, even if paths a and b are non-trivial.)

    option: MR with expression 
        instrument variables are really just perfect mediators

        genetic instruments
        as demonstrated by e.g. qinqin, use (r)eqtl as genetic instruments

        e.g. sigeclec10
        we have eqtls for it, but a bad idea since indep

        does this approach require external datasets?

        MRLocus: identifying causal genes mediating a trait through Bayesian estimation of allelic heterogeneity
        \url{https://www.biorxiv.org/content/10.1101/2020.08.14.250720v2}

\subsection{Triangulating the truth}

% https://www.nature.com/articles/d41586-018-01023-3

unification
    this thesis only first step

    triangulation

    here's where multiomics comes in
    diff omics layers have diff biases

    more intermediates:
        The proteome: disconnect between regulatory layers
        % https://pubmed.ncbi.nlm.nih.gov/32709985/
        % https://www.nature.com/articles/s41576-020-0258-4
        % Although protein and mRNA levels typically show reasonable correlation, we describe how transcriptomics and proteomics provide useful non-redundant readouts.

        % e.g. anti-TNF does not imply reduction in TNF expression, but we see only downstream fx through regulation

    MOFA
    multiomics
        % https://twitter.com/Eric_Fauman/status/1266128178692718598?s=19
        Pqtls more accurate?

    already lots of expression data

    combining systems immunology studies of genetic arch of immune parameters
            e.g. dejager2015ImmVarProjectInsights, zalocusky201810000Immunomes
        giving gmore intermediate phenotypes
        layering of evidence (triangulation)
            molqtls (e.g. foreshadowing, pqtls), MR, CIT, coloc

    with GWAS of immune phenotypes

    reQTL
    ASE
    vqtls: e.g. % doi: https://doi.org/10.1101/2020.07.28.225730 

\end{outline}

