%
% Chapter 4
%
% multiPANTS

\chapter{multiPANTS: response to biologic anti-TNF therapy for CD}

% Collab note
% - PANTS collab
% - RNAseq gen and quant
% - Cell count estimates

\begin{outline}

\section{Introduction}

\subsection{\Glsfmtlong{CD} and \glsfmtlong{IBD}}
\todo{difficult to separate out CD from IBD in literature, section probably needs to decide to focus on CD only or IBD}

% \1 <Description>
\gls{CD} is a chronic, inflammatory disease of the gastrointestinal tract.
Characterised by patchy inflammation, where lesions are interspersed with regions of normal mucosa. 
The lesions can be distributed anywhere in the gastrointestinal tract, and tend to be transmural, affecting all layers of the gut wall.
\gls{CD} is one of the two forms of \gls{IBD}.
The second form is \gls{UC}, is characterised by continuous inflammation, with lesions that are superficial rather than transmural, and restricted to the colon\autocite{roda2020CrohnDisease}.
Whilst the two are distinct forms of \gls{IBD}, similarities in clinical presentation, available therapies, and genetic architecture mean \gls{CD} and \gls{UC} have often been historically studied together.
This introduction will consider literature on \gls{CD}, \gls{UC}, and \gls{IBD}.
% The similarity is such that there is a subset of IBD-U patients with features of both CD and UC, and thus is difficult to classify as one or the other.
\gls{IBD} is an \gls{IMID}, a group of related diseases involving immune dysregulation of common inflammatory pathways.
Other \glspl{IMID} include \gls{T1D}, \gls{SLE}, \gls{RA}, \gls{MS}, and psoriasis\autocite{cotsapas2013ImmunemediatedDiseaseGenetics,david2018GeneticsImmunemediatedInflammatory}.

% \1 <Pathogenesis and host genetics>
Pathogenesis of \gls{CD} is not completely understood, but involves interaction of the immune system, environmental factors (such as smoking, stress, diet \autocite{ananthakrishnan2015EpidemiologyRiskFactors,roda2020CrohnDisease}), and gut microbial factors in a genetically-susceptible individual \autocite{desouza2016ImmunopathogenesisIBDCurrent}.
Since the seminal discovery in 2001 that genetic variation in \gene{NOD2} was linked to \gls{CD} risk\autocite{todd2001TacklingCommonDisease},
much progress has been made in establishing the genetic architecture.
% (NOD2 was the first gene to be genetically-associated with a common disease)
% The first IBD GWAS was in 2006 \url{https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4410764/}
The most recent \gls{GWAS} studies catalogue over 240 risk loci for \gls{IBD} \autocite{delange2017GenomewideAssociationStudy}.
% \2 IBD GWAS identify loci that are in pathways that are drug targets \autocite{delange2017GenomewideAssociationStudy}
% \2 Genetic correlation between UC and CD is high \autocite{cotsapas2013ImmunemediatedDiseaseGenetics,david2018GeneticsImmunemediatedInflammatory}
Most associations are shared between \gls{CD} and \gls{UC}, with some heterogenenity of effects such as \gene{NOD2} being more strongly linked to \gls{CD} risk\autocite{jostins2012HostMicrobeInteractions,liu2015AssociationAnalysesIdentify}.
% \2 Concordance rates amongst monozygotic twins are higher for CD (~50\%) than for UC (\textapprox{15}\%) suggesting a greater heritable component \autocite{roda2020CrohnDisease}

% \1 <Epidemiology and burden>
\gls{IBD} has historically been considered a disease of the Western world.
The highest prevalence and incidence of new cases of CD are in North America and western Europe\autocite{roda2020CrohnDisease},
although disease burden is now rising in newly industrialised countries Asia, Africa and South America\autocite{kaplan2015GlobalBurdenIBD,alatab2020GlobalRegionalNational}.
% \3 Migrants from low- to high- prevelance regions are at higher CD risk, suggesting there is an influence of Western lifestyle on disease risk. \autocite{roda2020CrohnDisease}
The modal age of onset of \gls{CD} is typically between late adolescence and early adulthood.
The disease is progressive: within 20 years of diagnosis, 50\% of patients with CD will develop gastrointestinal tract complications and approximately 15\% require surgical intervention \autocite{roda2020CrohnDisease}.
Given the rising prevelance and large impact on quality of life, there is active research into effective therapies with the goal of inducing complete mucosal healing\autocite{levin2016MechanismActionAntiTNF,roda2020CrohnDisease}.

\subsection{Anti-TNF biologic therapies for \glsfmtlong{CD}}
\todo{the subsection on anti-TNFs in the intro chapter may be merged here}

% <What is TNF>
\gls{TNF}, also known by the archaic name \gls{TNF}-\textalpha, is a proinflammatory cytokine produced mainly by immune cells such as monocytes, macrophages, NK, T cells, and B cells.
It is synthesised in transmembrane form, then enzymatically cleaved into its soluble form.
\gls{TNF} binds to receptors TNFR1 and TNFR2; most cells in the body express one receptor or the other.
Binding triggers a signalling cascade that in different contexts, can regulate inflammation, apoptosis, cell proliferation and survival. \autocite{aggarwal2003SignallingPathwaysTNF,kalliolias2016TNFBiologyPathogenic,digby-bell2019InterrogatingHostImmunity}
In the context of \gls{IBD} pathogenesis, high \gls{TNF} levels in \gls{IBD} contribute to maintaining inflammation in the gut.
One current model involves \gls{TNF} promoting apoptosis of monocytes, macrophages, and gut epithelial cells via TNFR1, while inhibiting apoptosis of mucosal CD4+ T cells via TNFR2\autocite{levin2016MechanismActionAntiTNF,adegbola2018AntiTNFTherapyCrohn,digby-bell2019InterrogatingHostImmunity}.

The development of biologic anti-\gls{TNF} therapy has revolutionised \gls{CD} patient care in the last two decades.
Infliximab and adalimumab are the two major drugs approved for treatment of \gls{CD}, \gls{UC}, and a number of other \glspl{IMID}.
Both are IgG1 monoclonal antibodies that bind soluble and transmembrane \gls{TNF}, inhibiting interaction with its receptors\autocite{lichtenstein2013ComprehensiveReviewAntitumor,adegbola2018AntiTNFTherapyCrohn}
Two major mechanisms for how anti-TNF antibodies work have been proposed: induction of CD4+ cell apoptosis in the gut mucosa by inhibiting the \gls{TNF}-TNFR2 interaction; and binding of the tail (Fc) region of the antibody to Fc receptors on monocytes, inducing their differentiation into wound-healing M2 macrophages\autocite{levin2016MechanismActionAntiTNF}.
% Also see https://gut.bmj.com/content/69/6/1053.full
% Fcγ-receptor (FcγR) signalling induces IL-10 production in macrophages.
% Anti-TNF therapy induces FcγR-dependent IL-10 production in mouse and human macrophages in vitro.
% Anti-TNF therapy polarises human and mouse macrophages towards CD206+ macrophages in an IL-10-dependent manner in vitro.
% The efficacy of anti-TNF therapy in the T-cell transfer model of colitis depends on IL-10 signalling, specifically in macrophages and not T cells.
% \3 Fc regions fixing complement can also lyse cells expressing membrane TNF \autocite{adegbola2018AntiTNFTherapyCrohn}
% \2 Treatment leads to drop in circulating markers like CRP \autocite{levin2016MechanismActionAntiTNF}
Adalimumab is a human antibody, typically administered subcutaneously via auto-injector pen, with two initial doses aimed to induce remission, then a dose every two weeks to to maintain remission. 
Infliximab is a chimeric mouse-human antibody administered via intravenous infusion, with a three-dose induction and a dose every eight weeks for maintenance\autocite{adegbola2018AntiTNFTherapyCrohn}.
\todo{I assume dosing follows the induction and maintenance schedule from \autocite{adegbola2018AntiTNFTherapyCrohn}. Can't find anything about it in the PANTS protocol, although Sim confirmed the 2w/8w frequencies.}
Anti-\gls{TNF} biologics are among the drugs with the highest global spendings; adalimumab and infliximab ranked first and fourth respectively in 2017,
with an estimated combined spending of 29 billion USD\autocite{aitken2019GlobalUseMedicine}.

% The choice of whether to give ADA or IFX to \gls{CD} patients is currently made on the basis of \enquote{practical considerations regarding mode and frequency of administration}\autocite{lamb2019BritishSocietyGastroenterology}.
% Also see:
        % \3 2010 ECCO: "all currently available anti-TNF therapies appear to have similar efficacy and adverse-event profiles, so the choice depends on availability, route of delivery, patient preference, cost and national guidance.” \url{https://www.nature.com/articles/nrgastro.2015.135}, also see "Figure 2: Biologic agents in IBD: a proposed algorithm for clinical practice."
        % \3 \autocite{dhaens2011LondonPositionStatement}
        % \3 2018 <Infliximab and adalimumab for the treatment of Crohn's disease> \url{https://www.nice.org.uk/guidance/ta187}>: "1.2 Treatment as described in 1.1 should normally be started with the less expensive drug (taking into account drug administration costs, required dose and product price per dose). This may need to be varied for individual patients because of differences in the method of administration and treatment schedules."

\subsection{Treatment failure}

Anti-\gls{TNF} therapy is not always effective at treating \gls{CD}.
Various types of treatment failure can occur:
\gls{PNR} within the induction period (usually 12-14 weeks for adalimumab and infliximab), developing secondary \gls{LOR} during maintenance after an initial response, failure to achieve remission after the treatment course, and adverse events that lead to treatment stoppage\autocite{roda2016LossResponseAntiTNFs}.
The incidence of \gls{PNR} is 10-40\%, and the incidence of secondary \gls{LOR} is 24-46\% in the first year of treatment\autocite{ben-horin2014OptimizingAntiTNFTreatments,flamant2018InflammatoryBowelDisease,kennedy2019PredictorsAntiTNFTreatment}.
Another factor affecting treatment outcome is immunogenicity, the generation of antibodies against the drug, 
thought to increase the probability of treatment failure by increasing drug clearance rate\autocite{lichtenstein2013ComprehensiveReviewAntitumor,kennedy2019PredictorsAntiTNFTreatment}.
As a chimeric antibody, infliximab is more immunogenic than adalimumab\autocite{vermeire2018ImmunogenicityBiologicsInflammatory,kennedy2019PredictorsAntiTNFTreatment}.
Although remission with complete mucosal healing is the gold standard for successful treatment\autocite{roda2020CrohnDisease}, \gls{PNR} and \gls{LOR} can be defined much earlier in the treatment course, and help guide changes in treatment such as dose intensification or switching to a drug class with a different mechanism of action\autocite{lichtenstein2013ComprehensiveReviewAntitumor,ben-horin2014OptimizingAntiTNFTreatments}.

Anti-\gls{TNF} biologics are near the top of the therapeutic pyramid for \gls{CD}, among the treatment options with the highest toxicity and costs
\autocite{rogler2015WhereAreWe}.
% adegbola2018AntiTNFTherapyCrohn
% "Consensus guidelines recommend “step-up” treatment with initial trials of immunosuppressive treatment (corticosteroid and steroid sparing immunomodulators) prior to starting anti-TNF agents [8,9], although there is increasing evidence advocating for early institution of anti-TNF treatment [10], the “top-down” approach, especially in severe disease phenotypes [11].  In this review, we focus on the use of anti-TNF therapy in Crohn’s disease, its mechanism of action, role and future directions in therapeutic use."
The traditional approach to disease management is \enquote{step-up}, beginning at the bottom of the pyramid with steroids\autocite{adegbola2018AntiTNFTherapyCrohn}.
This may undertreat patients that require more aggressive therapy, allowing the disease time to progress.
An inverted approach is beginning at biologic therapies, then stepping-down the pyramid if possible.
This risks exposing patients to aggressive therapies they may not have needed\autocite{flamant2018InflammatoryBowelDisease}.
The best approach would of course be to predict whether a particular treatment will be effective for each patient, especially given the costliness and patient risks associated with therapies near the top of the pyramid.
Baseline predictors would be especially valuable for stratifying patients to specific therapies.

\subsection{Predicting response to anti-\glsfmtlongpl{TNF}}

Reported clinical variables with association to anti-\gls{TNF} response include age, disease duration, \gls{BMI}, smoking, \gls{CRP}, faecal calprotectin, drug and anti-drug antibody concentrations.
These have mostly been found in small retrospective cohorts, and are rarely independently validated\autocite{dhaens2011LondonPositionStatement,ding2016SystematicReviewPredicting,kopylov2016PredictingDurableResponse,flamant2018InflammatoryBowelDisease,digby-bell2019InterrogatingHostImmunity,noor2020PersonalisedMedicineCrohn}.
In \gls{PANTS}, a prospective study of infliximab and adalimumab response in \gls{CD} patients and the largest to date (n=1610),
baseline obesity, smoking, and greater disease activity was associated with low serum drug concentration after induction.
Low drug concentration was in turn associated with \gls{PNR} and non-remission, suggesting that immunogenicity may be mediating treatment failure\autocite{kennedy2019PredictorsAntiTNFTreatment}.

Studies have also looked for transcriptomic predictors for anti-\gls{TNF} response\autocite{digby-bell2019InterrogatingHostImmunity,noor2020PersonalisedMedicineCrohn}.
% "complete endoscopic and histologic healing"
In gut biopsies, expression of sets of \enquote{signature} genes were found to be predictive of mucosal healing after infliximab treatment in cohorts of \gls{UC} (\gene{TNFRSF11B}, STC1, PTGS2, IL13RA2, IL11; n=46\autocite{arijs2009MucosalGeneSignatures}) 
and \gls{CD} patients (TNFAIP6, S100A8, IL11, G0S2, S100A9; n=19\autocite{arijs2010PredictiveValueEpithelial}).
% \2 Halloran 2014 \url{https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4985265/} "molecular disturbance" aka PC1 seems to sep R and NR in Pathogenesis-based transcript sets (PBTs)
Expression of \gene{OSM} is associated with anti-\gls{TNF} response defined by improved Mayo score, a multiparameter clinical score of \gls{UC} activity (n=227)\autocite{west2017OncostatinDrivesIntestinal}.
Most recently, single-cell \gls{RNAseq} identified a module of IgG plasma cells, inflammatory mononuclear phagocytes, activated T cells, and stromal cells associated with clinical remission after anti-\gls{TNF} therapy in two separate \gls{CD} cohorts (total n=340)\autocite{martin2019SingleCellAnalysisCrohn}.

As obtaining blood samples is non-invasive, there has been great interest in finding transcriptomic predictors of response in blood.
% It was recently shown that the majority of the genes in a signature for response to anti-TNF in IBD patients show higher expression in immune cell subsets, compared to other cells present in the biopsy tissues, suggesting that resident or infiltrating leucocyte populations represent a good target to investigate responses to anti-TNF therapy. However, a clear cell population has not yet been implicated in the response (215).
% https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6434926/
%
% gaujoux2019CellcentredMetaanalysisReveals "Taken together, the majority of anti-TNF response signature genes are more highly expressed by immune cell subsets"
While blood is not the disease-relevant tissue for \gls{CD},
many of genes in gut biopsy signatures have high expression in infiltrating immune cells, so blood gene expression may capture the precursors of those cells\autocite{gaujoux2019CellcentredMetaanalysisReveals}.
Blood \gene{TREM1} expression has been identified as a marker of anti-\gls{TNF} response in two studies with inconsistent directions of effect.
\textcite{gaujoux2019CellcentredMetaanalysisReveals} defined response based on  
\enquote{clinical and/or endoscopic improvement}.
\gene{TREM1} expression was lower in infliximab responders in gut biopsies (total n=72),
but higher in responders in a separate cohort measuring whole blood expression (n=22).
\textcite{verstockt2019LowTREM1Expression} defined response based on endoscopic remission, reporting \gene{TREM1} to be a marker of response 
with lower expression in responders to infliximab and adalimumab in both gut biopsies and blood.
Proposed reasons for the disrepancy include small sample sizes, patient ethnicity, and different definitions of response \url{https://pubmed.ncbi.nlm.nih.gov/30007919/} \autocite{digby-bell2019InterrogatingHostImmunity}.
% \2 \autocite{salvador-martin2020GeneSignaturesEarly} SMAD7, TLR2 and DEFA5. response based on decrease in Pediatric Crohn’s Disease Activity Index (PCDAI) and Pediatric Ulcerative Colitis Activity Index (PUCAI). pIBD, IFX/ADA. n=31

Attempts have also been made to find genetic markers for response,
as they are baseline by definition, and may also help form mechanistic hypotheses of treatment response.
Anti-TNF response does not necessarily share the same genetic architecture as disease risk.
Variants in \gls{TNF}-regulated genes that are also associated with \gls{IBD} risk (\gene{NOD2}, TNFR1, TNFR2) are not associated with response to infliximab \autocite{digby-bell2019InterrogatingHostImmunity,noor2020PersonalisedMedicineCrohn}.
% IL-13 receptor (IL13Rα2), IL-23 receptor (IL23R), TNF-receptor I (TNFRI), IgG Fc receptor IIIa (FcYRIIIa), neonatal Fc receptor (VNTR2/VNTR3), apoptosis-related genes (Fas ligands, caspase 9), and MAP kinases\autocite{burke2018GeneticMarkersPredict}
A number of candidate gene studies have identified \gls{SNP} associations with response in genes such as apoptosis-related Fas ligand and caspase-9 that have yet to be validated \autocite{flamant2018InflammatoryBowelDisease,burke2018GeneticMarkersPredict}.
% Attempts have also been made to combine genetic and clinical markers\autocite{burke2018GeneticMarkersPredict}.
Recently, larger cohorts have enabled \glspl{GWAS} of anti-\gls{TNF} response in \gls{IBD}.
% The HLA-DQA1*05 allele, carried by approximately 40% of Europeans,
% significantly increased the rate of immunogenicity (hazard ratio [HR],
% 1.90; 95% confidence interval [CI], 1.60–2.25; P ¼ 5.88  10–13).
% Both drugs, and regardless of immunomodulator
% Dominant effect
In \gls{PANTS}, although no associations to \gls{PNR} were genome-wide significant\footnote{},
HLA-DQA1*05 carriage was associated with higher anti-drug antibody levels\autocite{sazonovs2019HLADQA105Carriage}, which may mediate related phenotypes such as \gls{LOR}.

% Overall, variation in technology used to measure gene expression, analysis method, anti-TNF drug, response definition (both criteria and time of assessment), tissue, sample size, disease among all these studies make a consensus hard to establish.
No algorithms using clinical, transcriptomic, or genetic markers for predicting \gls{IBD} patient response to anti-\gls{TNF} therapy have been translated to clinical practice,
although several markers are currently undergoing validation\autocite{noor2020PersonalisedMedicineCrohn}.

\subsection{Chapter summary}

% TODO in this ch, I describe E differences associated with 
\todo{this subsection}
\1 <context>: summary of above
    \2 conflicting results in the literature and need for validation for transcriptomic signatures for R/NR to anti-TNF 
\1 <content: our approach> 
    \2 use PANTS cohort data, the largest RNAseq dataset to date on CD patients with anti-TNF therapy, to define expression differences between PR/PNR
    \2 also able to evaluate evidence for genetic control of blood expression in CD patients
\1 <conclusion: our results> 
    \2 there are some baseline differences between R and NR, weak effects that require further validation
    \2 there are strong transcriptomic differences after the induction period (12 weeks) between R and NR
    \2 these differences are maintained over time up to w54, suggesting NR phenotype is stable over time and many doses
    \2 change from baseline to w14 for expression is magnified in R vs NR, suggesting there may be a continuum of response
    \2 TREM1 baseline signature not replicated
    \2 weak evidence of interaction of anti-TNF therapy with genetic control of expression (reQTLs)

\section{Methods}

\subsection{The PANTS cohort}

\gls{PANTS} is a prospective, observational, UK-wide cohort study of response to anti-gls{TNF} therapy in \gls{CD} patients, described in detail by \textcite{kennedy2019PredictorsAntiTNFTreatment}.
The study was registered with ClinicalTrials.gov identifier NCT03088449, and the protocol is available at \url{https://www.ibdresearch.co.uk/pants/}.
% "The eligibility criteria were as follows: age 6 years or older; diagnosis of Crohn’s disease involving the colon, the small intestine, or both; and active luminal disease supported by a C-reactive protein (CRP) of more than 3 mg/L 90 days before the first dose, faecal calprotectin of more than 50 μg/g between 90 days before and 28 days after first dose, or both."
In brief, total enrollment was 1610 patients, who were at least 6 years old, had active luminal \gls{CD}, and were naive to anti-gls{TNF} therapy.
Patients were invited to attend up to ten major study visits over a maximum follow-up period of three years, or until drug withdrawal.

The anti-\gls{TNF} drugs evaluated were adalimumab and infliximab\footnote{The study also evaluated infliximab biosimilars. Data from patients who received a biosimilar is not included in this chapter}.
All major visits were scheduled immediately prior to a drug dose.
Although adalimumab and infliximab have different dosing schedules, the timing of major visits was chosen such that the same visit structure could be used for patients on both drugs.
Additional visits could also be scheduled in case of secondary loss of response, or exit due to drug withdrawal.

% Some stats:
% "1241 patients were assessable at week 14. 
% Primary nonresponse occurred in 170 (21·9%, 95% CI 19·1–25·0) of 775 patients treated with infliximab and 125 (26·8%, 22·9–31·1) of 466 patients treated with adalimumab (table 2). 
% After excluding primary non-responders, the estimated proportion of infliximab-treated patients who had loss of response by week 54 was 36·9% (32·7–40·9), and for adalimumab was 34·1% (28·4–39·4; appendix pp 15–16). 
% At week 54, 469 (60·9%; 57·4–64·0) of 770 infliximab-treated patients were classified as being in non-remission, compared with 295 (66·9%; 62·3–71·3) of 441 adalimumab-treated patients (table 2)."

\subsection{Definition of timepoints}
\label{subsubsec:multiPANTS_timepoints_def}

%
% Visit info from PANTS protocol
%
% visit_1
%     This visit must occur in the 7 days prior to commencing anti-TNF medication, and preferably on the day of the first infusion / injection
% visit_1X
%     3 days after first dose of anti-TNF therapy - OPTIONAL
% visit 2 (not in phenotypes)
%     w12
%     The patient’s gastroenterologist will decide whether to continue with the next infusion / injection at week 14, or stop, anti-TNF therapy.
%     NB if after this visit, the anti-TNF drug is stopped, or switched to an alternative anti-TNF drug then the patient’s next visits will be Exit visit 1 (scheduled for week 14) and Exit visit 2 (scheduled for week 22, patients on IFX, week 28 patients on ADA)
% visit_3
%     w14
%     Immediately prior to IFX infusion, ADA injection
% exit_1
%     Exit Visit 1 (Withdrawal of anti-TNF therapy before week 54 or non-elective withdrawal after week 54).
    % If the drug is withdrawn, or the patient switched to another anti-TNF drug, two study exit visits will be scheduled. Exit visit 1 is scheduled for 8 weeks after the last IFX infusion or 2 weeks after the last ADA injection. Exit visit 2 is scheduled for 16 weeks after the last IFX infusion or ADA injection.
% lor_3
%     Loss of response (LOR) will be defined after week 13 by reviewing the following:
%         Initial response to anti-TNF defined at week 12.
%         Symptomatic inflammatory IBD activity, which is of sufficient severity and duration to warrant an escalation in steroid, immunomodulatory, or anti-TNF therapy or surgical resection as directed by the patient’s gastroenterologist.
%         Patients who lose response before week 13 will be categorised as primary non-responders
%     The LOR visit should be scheduled immediately prior (preferably on the same day) to the next (modified) anti-TNF infusion / injection and will replace any standard visits. In the event of anti-TNF dose intensification this visit may occur earlier than planned or may involve a higher dose of anti-TNF drug.
%     NOTE: this means visits still continue to be scheduled as long as the drug is not withdrawn
%         For IFX:
%             Conduct LOR 1 visit which will replace the standard visit.
%             Standard visits will continue to be scheduled for each subsequent IFX infusion up to week 54.
%         For ADA:
%             Conduct LOR 1 visit which may replace a standard visit if this coincides with the scheduled ADA research visits at week 30 and week 54
%             After the LOR1 visit subsequent visits are scheduled every 3 months for 12 months. These visits may coincide with the scheduled research visits 4, 5, 6, 7, 8 and 9. If this is the case simply use these visits and labels. If not then use LOR 2 or LOR ES visits and labels.
% lor_4
% visit_4
%     w30
%     Immediately prior to IFX infusion, ADA injection
% visit_5
%     w54
%     Immediately prior to IFX infusion, ADA injection
%
% AbbVie definition of w14 timepoint:
%
% Excluding visits labeled “visit_1X” (72 hour visit), we labeled the first follow-up visit after baseline as a “primary_response” visit if it occurred <= 20 weeks after baseline, in line with what was reported in the paper, and independent of whether it was labeled “visit_3” or “exit_1” or “lor_3”.
%

%
% "Each visit was scheduled just immediately prior to the next dose of either infliximab or adalimumab and so it should be a trough level.
% Ie. Infliximab is given as an IV every 8 weeks.  So last dose of infliximab should be 8 weeks prior to the visit
% Adalimumab is given every 2 weeks. So last dose of adalimumab is 2 weeks prior.
% The visits are definitely timed just prior to when patient would be receiving drug as the study was designed such that we capture patients at the point of drug infusion so they would not have to come to hospital for an unnecessary visit."
The expression data for this chapter comes from \gls{PANTS} samples centered around four timepoints from the first year of follow-up: week 0, week 14, week 30, and week 54.
These are the specified timings for four major visits in the first year.
Whole blood samples were taken prior to drug administration, and preserved for \gls{RNAseq} in Tempus Blood RNA Tubes.
The study day that sampling occurred relative to the first drug dose was recorded.

For the purpose of measuring the transcriptome at trough drug levels, I mapped samples from major and additional visits to four timepoints centered around the four major visits.
As it could not be guaranteed that visits occurred on the exact day specified in the protocol, I considered the visit windows defined by \textcite{kennedy2019PredictorsAntiTNFTreatment}: week 0 (week \numrange{-4}{0}), week 14 (week \numrange{10}{20}), week 30 (week \numrange{22}{38}), and week 54 (week \numrange{42}{66})
Samples were mapped according to the following criteria:
\begin{itemize}
    \item Major visit samples were mapped to the corresponding timepoint, regardless of whether they fell within the corresponding window i.e. an available week 0 sample is always mapped to the week 0 timepoint.
    \item Samples taken at additional loss of response or exit visits falling within one of the windows were mapped to that timepoint, unless the patient also had a major visit sample inside that window.
\end{itemize}
Only a small minority of major visit samples fell outside their corresponding windows.
\todo{compute maximum deviation}
Inclusion of samples from additional visits was important as they often replaced major visits for patients with primary non-response or loss of response.
\todo{Still discussing with Sim on the exact def of LOR and exit visits to decide whether this is sensible.}
Samples included under both criteria should be representative of trough drug levels,
as major visits and loss of response visits were always scheduled prior to a drug dose,
and exit visits were scheduled for when the next drug dose would have been.

\subsection{Definition of primary response and non-response}
\label{multiPANTS:PR_definition}

% "there is agreement that primary nonresponse to anti‐TNF drugs should not be assessed prior to 14 weeks following initial induction infusions with use of infliximab, or prior to 12 weeks for adalimumab injections" \autocite{dhaens2011LondonPositionStatement}
I used a definition of primary non-response based on the decision tree from \textcite{kennedy2019PredictorsAntiTNFTreatment}.
Primary non-response was assessed at week 12, prior to the week 14 drug dose. The criteria for primary non-response was \emph{either} of the following: 
\begin{itemize}
    \item exit for treatment failure before week 14 (e.g. as decided by physician global assessment), \emph{or} corticosteroid use at week 14 (a continuing or new prescription).
    \item compared to week 0, a decrease in \gls{CRP} by less than 50\% or to >\SI{3}{\milli\gram\per\litre}, \emph{and} a decrease in \gls{HBI} by less than 3 points or to >4.
\end{itemize}
In addition, the primary non-responders in the \gls{RNAseq} dataset were selected to exclude patients in remission at week 54.
As this is an observational study that continues until drug withdrawal, so a patient's clinician may decide to continue anti-gls{TNF} therapy even if a patient has primary non-response.

The definition of primary response was to not have primary non-response, and also have \gls{CRP} \SI{\le 3}{\milli\gram\per\litre} and \gls{HBI} < 4 by week 14.
Grey zone patients that were intermediate between primary response and non-response were not analysed.

% Selection criteria for samples from Nick:
%
% "~200 for each drug, ~100 PNR, 100 remission. PNR had to be PNR at week 14 and non-remission at week 54 (or unknown at week 54). 
% Remission had to have active disease at baseline and be in remission or response at week 14 and remission at week 54 (or unknown at week 54 and remission at week 30).
% ideal_downstream_cohort <- bd_vr_clin %>%
%   filter(
%     (crp_visit_1 >= 4 | calpro_visit_1 > 100),
%     (pnr & (is.na(remission_visit_5) | !remission_visit_5)) |
%     ((status_visit_2_3 %in% c("Response", "Remission")) & (remission_visit_5 | (is.na(remission_visit_5) & remission_visit_4)))
%   )
% They also had to be 16 years or over and have a baseline serum sample. 
% Within the infliximab patients, there was propensity matching between PNR/non-PNR based on on_imms_visit_1 + on_steroids_visit_1 + age_at_first_dose + earliest_weight_category_4 + albumin_visit_1 + sex."

\subsection{Library preparation and \glsfmtshort{RNAseq}}

% From Mark:
% Here is an extended version of the protocol, with highlighting of the portion on poly-A selection and subsequent depletion steps:
%
% RNA and DNA were isolated from whole blood samples collected in Tempus Blood RNA Tubes.
%
% The Applied Biosystems Tempus Blood RNA Tube and Tempus Spin RNA Isolation kit was adapted to work in concert with the Qiagen QIAsymphony PAXgene Blood RNA (762635) and DNA DSP Midi (937255) Kit protocol for total RNA and DNA isolation from Tempus blood RNA tubes.
%
% Day 1: Batches of 48 tempus tubes were removed from -80°C and scanned into the electronic isolation record. Sample blood tube barcodes were matched to shipping barcodes and arranged by subject ID in visit order. Sample blood tubes were transferred to 4°C to thaw overnight. In preparation for DNA isolation, 48 14mL polystyrene culture tubes (BD352051) were labeled with Genomic Technologies barcodes and scanned into the electronic isolation record alongside the corresponding sample IDs.
%
% Day 2: Sample blood tubes were removed from 4°C and inverted 10 times to ensure efficient lysis. Samples were then left at room temperature to equilibrate for 2 hours. To obtain RNA and DNA from the same blood samples the following steps were performed: (1) Prepared the blood tube samples by following the Applied Biosystems manufacturer’s protocol “Processing Stabilized Blood before Purification” steps 1-5 (Nunc 50mL conical tubes 52000-004, PBS 1x -Ca2+ -Mg2+ pH 7.2 (20012050). After centrifugation, 2mL of the supernatant was aliquoted into the 14mL barcoded culture tubes for DNA isolation and placed in the 4°C until ready for the DNA isolation protocol with the Qiagen QIAsymphony DNA DSP Midi kit. The remaining supernatant was poured off of the RNA pellet and allowed to briefly dry. The blood samples then followed the Qiagen QIAsymphony PAXgene blood RNA manufacturer’s protocol at step 3. The following QIAsymphony instrument protocol was performed for isolation of Total RNA, RNA Isolation PAX RNA CR22332 ID 2915. The RNA samples were eluted in an 80uL volume and stored at -80°C, UltraPure DNase/RNase Free Distilled Water (10977023). The QIAsymphony was then loaded with the necessary reagents and consumables to perform the DNA isolation protocol. The DNA blood samples were removed from 4°C and loaded onto the QIAsymphony. The following QIAsymphony instrument protocol was performed for isolation of DNA, DNA isolation Blood_1000_V7-DSP. The DNA samples were eluted in a 200uL volume and stored at -80°C. RNA and DNA nucleic acid quantification were performed with the ThermoFisher QuBit BR RNA (Q10211) and QuBit BR dsDNA (Q32853) kits respectively, following the manufacturer’s protocol. RNA integrity analysis was performed with the Agilent RNA ScreenTape assay (5067-5579, 5067-5577, 5067-5576) on the Agilent 4200 TapeStation following the manufacturer’s protocol. Results were uploaded into the electronic isolation record and used RNA and DNA normalization.
%
% RNA library preparation from total RNA was conducted following the manufacturer’s protocol for the Kapa mRNA HyperPrep Kit. Briefly, 250 ng of total RNA was enriched for mRNA using magnetic oligo-dT beads. The remaining RNA was then fragmented by magnesium under elevated temperature. After fragmentation RNA was depleted of rRNA and globin mRNA using the QIAseq FastSelect RNA Removal Kit by Qiagen. The depleted RNA then underwent first strand synthesis using reverse transcriptase and random primers. Combined second strand synthesis and A-tailing incorporated dUTP into the second cDNA strand for stranded RNA sequencing and added dAMP to the 3’ ends for adapter ligation. The cDNA fragments were then ligated to sequencing adaptors (IDT xGEN Dual Index UMI adapters) and was enriched using 16 cycles of PCR. Final libraries were assessed using the Agilent Tapestation and Qubit (ThermoFisher) assay methods then sequenced on an Illumina HiSeq 4000 sequencer using 2x75bp read length.

Total RNA was extracted from whole blood samples preserved in Tempus Blood RNA Tubes, following the Qiagen QIAsymphony instrument protocol (RNA Isolation PAX RNA CR22332 ID 2915.
RNA was then quantified with the ThermoFisher QuBit BR RNA (Q10211), RNA integrity was assessed with the Agilent RNA ScreenTape assay (5067-5579, 5067-5577, 5067-5576) on the Agilent 4200 TapeStation.

Library preparation was done using the Kapa mRNA HyperPrep Kit, including enrichment for mRNA using magnetic oligo-dT beads, depletion of rRNA and globin mRNA using the QIAseq FastSelect RNA Removal Kit, and adapter ligation with IDT xGEN Dual Index UMI adapters.
Libraries were sequenced on the Illumina HiSeq 4000 using 75 bp paired-end reads.

\subsection{RNAseq quantification and quality control}

A total of 1141 samples were initially sequenced.
Sequencing data was demultiplexed with Picard.
% "To demultiplex and annotate the reads with the UMI information we use the Picard commands outlined in the attached file. We align using STAR (https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3530905/ [ncbi.nlm.nih.gov]), mark and remove duplicates using UMITools (https://github.com/CGATOxford/UMI-tools [github.com]), and get our gene counts using featureCounts (http://subread.sourceforge.net/ [subread.sourceforge.net]). For QC we use fastQC, Picard, and compile the results together using MultiQC."
%
% "With the UMI protocol, raw fastq files are generated per sample/per lane, so there are 8 files per sample and they don’t actually contain the UMI information, so they are probably not ideal for starting your work. We should speak about where along the pipeline you are comfortable with us sending you the PANTS RNAseq
% Roughly the pipeline:
% Raw .bcl files -> demultiplexed unaligned .bam file per sample/per lane (8 files per sample) ->  fastq without UMI information per sample/per lane -> alignment -> aligned .bam files without UMI information per sample/per lane -> merge aligned .bam with unaligned .bam including UMI information -> merge .bam files for a sample across lane (1 file per sample) -> mark duplicates using UMIs -> create deduplicated .bam file -> create counts matrix and run QC -> optional creation of fastq files with UMIs in read names
Reads were mapped to GRCh38 using STAR (2.6.1d) and deduplicated to unique reads using UMI-tools.
Gene expression was quantified against the Ensembl 96 gene annotation using featureCounts (1.6.4).

Total number of read pairs, sequence quality, overrepresented sequences, adapter content and sequence duplication rates were checked using FastQC.
Samples were filtered to remove outliers (>2SD from the mean) according to percentage of aligned reads in coding regions reported by Picard, percentage of unique reads, and number of unique reads.
Samples that could not be mapped to a timepoint according to \autoref{subsubsec:multiPANTS_timepoints_def} were removed.
Samples that came from patients with 
sex mismatch based on Y chromosome gene expression,
grey zone primary response, 
or missing data for variables considered in the variable selection process (\autoref{subsubsec:multiPANTS_var_selection}),
were removed.
A total of 814 samples remained after filtering.

The Ensembl 96 gene annotation contains \SI{58884} genes, many of which are not expressed in whole blood.
Normalised library sizes were computed using edgeR::calcNormFactors(method='TMM') for use in CPM calculations.
Genes were filtered to require 
>1.25 CPM in >10\% of samples, where 1.25 CPM is approximately 10 counts at the median library size;
and non-zero expression in >90\% of samples.
Globin genes and short ncRNAs were removed.
A total of 15511 genes remained after filtering.
Finally, counts were converted to the log2 CPM scale and precision weights to account for the mean-variance relationship were computed for each gene and sample using voomWithDreamWeights.

\subsection{\glsfmtlong{DGE}}

\subsubsection{Variable selection by variance components analysis}
\label{subsubsec:multiPANTS_var_selection}

For each gene, the form of the \gls{DGE} regression model has gene expression as the response variable, 
predictor variables of interest (such as primary non-response, drug, and timepoint),
and other selected predictor variables.
Many variables are available 
Variables in \autoref{fig:multipants_pheno_filtered_ggcorrplot} were included as predictors.

\1 In estimating the effect X\textrightarrow Y, of predictor X on response Y by regression, 
adjustment for a third variable Z can increase, decrease, or even reverse the effect estimate.
The regression model is agnostic to what causal role Z may play,
but different types of third variable can be distinguished conceptually if some relationships are actually causal.
% TODO
% \url{https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2819361/} <<<
% \url{https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2254615/}
In this section, I focus on identifying third variables that are covariates for inclusion into the \gls{DGE} model:
% TODO: note this is the formal def of covariate
% TODO: read Cinelli2020-CrashCourseBadControls.pdf
where Z is associated with Y and explains some variation in Y,
and conditioning on Z increases the efficiency of estimating X \textrightarrow Y.
% In theory
conditioning on a covariate that is not associated with the predictor ($X \nleftrightarrow Z \leftrightarrow Y$, aka. a precision variable, prognostic variable in clinical trials lit) reduces variance of the estimate,
% by explaining some var that would other wise go to error

% TODO note we are not assigning any causality here, so back off in this paragraph with a caveat
\1 If covariates are also associated with the predictor X, issues can arise depending on their causal role.
% In theory
Conditioning on a confounder (X \textleftarrow Z \textrightarrow Y) reduces bias,
conditioning on a collider (X \textrightarrow Z \textleftarrow Y) induces bias,
and conditioning on a mediator in the causal path (X \textrightarrow Z \textrightarrow Y) changes the effect estimated by removing the indirect effect mediated by Z.

\1 Here, the predictors in question are primary response status, drug, and study visit; and the response variable is gene expression.
\1 Many phenotypes and technical variables are available as potential covariates in the \gls{PANTS} cohort;
\autoref{fig:multipants_pheno_corrplot} shows their correlations with each other, and the predictor variables.
% From Mark:
% As a follow-up, here is what Sam shared about estimating cell composition in the data: “We used the Houseman method [bmcbioinformatics.biomedcentral.com], which is implemented with the estimateCellCounts() function from the minfi package in R.”
These include proportions of six common cell types in whole blood, 
estimated using the Houseman method (\software{minfi::estimateCellCounts} \url{https://academic.oup.com/bioinformatics/article/30/10/1363/267584}) 
from whole blood Illumina MethylationEPIC data also collected for the same patients and timepoints.
% TODO: 1 sentence on houseman method

\begin{figure}
    \centering
    \includegraphics[width=1.0\textwidth,page=1]{mainmatter/figures/chapter_04/process_pheno.pheno_filtered_dge.ggcorrplot.pdf}
    \caption{Correlation matrix of phenotypes considered as independent variables in DGE and eQTL models.}
    \label{fig:multipants_pheno_filtered_ggcorrplot}
\end{figure}

% TODO: any need for plot?
\1 visualised main factors that influence global gene expression by PCA (\autoref{fig:multipants_dream_prcomp})
    \2 main separation along PC1 is w0 anti-TNF naive samples from all other post-drug start samples 
    \2 TODO: color other PCs by other variables: sex, response status, library prep protocol

% TODO: recolor to 1 page
\begin{figure}
    \centering
    \includegraphics[width=1.0\textwidth,page=1]{mainmatter/figures/chapter_04/dream.prcomp.pdf}
    \caption{top 12 expression PCs of filtered expression data}
    \label{fig:multipants_dream_prcomp}
\end{figure}

\1 A variance components analysis was conducted to formally quantify the fractions of variation in expression explained by known variables
using \software{variancePartition}\autocite{hoffman2016VariancePartitionInterpretingDrivers}, which fits a mixed regression model.
% TODO covariates chosen here are used in all genewise models, we need a unified set.
Variables in \autoref{fig:multipants_pheno_filtered_ggcorrplot} were included as predictors.
    % Univariable analyses showed the strongest associations
    % with primary non-response to infliximab and
    % adalimumab were with week 14 drug and anti-drug
    % antibody concentrations (table 3; appendix p 17). Primary
    % non-response to infliximab was also associated with
    % older age at first dose, smoking at baseline, non-use of an
    % immunomodulator at baseline, lower baseline albumin
    % concentrations, and higher baseline white cell count.
    % Primary non-response to adalimumab was associated
    % with a higher body-mass index at baseline.
    \2 Includes prognostic factors from \autocite{kennedy2019PredictorsAntiTNFTreatment}
\1 Additional categorical variables were included for patient, \gls{RNAseq} plate, and library prep protocol version.
An additional continuous variable consisting of random numbers drawn from the standard normal distribution was also included as a null.
Granulocyte proportion estimates were dropped to relieve perfect multicollinearity.
Categorical variables were coded as random effects, and continuous variables as fixed effects.
Surprisingly, \textcite{hoffman2016VariancePartitionInterpretingDrivers} showed that variance proportion estimates are unbiased even when coding categorical variables with as few as two categories as random, 
as long as all model parameters are estimated jointly using \gls{ML} rather than \gls{REML}\footnote{
    \gls{REML} treats random effects as nuisance parameters and estimates fixed effects after first integrating out random effects).
}.
It was also shown that this approach also avoids over-estimates of variance proportions that occur if categorical variables with many levels are treated as fixed.

% aim to identify covariates, but will inevitably hit other types

\1 Variables were ranked by median variance proportion across all genes (\autoref{fig:multipants_varPart}).
The variables that explained the most variance included patient, cell proportions and \gls{RNAseq} plate.
    \2 most influential on interpretation are cell counts: there are pros and cons to using them
% should be given    special consideration as likely mediatgor
    \2 Cell proportions explain a lot of variance: this is expected \url{https://genome.cshlp.org/content/early/2020/06/24/gr.256735.119.abstract?papetoc} and even more so as they change a lot over time (\autoref{fig:multipants_cell_type_proportion_vs_Study_Day})
    \todo{anova for each cell prop over time for p values}
    \2 in the case of mediation i.e. PNR -> CC -> R
    \2 Should rarely find cell prop to be a collider, as in most genes, E -> cell prop is unlikely vs cell prop -> E
    \2 so keep them in as covariates
        \3 it's already popular for diff meth, and in DGE, can increase robustness \url{https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1878-x}
    \2 note that we end up with the adjusted effect: upregulation in this context is increase in transcripts because making more per cell, not more in the bulk sample
        \3 this may not be ideal in prediction context, as each var input needs to be measured, and may even attenuate ability to predict \autocite{gaujoux2019CellcentredMetaanalysisReveals}
    \todo{don't know if it matters if there are colliders among covariates, since we can't estimate any causal effects in DGE due to lack of a control group?}

\begin{figure}
    \centering
    \includegraphics[width=1.0\textwidth,page=1]{mainmatter/figures/chapter_04/dream.plotVarPart.pdf}
    \caption{variance partition analysis, distribution of genewise \% variance in expression explained by each variable}
    \label{fig:multipants_varPart}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=1.0\textwidth,page=1]{mainmatter/figures/chapter_04/dream.cell_type_proportion_vs_Study_Day.pdf}
    \caption{changes in cell proportions of 6 immune cell types over time}
    \label{fig:multipants_cell_type_proportion_vs_Study_Day}
\end{figure}

\1 Variables that did not explain more variation on average than the null could still have high maximum values, 
    indicating their importance for specific genes only, such as genes with sex-specific expression.
    \2 would be best to customise per gene, but less comparable interpr between genes
    \2 Included, as we need a consensus set of covariates, and the penalty is only 1 df.
    \2 final list of covariates is [TODO: basically select all, except Gran, since we have proportions; and ever immunomodulator, which is low in median and max var explained]
    \todo{the var explained by Gran will be redistributed among highly cor vars anyways}
% Sex
% Age
% BMI
% Age of Onset
% Crohn’s Surgery
% Ever Immunomodulator
% Current Smoker
% Proportions of the 6 cell types: CD4+ T cells, CD8+ T cells, B cells, NK cells, monocytes, and granulocytes

\subsubsection{Contrasts for pairwise group comparisons}

\1 model form is E ~ [...] don't forget random fx!
\1 can we pool drugs with a mean term for this comparison? i.e. move from 8 to 4 means model
    \2 test interaction between drug and response at w0, and at w14 i.e. is there a diff in the diff between R vs NR between drugs?
    \2 no significant single gene hits at either timepoint
    \2 but unclear if preregistered power calculations would extend to this test \url{https://statmodeling.stat.columbia.edu/2018/03/15/need-16-times-sample-size-estimate-interaction-estimate-main-effect/}

    \2 there are baseline diffs are evident in clinical data \todo{because this is non-randomised, baseline differences do matter??}
    \2 “Several baseline characteristics were significantly different between the infliximab-treated and adalimumab-treated patients, including age, smoking, body-mass index, disease duration, disease location, and disease behaviour. Patients treated with infliximab had more active disease at baseline than did patients treated with adalimumab, as evidenced by higher serum CRP and faecal calprotectin concentrations (table 1).” \autocite{kennedy2019PredictorsAntiTNFTreatment}
    \2 I include many but not all of these covariates in the DGE model

\1 used \software{dream} hoffman2018DreamPowerfulDifferential
\1 Group-means parametrisation with 8 means
    \2 equiv to to 3 way interactions between drug/response/visit
    \2 no intercept, so each group coef is a mean estimate
\1 specific hypotheses tested using sum to zero contrasts 
\1 Model also fit that used Group-means parametrisation with 4 means: pooling the two drugs
    % TODO
    % The grand mean or pooled mean is the mean of the means of several subsamples, as long as the subsamples have the same number of data points.
    % \2 note this is not the same as the same as the mean of means contrast as pooled is not mean of two means
\1 TODO: model equations
    \2 for dream analysis, unlike variancePartition, use REML over ML, so use fixed effects for small numbers of levels
    \2 also, need fixed effects for tested predictors
    % However, in the lme4 package in R the standards for evaluating significance of
    % fixed effects in these models (i.e., obtaining p-values) are somewhat vague.
    % There are good reasons for this, but as researchers who are using these models
    % are required in many cases to report p-values, some method for evaluating the
    % significance of the model output is needed.
    %
    % The primary motivation for this omission is that in linear mixed models it is
    % not at all obvious what the appropriate denominator degrees of freedom to use
    % are, except perhaps for some simple designs and nicely balanced data.
    \2 to get p values for papers, Dream uses lmerTest approximation Satterthwaite df \url{https://link.springer.com/article/10.3758/s13428-016-0809-y} with REML
    \2 this combo controls type 1 error for n>144 in lmerTest simulations
    \2 FDR BH separately per comparison: "The default method="separate" and adjust.method="BH" settings are appropriate for most analyses. method="global" is useful when it is important that the same t-statistic cutoff should correspond to statistical significance for all the contrasts." \url{https://rdrr.io/bioc/limma/man/decideTests.html}

% TODO: confidence intervals for FDR https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-12-288
    % https://en.wikipedia.org/wiki/False_coverage_rate

\subsubsection{Spline model for difference over all timepoints}

% TODO there are methods, but did not use due to shortness of time series
% "Comparative analysis of differential gene expression tools for RNA sequencing time course data"
% Surprisingly, TC tools were outperformed by the classical pairwise comparison approach on short time series (<8 time points) in terms of overall performance and robustness to noise, mostly because of high number of false positives, with the exception of ImpulseDE2.

\1 aim is to uses info in samples from other timepoints, avoiding a large number of pairwise comparisons
    \2 a simple study day x responder interaction over time assumes linear change
\1 could aosl treat time as categorical visits (like baseline/w14 analysis above), then f test all interactions
    \2 but there is variation in study day in the visit windows

\1 spline model tests over all timepoints, are there diff trajectories for R and NR?
    \2 Internal Knots set at w14 and w30, since drug dose just after the visit, so slope should be allowed to change until next dose
    \2 cubic between knots, linear outside external knots
    \2 f test over 3 interaction terms between spline basis and study day
        \3 can include all data
        \3 TODO: read this for maths behind basis functions \url{https://bmcmedresmethodol.biomedcentral.com/articles/10.1186/s12874-019-0666-3}

        % FDR separate

% TODO
% f-test
% is a wald test
% https://support.bioconductor.org/p/6124/
% The above mentioned statistics are computed for every contrast for each gene. The eBayes() function computes one more useful statistic. The moderated F-statistic (F) combines the t-statistics for all the contrasts for each gene into an overall test of significance for that gene. The moderated F-statistic tests whether any of the contrasts are non-zero for that gene, i.e., whether that gene is differentially expressed on any contrast. The moderated-F has numerator degrees of freedom equal to the number of contrasts and denominator degrees of freedom the same as the moderated-t. Its p-value is stored as fit$F.p.value. It is similar to the ordinary F-statistic from analysis of variance except that the denominator mean squares are moderated across genes.

\1 TODO: clustering spline hits
    \2 note more accurate to use partial expression, but complicated for DREAM, so used unadjusted expression
    % center, no scale
    % pearson, which is scale invariant on centered data https://www.researchgate.net/publication/26290974_A_new_approach_for_clustering_gene_expression_time_series_data
    \2 Centroids defined by simple mean in each visit

less rudimentary

a more direct connection to spline model fit
e.g. clustering basis function results
e.g. predicted values of expression

how to account for levels of noise in expression data?
% TMixClust
% measurements come withintrinsic noise which makes their time series clustering a difficult task. Here, we show howto cluster such data with the package TMixClust. TMixClust is a soft-clustering methodwhich employs mixed-effects models with nonparametric smoothing spline fitting and is ableto robustly stratify genes by their complex time series patterns.
% \url{https://www.nature.com/articles/s41598-018-36135-3}

\subsubsection{Gene set analysis ranked}

% camera is developed to use mod t, but in practice ranks are comparable between t and z.std, even though dream says otherwise
% https://bioconductor.org/packages/release/bioc/vignettes/variancePartition/inst/doc/dream.html#dream-analysis
\1 TODO: grab tmod paragraph from ch2
    \2 Genes are ranked by their test statistics, meaning we are ranking by significance
    % TODO
    % other more sophisticated ranking methods
    % Minimum Significant Difference \url{https://link.springer.com/article/10.1186/s12859-017-1674-0#Abs1} \url{https://cran.r-project.org/web/packages/tmod/vignettes/tmod.pdf}
    % treat method \url{http://web.mit.edu/~r/current/arch/i386_linux26/lib/R/library/limma/html/ebayes.html}
        \3 practice ranks are comparable between t and z.std, even though dream says otherwise, very high spearman cor
        % TODO: add spearman rank corr
    \2 approx 8k genes in the gene set annotation for tmod

% TODO: add explanation of interaction term column
    % note interpretation depends on the original signs of effects

The gene sets used were \glspl{BTM} from\autocite{li2013MolecularSignaturesAntibody}
"DC" (from Chaussabel et al. 2008)

\subsection{Genotyping and genotype data preprocessing}

% sazonovs2019HLADQA105Carriage
% DNA was extracted from pretreatment blood samples from
% 1524 patients in the PANTS cohort and genotyping undertaken
% using the Illumina CoreExome microarray
%
% After quality
% control, 1323 patients remained in the study, of which
% 1240 had drug and anti-drug antibody level data available
% (Supplementary Figure 2).
%
% 7,578,947 variants with an information content metric score .4 were subsequently taken forward for analysis
%
% HLA types were imputed at 2- and 4-digit resolution for the
% following loci: HLA-A, HLA-C, HLA-B, HLA-DRB1, HLA-DQA1,
% HLA-DQB1, and HLA-DPB1.
%
% sex,
% drug type (infliximab or adalimumab), immunomodulator use,
% and the first within-sample principal component were
% included as covariates (Supplementary Table 2).

\1 Whole blood samples were collected into EDTA tubes for genotypeing at w0
\1 TODO: scan Alex's thesis for genotyping and QC details
% TODO: check no batch effects e.g. chip, which would necc an insample PCA
    \2 autosomal only
    \2 TODO describe strange limix behaviour that lead to deduping visits by patient in sample filtering
    % although it may seem possible to have duplicate patients
    %     following the stacking logic of TODOcite,
    %     it estimation of the alternative model log likelihood
    %     excluding just a 2 duplicates at w14
    %     makes LL alt much lower
    %     LRT returns false negatives
    %
    %     resulting p value distribution is not well behaved:
    %     large spike at 1
    %
    %     but the betas and stes remain comparable
    %     since we use mashr to get signif values,
    %     can keep these in or out.


\subsection{Computing genotype PCs}

\1 used weights from akt for 1000G to do projection \autoref{fig:multipants_genotype_akt_1000g_pca}
    % just cases used. do not use in sample PCs?
    % TODO: read
        % Efficient toolkit implementing best practices for
        % principal component analysis of population
        % genetic data
    % would use FlashPCA2 if wanted to do myself e.g. between chip batches
\1 chose top 5 PCs for use in eQTL model
    \2 there is less pop structure here than HIRD. in HIRD, i did the PCA myself, and found 4 significant PCs with tracy-widom
    \2 from EIGENSTRAT paper, results not sensitive to number of PCs anyway, as long as it is sufficient \url{https://www.nature.com/articles/ng1847}
    % This implies that EIGENSTRAT results are not sensitive to the number of axes of variation used, as long as there is a sufficient number of axes to capture true population structure effects.
    % We see that in each SNP category, results are virtually identical for K = 1, 2, 5 or 10.

\begin{figure}
    \centering
    \includegraphics[width=1.0\textwidth,page=1]{mainmatter/figures/chapter_04/pants_samples.sampleids_cleaned_to_lowercase.filtered.GRCh38.sorted.multiPANTS.projection_1000G_pca.pdf}
    \caption{projection of PANTS samples onto 1000G genotype PC axes}
    \label{fig:multipants_genotype_akt_1000g_pca}
\end{figure}

\subsection{reQTL analysis}

\1 same overall strat as ch3

% Alternative:
% In the ANCOVA approach, the whole focus is on whether one group has a higher mean after the treatment. It’s appropriate when the research question is not about gains, growth, or changes.

% https://www.theanalysisfactor.com/pre-post-data-repeated-measures/
% ANCOVA vs repeated measures vs mixed model

% The biggest advantage of mixed models is their incredible flexibility.  They can handle clustered patients as well as repeated measures (even in the same model).  They can handle crossed random effects, where there are repeated measures not only on an patient, but also on each stimulus.
% https://www.theanalysisfactor.com/repeated-measures-approaches/

% Then work out if genetics changes trajectories for any gene i.e. DGE models with a snp as predictor
% First need to eQTL scan in general with mashr and find the snps in the most reQTLish genes, since this modelling is probably expensive

\subsubsection{Finding hidden confounders in expression data}

\1 PEER (same as ch3)
    \2 Used DESeq2 vst for between sample norm e.g. sequencing depth 
    \2 Chose n PEER to maximise cis-eQTLs on chr1 \autoref{fig:multipants_reqtl_PEER_k_choice} 

\begin{figure}
    \centering
    \includegraphics[width=1.0\textwidth,page=1]{mainmatter/figures/chapter_04/count_eGenes.signif_eGenes_vs_PEER_n.dataset_multiPANTS.chr_1.pdf}
    \caption{number of eGenes on chr1 used to choose number of PEER factors for each timepoint}
    \label{fig:multipants_reqtl_PEER_k_choice}
\end{figure}

\subsubsection{Computing GRMs}

\1 LDAK, same as ch3

\subsubsection{limix model}

\1 same as ch3, limix
    \2 n at each timepoint was [...]
    \2 AC thresh 15
    \2 extra filter to avoid small numbers of minor hom, as without sufficient numbers, leads to data points with high leverage that may be unduely influential on the beta
    % despite mashr migitation by shrinkage? is it sufficient?
    % when the whole point is calling signif changes in beta
    \2 used a >5 min hom filter
        % gated by lowest n among 4 visits, but 15 to 25 MAF diff only

\1 find lead eQTL for each gene in any condition by lowest lfsr
    \2 breaking ties by highest imputation INFO, highest \gls{MAF}, shortest dist to \gls{TSS}, and genomic coordinate.

\subsubsection{mashr joint analysis}

\1 same as ch3
    \2 TODO describe mashr bug for negative betas
    \2 reQTLs defined by difference in betas test between timepoints
    \2 BH FDR separate per comparison, not globally

\subsubsection{Clustering reQTLs}

% TODO
% Clustering time-course Microarray data using functional Bayesian innite mixture model
% clustering folder in downloads
% https://2-bitbio.com/2017/04/clustering-rnaseq-data-making-heatmaps.html
% https://hbctraining.github.io/DGE_workshop/lessons/08_DGE_LRT.html

\1 <pipeline>
    \2 align
    \2 Centering, no scaling
        \3 ensure comparability between gene
        \3 Amplifies noise? Migitate by prefiltering on nominal signif diff between two timepoints
    % https://hdbscan.readthedocs.io/en/latest/comparing_clustering_algorithms.html
    \2 dist\_cor(method='pearson')
    \2 fastcluster::hclust(method='complete')
    \2 distance metric 1-cor(pearson)
    % •	https://www.rdocumentation.org/packages/NbClust/versions/3.0/topics/NbClust
    % o	Possible statistics: the index to be calculated. This should be one of : "kl", "ch", "hartigan", "ccc", "scott", "marriot", "trcovw", "tracew", "friedman", "rubin", "cindex", "db", "silhouette", "duda", "pseudot2", "beale", "ratkowsky", "ball", "ptbiserial", "gap", "frey", "mcclain", "gamma", "gplus", "tau", "dunn", "hubert", "sdindex", "dindex", "sdbw", "all" (all indices except GAP, Gamma, Gplus and Tau), "alllong" (all indices with Gap, Gamma, Gplus and Tau included).
    \2 Number of clusters: gap stat fviz\_nbclust

\subsubsection{gprofiler}

\section{Results}
\todo{what to put in results vs discussion. going with the pattern of providing enough info for the reader to intepret the data in the results, then doing a summary and my own interpretation in the discussion}

\subsection{Longitudinal \glsfmtshort{RNAseq} data from the \gls{PANTS} cohort}

To define transcriptomic differences between primary responders and non-responders to anti-gls{TNF} therapy in the \gls{PANTS} cohort, 
I analysed whole blood \gls{RNAseq} gene expression measured at up to four timepoints per patient:
week 0 baseline before commencing anti-\gls{TNF} therapy, and weeks 14, 30 and 54 after commencing anti-\gls{TNF} therapy.
\todo{Some primary non-responders have loss of response samples. Not sure why.}
After quality control, expression data was available for 15584 genes and 814 samples (\autoref{fig:multipants_studyDay_boxplots}).
These samples come from 324 patients, with a median of three samples per patient (\autoref{fig:multipants_visits_upset}).

Patient characteristics are shown in \autoref{tab:multipants_table1}.
The proportion of primary non-responders is high (43.8\%) compared to the overall proportion in the \gls{PANTS} cohort (23.8\%, \autocite{kennedy2019PredictorsAntiTNFTreatment}).
This is due to sample selection for \gls{RNAseq} to balance the sample size for each combination of drug and primary response status.

\begin{figure}
    \centering
    \includegraphics[width=1.0\textwidth,page=1]{mainmatter/figures/chapter_04/process_pheno.pheno_filtered_dge.Study_Day_vs_Visit_Label.pdf}
    \caption{Distribution of samples among defined study visit windows. lor and exit are additional visits that fall into the windows.}
    \label{fig:multipants_studyDay_boxplots}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=1.0\textwidth,page=1]{mainmatter/figures/chapter_04/process_pheno.pheno_filtered_dge.Visit_Label_upset.pdf}
    \caption{}
    \label{fig:multipants_visits_upset}
\end{figure}

% TODO: add caption and label after finalising
% Table 1. 
\input{mainmatter/figures/chapter_04/process_pheno.pheno_filtered_dge.table1.tex}

\subsection{Baseline gene expression associated with response}

% TODO:  increased expression vs upreg: make this all consistent

Patient primary response to anti-\gls{TNF} was defined after the induction period, between week 12 and week 14 according to the clinical decision algorithm from \textcite{kennedy2019PredictorsAntiTNFTreatment} described in \autoref{multiPANTS:PR_definition}, 
which integrates clinician assessment with change in \gls{CRP} level and \gls{HBI} score.
To identify differences in baseline gene expression associated with future primary response, 
I fit gene-wise linear models at 15511 genes, comparing week 0 gene expression in primary responders with primary non-responders.
Comparisons were performed both within infliximab-only and adalimumab-only subgroups, and with both drugs pooled.
Models were run both adjusting for cell composition estimates of six immune cell types, and without adjustment.
Throughout this section, the significance threshold was set at FDR < 0.05 for each comparison, and upregulation refers to increased expression in primary responders versus non-responders.

Without adjusting for cell composition, the largest effects were infliximab-only, with 859 genes differentially expressed.
Only \gene{KCNN3} (log2FC=\num{-0.843455423}) was significant for the adalimumab-only comparison, 
% logFC      AveExpr         t      P.Value   adj.P.Val     z.std
% 0.34597454 6.55605 4.9264804 1.094745e-06 0.016980595 4.8737951
% 0.31379161 6.55605 4.6913403 3.384743e-06 0.050291627 4.6459748
and only \gene{SIGLEC10} (log2FC=\num{0.34597454}) was significant in the pooled analysis (\autoref{fig:multipants_dge_volcano_week_0_R_N}).

After adjustment for cell composition, there were no longer any significant genes in the infliximab-only analysis, 
with 856/859 genes that were significant before the comparison having a dampened effect size after correction (smaller absolute effect and same sign), suggesting many effects may be mediated by cell composition.
\gene{SIGLEC10} in the combined analysis was also non-significant after adjustment (adjusted log2FC=\num{0.31379161}, FDR=\num{0.050291627}).
Conversely, at three genes downregulated in the adalimumab-only analysis that were the only significant genes post-adjustment, I observed increased significance:
%       logFC  AveExpr          t      P.Value  adj.P.Val      z.std
% -0.33495771 2.569096 -4.5040835 7.723531e-06 0.05989985 -4.4727077
% -0.35115926 2.569096 -5.0627658 5.212334e-07 0.00677703 -5.0183272
\gene{PDIA5} (unadjusted log2FC=\num{-0.33495771}, adjusted log2FC=\num{-0.35115926}), 
% -0.843455423 -0.2360221 -4.7304059 2.689467e-06 0.04171633 -4.69321336
% -0.879776701 -0.2360221 -4.9610155 8.738353e-07 0.00677703 -4.91811062
\gene{KCNN3} (\num{-0.843455423}, \num{-0.879776701}), 
% -1.1529588 1.169468 -4.212961 2.854891e-05 0.1476074 -4.184743
% -1.2231591 1.169468 -4.479679 8.738258e-06 0.0451797 -4.446249
and \gene{IGKV1-9} (\num{-1.1529588}, \num{-1.2231591}).

To identify coordinately up and downregulated gene sets and increase sensitivity for detecting differences between primary responders and non-responders,
I performed ranked gene set analysis in on the gene-wise z-statistics using blood transcriptomic modules: annotated sets of coexpressed genes in peripheral whole blood from \textcite{li2013MolecularSignaturesAntibody} (prefixed \enquote{LI}).
This module-level analysis was also run unadjusted (\autoref{fig:multipants_dge_panelPlot_week_0_R_N_cellPropF}) and adjusted for cell composition (\autoref{fig:multipants_dge_panelPlot_week_0_R_N_cellPropT}).

Despite only \gene{SAMD10} having a significantly different effect between drugs at the gene-level (an interaction between drug and response at week 0), 
the large global differences observable in \autoref{fig:multipants_dge_volcano_week_0_R_N} were detected in the module-level analysis\footnote{It is likely the study is not powered to detect gene-level three-way interaction effects between timepoint, drug and response, although I am not aware of which subgroup analyses may have been prespecified during the study design and sample size calculations for the \gls{PANTS} \gls{RNAseq} cohort.}.
Without adjusting for cell composition, many of the most significantly upregulated modules in the pooled analysis, 
including upregulation of monocyte (LI.M11.0, LI.S4), neutrophil (LI.M37.1, LI.M11.2), and dendritic cell (LI.M165, LI.S11),
appear to be driven by infliximab.
These modules have heavily reduced significance after adjusting for cell composition.
The new modules that are most upregulated in the pooled analysis after adjustment have more consistent effects between drugs, such as
MHC-TLR7-TLR8 cluster (LI.M146), antigen presentation (LI.M71, LI.M95.0), and myeloid cell enriched receptors and transporters (LI.M4.3).

For downregulated modules before adjustment, I observed infliximab-specific effects for for NK cell (LI.M7.2) and T cell (LI.M7.0, LI.M7.1) modules.
Adalimumab-specific effects were observed for plasma cell, B cell and immunoglobulin modules (LI.M156.0, LI.M156.0, LI.S3); and cell cycle and transcription modules (LI.M4.0, LI.M4.1).
After adjustment, the significance of infliximab-specific modules was reduced, 
but the significance of adalimumab-specific modules and the corresponding interaction effects was increased.

\begin{figure}
    \centering
    \includegraphics[width=1.0\textwidth,page=1]{mainmatter/figures/chapter_04/plot_gene_set_enrichment.dge_result_volcano_simple_C_1RI_1NI,C_1RA_1NA,C_1R_1N.pdf}
    \caption{DGE volcano plot for PR vs PNR at week 0}
    \label{fig:multipants_dge_volcano_week_0_R_N}
\end{figure}

% \begin{figure}
%     \centering
%     \includegraphics[width=1.0\textwidth,page=1]{mainmatter/figures/chapter_04/dream.E_vs_Study_Day.GENEID_ENSG00000142512.SYMBOL_SIGLEC10.pdf}
%     \caption{Unadjusted, normalised SIGLEC10 expression over time}
%     \label{fig:multipants_dge_SIGLEC10}
% \end{figure}

\begin{figure}
    \centering
    \includegraphics[width=1.0\textwidth,page=1]{mainmatter/figures/chapter_04/plot_gene_set_enrichment.tmodCERNO_panelplot_reversed_C_1RI_1NI,C_1RA_1NA,C_(1RI_1NI)_(1RA_1NA),C_1R_1N.cell_prop_correction_FALSE.pdf}
    \caption{Panel plot of module enrichent analysis for PR vs PNR at week 0. Length of bar is effect size, shade is FDR. Blue is upreg in R, red is downreg in R.}
    \label{fig:multipants_dge_panelPlot_week_0_R_N_cellPropF}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=1.0\textwidth,page=1]{mainmatter/figures/chapter_04/plot_gene_set_enrichment.tmodCERNO_panelplot_reversed_C_1RI_1NI,C_1RA_1NA,C_(1RI_1NI)_(1RA_1NA),C_1R_1N.cell_prop_correction_TRUE.pdf}
    \caption{Panel plot of module enrichent analysis for PR vs PNR at week 0. Length of bar is effect size, shade is FDR. Blue is upreg in R, red is downreg in R.}
    \label{fig:multipants_dge_panelPlot_week_0_R_N_cellPropT}
\end{figure}

\subsection{Assessing previously reported baseline predictors of response}

In addition to hits from this study, \autoref{fig:multipants_dge_volcano_week_0_R_N} is annotated with genes whose expression in gut biopsies or blood have been tested as baseline predictors of primary response in the literature\autocite{arijs2009MucosalGeneSignatures,arijs2010PredictiveValueEpithelial,verstockt2019LowTREM1Expression,salvador-martin2020GeneSignaturesEarly}.
Some genes expressed in gut mucosa (e.g. \gene{IL13RA2}) were not appreciably expressed in this whole blood dataset, 
and most other genes that were expressed were not significantly differentially expressed.
Only \gene{TNFRSF1B} and \gene{PTGS2} were associated with primary response, specifically in the infliximab-only comparison unadjusted for cell composition.

A previously identified marker in blood \gene{TREM1}, found to have opposing in two studies from \autocite{gaujoux2019CellcentredMetaanalysisReveals,verstockt2019LowTREM1Expression}
was not significantly associated with response in this study neither
before (logFC=\num{0.29283535}, FDR=\num{0.0595323})
%      logFC  AveExpr          t    P.Value adj.P.Val      z.std
% 0.29283535 7.668094  2.8665957 0.00429233 0.0595323  2.8558388
nor after adjusting for cell composition (logFC=\num{0.04625626}, FDR=\num{0.9864125}).
% 0.04625626 7.668094  0.6661333 0.50560221 0.9864125  0.6657010

% TODO: add effect sizes from those studies

% TODO: confint not implemented in variancePartition for topTable on dream objects, and multiplicity adjustment is non-trivial, see \url{https://amstat.tandfonline.com/doi/abs/10.1198/016214504000001907}

% \begin{figure}
%     \centering
%     \includegraphics[width=1.0\textwidth,page=1]{mainmatter/figures/chapter_04/dream.E_vs_Study_Day.GENEID_ENSG00000124731.SYMBOL_TREM1.pdf}
%     \caption{Unadjusted, normalised TREM1 expression over time}
%     \label{fig:multipants_dge_TREM1}
% \end{figure}

\subsection{Post-induction gene expression associated with response}

The same methodology applied at week 0 was applied at week 14 to identify differences in post-induction expression associated with primary response.
A larger proportion of the transcriptome is differentially expressed at week 14: 
1364 for the infliximab-only comparison, 
1544 for the adalimumab-only comparison, 
and 4841 pooling both drugs (\autoref{fig:multipants_dge_volcano_week_14_R_N}).
No significant interactions between drug and primary response were detected at the gene-wise level.
Given that sample sizes at week 0 and week 14 are comparable (\autoref{fig:multipants_studyDay_boxplots}), the overall signal is much stronger than at baseline.

Adjusting for cell proportions, 1320/1367, 1515/1544 and 4653/4841 have dampened effects;
and the numbers of significant genes drop to 379, 177, and 1302 for infliximab, adalimumab, and pooled analyses respectively.
This again suggests many effects are mediated by differences in immune cell composition between primary responders and non-responders.

% TODO: add module names
% TODO: add up reg modules
Modules including generic immune activation, monocytes, TLR and inflammatory signalling, and neutrophils were downregulated in primary responders; 
whereas B cell and plasma cell modules were upregulated \autoref{fig:multipants_dge_panelPlot_week_14_R_N_cellPropF}.
These modules remained differentially expressed with the same direction of effect after adjusting for cell composition \autoref{fig:multipants_dge_panelPlot_week_14_R_N_cellPropF}, 
suggesting there is per-cell up or downregulation on top of abundance changes of the cell types expressing these modules.
Modules related to antigen presentation (LI.M71, LI.M97.0, LI.M5.0),
interferon (LI.M75, LI.M127, LI.M111.1),
and dendritic cells (LI.M64, LI.M165)
also appear among significantly downregulated modules after cell proportion adjustment.
Directions of effect for the most significant modules were largely consistent between drugs.
The significance of drug by response interaction effects compared to main effects is less prominent 
than in the baseline analysis, where many of the strongest effects were driven by one drug.

\gene{SIGLEC10} from the baseline analysis retains its significant association with primary response post-induction, 
with the same direction of effect (adjusted logFC=\num{0.36610032}).
\todo{highlight a few more individual genes specific to this analysis too? not sure how to pick them at the moment.}
Some genes previously proposed as baseline markers of response in gut mucosa: \gene{G0S2}, \gene{TNFAIP6}, \gene{S100A8} and \gene{S100A9} by \autocite{arijs2010PredictiveValueEpithelial}; and \gene{OSM} by \autocite{west2017OncostatinDrivesIntestinal},
were differentially expressed in post-induction blood in this study.
The direction of effect in both cases, downregulation of markers in primary responders, also matches this study.

\begin{figure}
    \centering
    \includegraphics[width=1.0\textwidth,page=1]{mainmatter/figures/chapter_04/plot_gene_set_enrichment.dge_result_volcano_simple_C_3RI_3NI,C_3RA_3NA,C_3R_3N.pdf}
    \caption{DGE volcano plot for PR vs PNR at week 14}
    \label{fig:multipants_dge_volcano_week_14_R_N}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=1.0\textwidth,page=1]{mainmatter/figures/chapter_04/plot_gene_set_enrichment.tmodCERNO_panelplot_reversed_C_3RI_3NI,C_3RA_3NA,C_(3RI_3NI)_(3RA_3NA),C_3R_3N.cell_prop_correction_FALSE.pdf}
    \caption{Panel plot of module enrichent analysis for PR vs PNR at week 14, adjusted for cell comp. Length of bar is effect size, shade is FDR. Blue is upreg in R, red is downreg in R. top 20 by max}
    \label{fig:multipants_dge_panelPlot_week_14_R_N_cellPropF}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=1.0\textwidth,page=1]{mainmatter/figures/chapter_04/plot_gene_set_enrichment.tmodCERNO_panelplot_reversed_C_3RI_3NI,C_3RA_3NA,C_(3RI_3NI)_(3RA_3NA),C_3R_3N.cell_prop_correction_TRUE.pdf}
    \caption{Panel plot of module enrichent analysis for PR vs PNR at week 14, adjusted for cell comp. Length of bar is effect size, shade is FDR. Blue is upreg in R, red is downreg in R. top 20 by max}
    \label{fig:multipants_dge_panelPlot_week_14_R_N_cellPropT}
\end{figure}

\subsection{Magnification of expression change from baseline to post-induction in responders}

Given the stronger differences in expression between primary responders and non-responders at week 14 versus week 0,
I estimated the change in expression from week 0 to week 14 within the two groups, and also estimated the timepoint by response interaction.
I perform only the pooled comparison here both to simplify the analysis, and because similarly to the within week 14 comparison, change from week 0 to week 14 was relatively consistent between drugs, with exceptions noted.

Without adjusting for cell proportions,
12862 genes were differentially expressed in primary responders at week 14 vs week 0 in the pooled analysis,
8310 genes in primary non-responders,
and 6320 genes had a significant interaction.
After adjusting for cell proportions, 
5572 genes were differentially expressed in primary responders,
626 genes in primary non-responders,
and 179 genes had a significant interaction.
Of the genes differentially expressed between week 14 and week 0 in both primary responders and non-responders,
and with a significant interaction between timepoint and response, 
nearly all (4885/4891 unadjusted for cell composition, 31/32 adjusted) were magnified by primary response,
such that the same genes have larger fold-changes in the same direction for primary responders (\autoref{fig:multipants_dge_logFC_C_3R_1R_vs_logFC_C_3N_1N}).

\begin{figure}
    \centering
    \includegraphics[width=1.0\textwidth,page=1]{mainmatter/figures/chapter_04/plot_gene_set_enrichment.logFC_C_3R_1R_vs_logFC_C_3N_1N.pdf}
    \caption{}
    \label{fig:multipants_dge_logFC_C_3R_1R_vs_logFC_C_3N_1N}
\end{figure}

% NOTE: choosing these to be present in both drugs and pooled
The most significant modules that change from week 0 to week 14 in responders included
upregulation of B cell (LI.M47.0), plasma cell (LI.M156.0), and T cell activation (LI.M7.1);
and downregulation of immune activation (LI.M37.0), monocyte (LI.M11.0), neutrophil (LI.M37.1) and TLR and inflammatory signalling (LI.M16) modules (\autoref{fig:multipants_dge_panelPlot_week_14_0_R_N_cellPropF}).
Many of these are the same modules associated with response within the week 0 and week 14 timepoints.

Adjusting for cell composition decreases the significance of a majority of modules (\autoref{fig:multipants_dge_panelPlot_week_14_0_R_N_cellPropT}),
with T cell modules in the adalimumab-only analysis especially decreased.
Magnification is also observed at the module level, with nearly all module effects aligned in the same direction in responders and non-responders, with significant interactions also in the same direction.

\begin{figure}
    \centering
    \includegraphics[width=1.0\textwidth,page=1]{mainmatter/figures/chapter_04/plot_gene_set_enrichment.tmodCERNO_panelplot_reversed_C_3R_1R,C_3N_1N,C_(3R_1R)_(3N_1N).cell_prop_correction_FALSE.pdf}
    \caption{Panel plot of module enrichent analysis for PR vs PNR for week 14 - week 0 change. Length of bar is effect size, shade is FDR. Blue is upreg in R, red is downreg in R.}
    \label{fig:multipants_dge_panelPlot_week_14_0_R_N_cellPropF}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=1.0\textwidth,page=1]{mainmatter/figures/chapter_04/plot_gene_set_enrichment.tmodCERNO_panelplot_reversed_C_3R_1R,C_3N_1N,C_(3R_1R)_(3N_1N).cell_prop_correction_TRUE.pdf}
    \caption{Panel plot of module enrichent analysis for PR vs PNR for week 14 - week 0 change. Length of bar is effect size, shade is FDR. Blue is upreg in R, red is downreg in R.}
    \label{fig:multipants_dge_panelPlot_week_14_0_R_N_cellPropT}
\end{figure}

\subsection{Interferon modules with opposing differential expression in responders and non-responders}
\label{subsec:multipants_dge_opposing_interferon}

\autoref{fig:multipants_dge_logFC_C_3R_1R_vs_logFC_C_3N_1N}) also contained genes that were downregulated from week 0 to week 14 in responders, but upregulated in non-responders.
I extended my module analysis to also considering modules from \autocite{chaussabel2008ModularAnalysisFramework} in \software{tmod} (prefixed \enquote{DC}).
Although these modules are on the whole poorly annotated compared to modules from \autocite{li2013MolecularSignaturesAntibody}, interferon modules are annotated.
\gene{STAT2}, \gene{GBP5}, and \gene{PARP14} are annotated into an interferon module (DC.M3.4, tmodHGtest FDR=\num{0.0001261719}).
\gene{IFIT3} and \gene{GBP2} are also annotated into separate interferon modules (DC.M1.2, DC.M5.12).
Adjusted for cell composition, these modules are significantly upregulated at week 14 in non-responders only: 
DC.M3.4
FDR=\num{3.446797e-21},
DC.M1.2
FDR=\num{9.492256e-16},
DC.M5.12
FDR=\num{1.355467e-13}
(\autoref{fig:multipants_dge_evidencePlots_C_3_1_Interferon}).
Similar opposing effects after cell composition adjustment were also observed in \autocite{li2013MolecularSignaturesAntibody} modules for
antiviral interferon signature (LI.M175), type I interferon response (LI.M127), and antigen presentation (LI.M95.0) (\autoref{fig:multipants_dge_panelPlot_week_14_0_R_N_cellPropT}).

\begin{figure}
    \centering
    \includegraphics[width=1.0\textwidth,page=1]{mainmatter/figures/chapter_04/plot_gene_set_enrichment.evidencePlots_C_3_1_Interferon.pdf}
    \caption{}
    \label{fig:multipants_dge_evidencePlots_C_3_1_Interferon}
\end{figure}

\subsection{Sustained expression differences between primary responders and non-responders during maintenance}

As \gls{PANTS} is an observational study, it was able to include some patients who continued with anti-\gls{TNF} therapy
even after meeting the definition of primary non-response at week 14.
For both primary responders and non-responders, expression data could also be available from blood samples around week 30 and week 54, and at additional visits scheduled in the event of secondary LOR.

I fit a natural cubic spline to the expression of each gene as a function of study day, and tested for general differences in expression over time between primary responders and non-responders.
This analysis was done only with drugs pooled due to lower sample sizes for later timepoints.
Without adjusting for cell composition, 
4426 genes were differentially expressed between primary responders and non-responders;
210 genes were differentially expressed after adjustment.
To identify distinct trajectories of expression over time, I hierarchically clustered those 210 genes by
their mean expression in responders and non-responders at each timepoint, 
and determined the optimal number of clusters by the gap statistic method (\autoref{plot_gene_set_enrichment.spline_fviz_nbclust_gap_stat.pdf}).
Six distinct clusters were proposed (\autoref{fig:multipants_dge_spline_cluster_trajectories}).

Many of these genes had previously been identified as having significant differences in expression between responders and non-responders
either within week 14, or for change in expression from week 0 to week 14.
Cluster 1 contained mainly previously identified genes (\autoref{fig:plot_gene_set_enrichment.spline_cluster_venns.pdf}),
and was enriched for modules including 
myeloid cells and monocytes (LI.M81, hypergeometric test, FDR=\num{2.114861e-06}),
platelet activation (LI.M196, FDR=\num{1.347609e-05}),
immune activation (LI.M37.0, FDR=\num{1.435659e-04}),
and TLR and inflammatory signalling (LI.M16, FDR=\num{2.364913e-03}).
The spline analysis highlighted that expression differences at week 14 are maintained at week 30 and week 54.

The highest proportion of genes uniquely identified as significant by the spline analysis were in cluster 2 (26/31) and cluster (15/20).
Cluster 2 was enriched in \autocite{li2013MolecularSignaturesAntibody} B cell modules (LI.M47.0, FDR=\num{1.525815e-06}; LI.M47.1, FDR=\num{4.531246e-05})
previously identified as having a greater increase from week 0 to week 14 in primary responders versus primary non-responders (\autoref{fig:multipants_dge_panelPlot_week_14_0_R_N_cellPropT}),
matching the observed cluster trajectory.
Cluster 4 was not enriched in any modules from \textcite{li2013MolecularSignaturesAntibody}, but is enriched for a B cell module (DC.M4.10, FDR=\num{0.001367196}) from \textcite{chaussabel2008ModularAnalysisFramework}.
Although no genes were significantly associated with response at week 0 (\autoref{fig:multipants_dge_volcano_week_0_R_N}),
the genes in cluster 4 are coordinately downregulated as a set in primary responders (CERNO test, p=\num{6.175642e-25}).

% AL353807.3
% STAT1
% BATF2
% GBP1
% GBP5
% IRF1
% TAP1
% APOL2
% APOL1
Cluster 3 is also of interest, enriched for type I interferon response (LI.M127, FDR=\num{0.005680666}) and interferon (DC.M3.4, FDR=\num{0.0005266239}) modules,
as well as genes that contain putative transcription factor binding motifs for interferon regulatory factors \gene{IRF7} (\software{g:Profiler} term ID TF:M00453\_1, adj. p value=\num{0.0050514484}) 
and \gene{IRF8} (TF:M11684\_1, adj. p value=\num{0.0077680709}; TF:M11685\_1, adj. p=\num{0.0103433288}).
The cluster trajectory shows direction of expression change is opposing in responders and non-responders from week 0 to week 14, followed by sustained differences at week 30 and week 54.
The trajectory and interferon-related gene set enrichments are consistent with those identified in \autoref{subsec:multipants_dge_opposing_interferon}.
Of the 9 genes in this cluster, 8 genes (\gene{STAT1}, \gene{BATF2}, \gene{GBP1}, \gene{GBP5}, \gene{IRF1}, \gene{TAP1}, \gene{APOL1}, \gene{APOL2})
have significant interaction between week 0 to week 14 expression change and response status,
whether or not correcting for cell composition.
However, only \gene{GBP5} was differentially expressed from week 0 to week 14 in both responders and non-responders,
and only when unadjusted for cell composition (\autoref{fig:multipants_dge_logFC_C_3R_1R_vs_logFC_C_3N_1N}).
This indicates that such small and opposite effects in responders and non-responders may only be detected at a single-gene level in the interaction analysis that tests the difference,
and in the spline analysis, with the support of additional data from week 30 and week 54.

% TODO: add siglec 10 trend raw, and add to discussion

\begin{figure}
    \centering
    \includegraphics[width=1.0\textwidth,page=1]{mainmatter/figures/chapter_04/plot_gene_set_enrichment.spline_fviz_nbclust_gap_stat.pdf}
    \caption{}
    \label{fig:multipants_dge_spline_fviz_nbclust_gap_stat}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=1.0\textwidth,page=1]{mainmatter/figures/chapter_04/plot_gene_set_enrichment.spline_cluster_trajectories.pdf}
    \caption{}
    \label{fig:multipants_dge_spline_cluster_trajectories}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=1.0\textwidth,page=1]{mainmatter/figures/chapter_04/plot_gene_set_enrichment.spline_cluster_venns.pdf}
    \caption{}
    \label{fig:plot_gene_set_enrichment.spline_cluster_venns.pdf}
\end{figure}
    
\subsection{Limited evidence for change in genetic architecture of gene expression over time}

Given the substantial changes in expression from baseline to post-induction,after starting the drug, and the differing trajectories observed in responders and non-responders, 
I performed \gls{eQTL} mapping to detect common genetic variants associated with expression.
Variants cis (within \SI{1}{\mega b} of the TSS) to 15040 genes were tested for association.
Mapping was done within each timepoint (weeks 0, 14, 30, and 54),
followed by joint analysis of per-timepoint \gls{eQTL} summary statistics and control for multiple testing were done using \software{mashr}.

% TODO
% also drugs pooled due to much more power for eqtl required than DGE, may be inappropriate

The majority 11156/15040 (\percentage{0.74175531914}) of genes were eGenes (a gene with at least one significant cis-eQTL) in at least 1 timepoint (lfsr < 0.05).
The variant with the lowest lfsr in any timepoint for each gene was 
chosen as the lead variant (eSNP) for that gene.
Most eSNPs are significant in multiple timepoints: 999 significant in 1 timepoint, 381 significant in 2 timepoints, 526 significant in 3 timepoints, 9250 significant in all 4 timepoints.
I compared eSNP effect sizes between week 0 and each of weeks 14, 30 and 54.
to identify \glspl{reQTL} with significant difference in effect versus baseline,
and may be associated with changes in expression from baseline.
Most eSNPs were shared across timepoints;
only six eSNPs-eGene pairs were significant at BH FDR < 0.05:
with 1/6 for week 30 versus week 0 and 5/6 for week 54 versus week 0 (\autoref{fig:multipants_reQTL_pm_w30_vs_w0_and_w54_vs_w0}).
\todo{any need to add w14 to plot?}
Of the six eGenes, \gene{NMI} and \gene{EPSTI1} both had their respective eSNPs having magnified effects on expression at week 54 compared to week 0,
and both are annotated to contain putative binding motifs for \gene{IRF8} and \gene{IRF2} (\software{g:Profiler} term IDs TF:M11685\_1 and TF:M11665\_1).
However, there may be complications of confounding \todo{right word???} by cell composition for reQTLs in bulk expression data (discussed in \autoref{subsec:hird_reQTL_methods_cellTypeInteraction}).

% \begin{figure}
%     \centering
%     \includegraphics[width=1.0\textwidth,page=1]{mainmatter/figures/chapter_04/}
%     \caption{}
%     \label{fig:multipants_dge_}
% \end{figure}

\begin{figure}
    \centering
    \includegraphics[width=1.0\textwidth,page=1]{mainmatter/figures/chapter_04/plot_dge_eqtl.pm_w30_vs_w0_and_w54_vs_w0}
    \caption{Week 30 and week 54 eQTL effect sizes vs baseline. Significant reQTLs in blue.}
    \label{fig:multipants_reQTL_pm_w30_vs_w0_and_w54_vs_w0}
\end{figure}

% \1 clustering of eQTL effect sizes across 4 timepoints to identify general patterns of change in beta \todo{similar to the idea of moving from single gene to gene set analyses for more sensitivity}
%     \2 start with prefilter for any reQTL significant at nominal p < 0.05. There were 344.
%         \3 327/344 significant in all timepoints
%     \2 3 main clusters: high effect at w54, high effect at w0 and intermediate cluster (\autoref{fig:multipants_reQTL_clusters})
%         \3 ADCY3 is in cluster 3
%     \2 GSEA on cluster 1 revealed enrichment of genes with interferon regulatory motifs in cluster 1 (\autoref{fig:multipants_reQTL_clusters_gprofiler})
%     \2 2 INF genes: NMI EPSTI1

% \begin{figure}
%     \centering
%     \includegraphics[width=1.0\textwidth,page=1]{mainmatter/figures/chapter_04/plot_dge_eqtl.reQTL_clusters.pdf}
%     \caption{Clustering of eQTL betas over the 4 visits}
%     \label{fig:multipants_reQTL_clusters}
% \end{figure}

% \begin{figure}
%     \centering
%     \includegraphics[width=1.0\textwidth,page=1]{mainmatter/figures/chapter_04/reQTL_cluster_1_gene_sets_Screenshot 2020-09-04 at 00.18.57.png}
%     \caption{gene set enrichment using gprofileR for cluster 1 genes}
%     \label{fig:multipants_reQTL_clusters_gprofiler}
% \end{figure}

\section{Discussion}

In \gls{PANTS}, a cohort of \gls{CD} patients receiving infliximab or adalimumab anti-TNF therapy for the first time,
there were substantial differences in whole blood gene expression between primary responders and non-responders.
At baseline, the greatest differences in expression were observed between future responders and non-responders to infliximab,
with increased expression of monocyte, neutrophil and dendritic cell gene modules in responders,
and decreased expression of T cell and NK cell modules.
\todo{So many modules associations here, maybe try to Google some of them...}
These effects appear to be infliximab-specific, and are attenuated after adjusting for the proportions of six major immune cell types,
suggesting expression differences may be driven by mediation via the proportions of these cell types.
% Although the sample size for infliximab is greater than for adalimumab,
% this is unlikely to be the only factor, as the drugs are much more comparable after adjusting for cell composition.

In contrast, future responders to adalimumab had lower baseline expression of plasma cell and cell division modules.
Th module-level results line up with the three gene-wise hits for the adalimumab-only analysis:
\gene{IGKV1-9} encodes the immunoglobulin light chain variable region that forms part of antibodies produced by plasma cells,
\gene{KCNN3} is annotated to plasma cell surface signature module (LI.S3\autocite{li2013MolecularSignaturesAntibody}), 
and the expression of both \gene{KCNN3} and \gene{PDIA5} are correlated with blood plasmablast frequencies\autocite{tsang2014GlobalAnalysesHuman}.
It was reported by \textcite{gaujoux2019CellcentredMetaanalysisReveals} that baseline plasma cell abundances are lower for infliximab responders,
hypothesising that plasma cell survival is supported by increased \gls{TNF} levels in non-responders.
Plasma cells also formed a part of a correlated module of cell populations identified by \autocite{martin2019SingleCellAnalysisCrohn},
where lower module expression was associated with better response to anti-\gls{TNF} in a cohort with patients taking both infliximab and adalimumab.
However, both these studies were done in gut biopsy samples, and there was no report of strong between-drug heterogenenity.

The adalimumab-specific associations I find were more significant after cell proportion adjustment, 
which may indicate per-cell downregulation rather than cell abundance being associated with response.
However, cell composition differences mediated by rarer cell types that have abundances poorly captured by the six major types used in the model will be poorly adjusted for.
For example, plasma cell proportions are only weakly correlated with other immune cell types in the healthy immune system\autocite{zalocusky201810000Immunomes}, 
although the relationship may differ for \gls{CD} patients.
It has also been shown that differentially expressed genes in blood with correction for only major cell types will 
identify associations that are proxies for rare cell types\url{https://www.biorxiv.org/content/10.1101/2020.05.28.120600v1}.
If this is the case, the role of the cell composition estimates for adalimumab-specific effects may be more akin to precision variables, 
which would be consistent with increased significance after adjusting.
% although formal tests for difference in effect between adjusted and non-adjusted estimates would be required to confirm this.

The differences between drugs are puzzling, especially the greater effect of cell composition adjustment for infliximab. 
Baseline patient differences between drugs may offer a partial explanation.
% Due to lack of randomisation, patients that took infliximab and adalimumab are likely to differ by more than drug.
In the full \gls{PANTS} cohort, baseline characteristics other than those in \autoref{tab:multipants_table1} differ between patients on different drugs\autocite{kennedy2019PredictorsAntiTNFTreatment}.
In particular, lower albumin, higher \gls{CRP}, and higher faecal calprotectin in infliximab patients suggest that they may have had greater disease severity.
% Nick: “In general, sicker patients have often been given infliximab. At the time, children were rarely given adalimumab, but that doesn’t affect the RNA seq cohort because children were excluded.”
Differences may be driven by patient or physician preference, for example, patients with more severe disease are often given infliximab rather than adalimumab\footnote{Kennedy, N. A., personal communication, 4 June (2020).}.
I have not yet been able to access clinical variables such as \gls{CRP} and faecal calprotectin levels to consider as variables to adjust for in my modelling.
A richer phenotype dataset containing some of these variables has been requested from collaborators.

The strongest single-gene association in the pooled analysis was \gene{SIGLEC10}, 
which had reduced significance post-adjustment with a comparable effect size,
where baseline expression was approximately 25\% higher in responders.
Direction of effect was consistent between drugs, but most significant in infliximab without cell composition adjustment.
In \gls{IBD}, small molecules called \glspl{DAMP} are released due to tissue damage and cell death, 
and further promote inflammation through pathogen sensing \gls{PRR} pathways that include \gls{TLR} family receptors\autocite{boyapati2016GutMucosalDAMPs,desouza2016ImmunopathogenesisIBDCurrent}.
For instance, faecal calprotectin, a marker for \gls{IBD} activity, is a complex of two \glspl{DAMP}, S100A8 and S100A9\autocite{desouza2016ImmunopathogenesisIBDCurrent}.
SIGLEC10 has been shown to repress \gls{DAMP}-mediated inflammation through binding CD24\autocite{boyapati2016GutMucosalDAMPs}.
\gene{SIGLEC10} is expressed on B cells, monocytes and eosinophils \url{https://www.nature.com/articles/nri2056},
and of these cell types, module level results posit monocytes as the most likely candidate cell type to have increased module expression in responders.
In monocytes, \gene{SIGLEC10} expression is more specific to the CD16+ monocytes\autocite{martinez2009TranscriptomeHumanMonocyte},
and in particular the CD14+CD16++ non-classical monocytes rather than the classical CD14++CD16- or intermediate CD14++CD16+ subsets\autocite{villani2017SinglecellRNAseqReveals}.
% Non-classical (anti-inflammatory) monocytes have a role in many chronic inflam diseases \url{https://www.annualreviews.org/doi/full/10.1146/annurev-immunol-042617-053119#_i16} 
% including CD \url{https://www.researchgate.net/profile/Siew-Cheng_Wong/publication/221723208_The_three_human_monocyte_subsets_Implications_for_health_and_disease/links/09e415101253562a3c000000.pdf}
% Also from above: "The two main forms of human inflammatory boweldiseases (IBD) are Crohn’s disease (CD) and ulcerative colitis (UC), and the expansion of CD16+ monocytes had been reported in both forms of IBD [87,88]. In CD, it is during active disease that the monocyte subsets perturbation was observed [87,89]."
In \gls{PANTS}, it was suggested by \textcite{kennedy2019PredictorsAntiTNFTreatment} that higher inflammatory load as indicated by low baseline albumin levels 
may result in low week 14 drug levels due to faster drug clearance,
and low drug levels at week 14 were in turn associated with non-response.
A hypothetical model might be imagined where high baseline \gene{SIGLEC10} expression reflecting 
higher proportions of CD16+ monocytes (or lower proportions of CD16- monocytes),
decreased \gls{DAMP}-mediated inflammation, 
and increased chance of primary response, possibly by affecting drug clearance rate.
This is an extremely tentative model:
both the cell proportion estimates and module definitions used thus far only represent monocytes as a whole,
lacking the resolution to properly explore shifts in the three monocyte subsets.
It may be possible to use expression of monocyte subset marker genes such as those identified by \textcite{villani2017SinglecellRNAseqReveals} to improve the resolution of the cell proportion estimates.

Despite the strong heterogenenity in effects between drugs, one consistent effect that emerged after adjusting for cell composition was 
baseline upregulation of MHC-TLR7-TLR8, antigen presentation, and interferon modules in responders.
\todo{How does the MHC come in here?}
As mentioned above, \gls{TLR} receptors are involved in pathogen sensing, and TLR7 and TLR8 are endosomal proteins primarily expressed in monocytes, macrophages and \glspl{DC},
part of an antigen presentation pathway that senses bacterial DNA and activates downstream innate immune pathways including type I interferon response \url{https://www.nature.com/articles/cmi201238}.
Type I interferons have pathogenic or protective roles in many \glspl{IMID}\url{https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4084561/}.
It has been suggested that type I interferon responses induced via TLR7 and TLR8 can suppress colitis in mouse models, and play a role in maintaining gut homeostasis
\url{https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5797585/}
\url{https://www.frontiersin.org/articles/10.3389/fmed.2018.00032/full},
so upregulation here may again represent a less severe baseline disease in future responders.

Most previously reported baseline markers in blood and gut biopsies were non-significant in this study.
% One exception is \gene{PTGS2}, which is also specific to Mono3 in \textcite{villani2017SinglecellRNAseqReveals}.
For gut markers, this may not be unexpected.
Although a subset of gut infiltrating immune cells and their precursors may also be circulating,
genes specific to epithelium and immune cell types that differentiate after they migrate into tissues (e.g. monocyte-derived macrophages),
will be difficult to observe in blood.
For blood markers, I sought to clarify the conflicting results in the literature about the association of \gene{TREM1} expression in blood with anti-\gls{TNF} response\autocite{gaujoux2019CellcentredMetaanalysisReveals,verstockt2019LowTREM1Expression}.
I did not find \gene{TREM1} to be significantly differentially expressed in \gls{PANTS},
although the direction of effect is increased expression in responders,
matching the \textcite{gaujoux2019CellcentredMetaanalysisReveals} direction of effect in blood.
\gene{TREM1} is expressed on myeloid lineage cells such as monocytes and macrophages;
\textcite{villani2017SinglecellRNAseqReveals} reported that \gene{TREM1} expression is most specific to 
classical monocytes and a newly identified subtype within the intermediate monocytes (\enquote{Mono3}).
The \gene{TREM1} effect is one of the infliximab-specific differentially expressed genes that is much stronger without cell proportion adjustment,
so it may reflect association of baseline monocyte cell proportions with response.

There are many factors could explain failures to replicate reported markers, or identification of different markers from study to study.
Many existing studies pool cohorts with different anti-\gls{TNF} biologics due to the scarcity of large datasets,
yet even within this study, there is heterogenenity between drugs.
There are between-study differences in the definition of primary response,
such as endoscopic healing\autocite{gaujoux2019CellcentredMetaanalysisReveals} versus scoring on clinical parameters\autocite{verstockt2019LowTREM1Expression}.
Any two studies are unlikely to have adjusted for the same combinations of covariates in modelling,
and some covariates like cell composition are very influential for bulk expression data.
Finally, small sample sizes have considerable sampling error.
Set-based association tests that draw on changes in multiple genes,
% such as the cell abundance module associations reported for gut biospies by \textcite{martin2019SingleCellAnalysisCrohn},
such as the expression module associations for blood in this study,
may be more reproducible compared to single-gene markers. 

Although this study is purely descriptive, a future aim will be to see if the identified baseline module associations also imply that response status can be predicted from baseline expression.
% Correction for cell proportions has thus far provided complementary interpretations of total effect and effects not mediated by cell count.
Because modules associated with response appear to be mediated by cell proportions,
much of the predictive ability may also lie in differences in cell proportions between responders and non-responders.
% Adjusting would be counterproductive.
Indeed, \textcite{gaujoux2019CellcentredMetaanalysisReveals} noted that adjusting expression for cell composition resulted in gut gene signatures that were worse at discriminating responders from non-responders.
Testing specific subpopulations such as CD16+ monocyte or plasma cell abundance for association with response
can also be viewed as a type of set-based test that represents a set of cell-type specific genes,
and thus may also be more reproducible than single-gene markers.

% Stronger signal to noise ratio
Much larger proportions of the transcriptome are associated with response after the induction period at week 14.
Module associations
showed downregulation of immune activation, \gls{TLR}, inflammatory, monocyte and neutrophil modules in responders;
and upregulation of B and T cell modules.
Similar module associations were also found when considering modules differentially expressed from week 0 to week 14.
The differences between responders and non-responders at week 14 
were qualitatively similar to the differences pre and post anti-\gls{TNF} induction,
suggesting there may be relatively little change in the transcriptome of non-responders after induction.
Associations were generally consistent between drugs for both the within week 14 and change from week 0 to week 14 analyses,
perhaps because the effect of baseline differences between patients taking different drugs on the transcriptome is diluted
by the large transcriptomic perturbation caused by taking an anti-\gls{TNF} drug.
Many of the same modules were also significant regardless of cell proportion correction.

There appears to be a general reduction in immune activation in responders at week 14,
presumably due to successful inhibition of \gls{TNF},
which is consistent with reduced neutrophil activation and reduced monocyte recruitment \url{https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5820413/}.
Apoptosis of monocytes induced by anti-\gls{TNF} in \gls{CD} patients has also been previously observed \url{https://www.gastrojournal.org/article/S0016-5085(01)32104-2/fulltext}.
Certain B cell subsets are reduced in the blood \gls{IBD} patients compared to controls \url{https://www.frontiersin.org/articles/10.3389/fimmu.2019.00361/}, 
so upregulation of B cell modules after treatment may represent a shift towards health.
Another potential explanation would be increased immunogenicity due to higher drug levels in responders\autocite{kennedy2019PredictorsAntiTNFTreatment}.
although lack of between-drug heterogenenity does not support the greater immunogenicity of infliximab versus adalimumab.
Overall, it is difficult to determine exact mechanisms in this observational study design, with bulk expression data, using such broad module definitions.

Some previously identified baseline gut markers of response that were not differentially expressed in blood at week 0, were differentially expressed at week 14.
\gene{S100A8} and \gene{S100A9}, identified as markers by \textcite{arijs2010PredictiveValueEpithelial}, which encode components of the inflammatory marker \gls{CRP}, were downregulated in week 14 responders.
The cytokine \gene{OSM}, which promotes inflammation in gut stromal cells\autocite{west2017OncostatinDrivesIntestinal}, was similarly downregulated.
Although it is essentially pointless to use a week 14 marker to predict a response that is defined at week 14,
this does demonstrate that gut markers can coincide with blood markers if expressed in immune cells present in both tissues.

When considering the interaction between change from week 0 to week 14 and response,
the general pattern is magnification in responders,
where the same expression changes occurring in both responders and non-responders tend to have greater magnitude in responders.
A potential hypothesis is a continuum of response from non-response to response.
\textcite{gaujoux2019CellcentredMetaanalysisReveals} found changes in cell proportions in response to anti-\gls{TNF} treatment were magnified in responders, 
also supporting response as continuous phenotype.
This study confirms a similar trend at the transcriptional level.
% TODO
% https://www.nature.com/articles/s41467-018-07841-3
% Notably, the corticosteroid response gene signature showed a substantial overlap with genes previously associated with anti-TNF response, and exhibited a similar difference between responders and nonresponders to anti-TNF or anti-integrin α4β7 therapies. These similarities support an emerging concept in the field that the mucosal inflammatory state as measured by gene expression may better define the likelihood of response to current treatment approaches than conventional clinical measures of severity.
% (Oncostatim M (OSM) and TREM122)

% \1 PNR definition
%     \2 its a very complex binary, but certainly useful
%         \3 kennedy2019PredictorsAntiTNFTreatment Univariable analysis showed, for both drugs, that the most significant determinant of non-remission at week 54 was clinical status at week 14 (table 4; appendix pp 21–22).
%         \3 DGE analysis also agrees with kennedy2019PredictorsAntiTNFTreatment: once PNR, no point in continuing
%     \2 if continuum of PNR? perhaps model decomposed pheno?
%         \3 e.g. "Genetic Loci associated with C-reactive protein levels and risk of coronary heart disease."
%         \3 take advantage of richness of dataset
%     \2 although other mediators of NR could be modelled using genetic instruments e.g drug level

There were some rare exceptions to magnification for genes and modules in the type I interferon pathway.
These showed upregulation in non-responders from week 0 to week 14,
yet were either downregulated or not significantly different for responders.
Single-gene examples include the interferon-induced guanylate-binding proteins \gene{GBP2} and \gene{GBP5} \url{https://rupress.org/jem/article/216/3/482/120360/Interferon-induced-guanylate-binding-proteins}, 
and \gene{STAT2}, a key transcription factor for interferon-stimulated genes\autocite{schneider2014InterferonStimulatedGenesComplex}.
Genes such as \gene{IFIT3} and \gene{STAT2} are more strongly induced by type I interferons compared to type II \url{https://www.pnas.org/content/109/11/4239}.
A study of \gls{RA},
an \gls{IMID} also treated with anti-\gls{TNF} drugs,
also found increases in type I interferon-regulated gene expression in blood after infliximab treatment associated with poor clinical response \url{https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2875639/}.

A spline model of expression over all four timepoints confirmed the above observations made in week 0 and week 14 samples.
\todo{move some of this to results section}
% TODO: add KREMEN1 plot
Two main clusters of genes (clusters 1 and 5) contained mostly genes significantly associated with response in the two pairwise comparisons: within week 14, and change from week 0 to week 14.
An example is the most significant single-gene association from the cluster 1 in spline model, \gene{KREMEN1}, 
which was also among most significant associations in the pairwise comparisons.
\gene{KREMEN1} is part of an inflammatory apoptotic pathway in gut epithelium \url{https://academic.oup.com/ibdjournal/article/14/suppl_1/S4/4653822}, 
and is downregulated in responders post-induction.
The trajectories of expression for genes in these cluster 1 and 5 confirmed changes in expression post-induction were generally greater for responders.
and in addition demonstrates that post-induction expression differences between responders and non-responders are sustained
in samples taken around week 30 and week 54 during the anti-\gls{TNF} maintenance period.
In \gls{PANTS}, \enquote{continuing standard dosing regimens after primary non-response was rarely helpful} for inducing remission by week 54\autocite{kennedy2019PredictorsAntiTNFTreatment}.
This phenomenon may have a transcriptomic basis.
% \textcite{gaujoux2019CellcentredMetaanalysisReveals} suggesting increased drug doses may be effective in treating a non-responsive individual.

Making use of data from later timepoints allowed more subtle effects to be detected in the spline analysis.
Clusters 2 and 4 are enriched for B cell genes that were not significantly different at the single-gene level in the within week 0 comparison,
although some downregulation of B cell and plasma cell modules were detected.
% AL353807.3
% STAT1
% BATF2
% GBP1
% GBP5
% IRF1
% TAP1
% APOL2
% APOL1
Cluster 3 reproduced the observation that interferon-induced genes have opposing trajectories of expression in responders and non-responders.
Expression of these genes was higher in responders at week 0 and lower at all post-treatment timepoints.
Here, the cluster contains genes such as \gene{STAT1}, \gene{IRF1} and \gene{TAP1} that are induced by both type I and type II interferons \url{https://www.pnas.org/content/109/11/4239}.
% TODO: add TREM1 plot
I propose that blood expression of interferon-related genes is an attractive target for future studies of the biological basis of anti-\gls{TNF} response,
or prediction of primary response status.
If one assumes response and non-response lie along a phenotypic continuum,
it is less likely that the opposing directions of regulation can be explained by unmodelled variables such as serum drug level being higher in responders.
% Or indeed TNF being lower in responders.
\todo{not entirely sure this this is statistically rigourous due to third var effects?}
Since the difference is maintained until week 54, by which time patients would have recieved many doses of drug,
it is also more likely that response is due to some biological property of an individual patient.
Studies of anti-\gls{TNF} response in \gls{RA} patients have also 
found high baseline interferon activity in blood to be associated with good clinical response
\url{https://academic.oup.com/rheumatology/article/54/1/188/1840617}
\url{https://onlinelibrary.wiley.com/doi/full/10.1002/art.27226}.

It should be noted that the number of clusters is only the optimal number determined in this dataset,
and does not imply that genes in different clusters represent biologically distinct pathways.
Clusters 2 and 4 have similar trajectories and enrichments for B cell genes,
and interferon pathway genes appear in both clusters 1 and 3.
Repeating the analysis setting a different number of clusters, or with bootstrapping, will help to determine a robust number of clusters.

% \1 reQTL mapping to identify changes in genetic control of expression over the timepoints
%     \2 only 6 reQTLs at per-comparison FDR 0.05
%     \2 2 main patterns of reQTLs over time
%     \2 change from baseline expected, but no enrichments
%     \2 but can only speculate on why INF-stimulated genes had change in E from baseline to w54 genetically controlled
%     \2 biases?
%         \3 winner's curse caused by combo of low power and a signif threshold?
%         TODO
%         \3 EE vs EZ model with varying sample size?  https://stephenslab.github.io/mash-application-immune/
%         If we define the reQTL as different standardized effects, the result is EZ V1 model reQTL.
%     \2 vs ch3 HIRD, fewer strong reqtls
%         \3 probably due to low signal: less changes in cell comp across timepoints? or different FDR?
%         \3 and high noise: variation in time since last dose, and much longer timeframe (weeks between doses, vs days since vacc) overall, so more chance for env factors to act
% since E is diff between R and NR,
%     why not put in an interaction between R and NR? not sure if powered
%     not sure if estmating the interaction effect would require more power
%     than losing the averaging effect
Finally, I also attempted to determine if there were changes in genetic architecture of expression over time,
which can indicate that expression response to anti-\gls{TNF} has a genetic component.
Out of all significant lead \glspl{eQTL} for 11156 genes, only six \glspl{reQTL} were detected with significantly different effect sizes at baseline versus one of the three post-treatment timepoints.
Although no enrichment analyses are reported due to the small number of associations, 
\gene{NMI} and \gene{EPSTI1} are both interferon-induced genes with significant \glspl{reQTL} that have their strongest effect size on expression at week 54.
Given the issues with doing this \gls{reQTL} analysis in bulk expression data are similar to those encountered in \autoref{chap:hird_reQTL},
I did not place emphasis on interpreting these small numbers of associations.
Before proceeding, I would also like to verify that these significant \glspl{reQTL} are not artifacts from shrinkage of effect sizes in the joint \gls{eQTL} model,
as their posterior effect sizes from \software{mashr} were very different from the inputs from the timepoint-stratified models.
If these hits are indeed reproducible by complementary methods such as \gls{ASE}\autocite{gutierrez-arcelus2020AllelespecificExpressionChanges},
it may then be worth introducing genotype-response interaction terms in the \gls{eQTL} models
to prioritise \glspl{eQTL} with differing effects in responders and non-responders.
% It would still be difficult to determine cell types responsible
Given there is prior interest in the interferon pathway from \gls{DGE} analyses,
a more statistically powerful approach may be to generate a continuous interferon pathway score for each sample,
which would then act as the interacting variable,
similar to the approach of \textcite{davenport2018DiscoveringVivoCytokineeQTL}.

Several threats to the validity of the study remain to be discussed.
The most pressing may be the meaning of time in the study.
For pairwise \gls{DGE} comparisons and \gls{reQTL} mapping, 
samples were divided into four timepoints that corresponded to the major visits in \gls{PANTS}.
The \gls{DGE} spline model used study day directly.
The actual sample study day has substantial variation around the target for later timepoints.
The particular targets (weeks 14, 30, 54) for post-baseline samples were chosen so that patients on infliximab (8 weeks between doses) and adalimumab (2 weeks between doses) could both be sampled with the same visit structure.
Drug levels peak sharply after each dose and decline exponentially over time.
Visits were scheduled to be as close as possible (within a week) to the next scheduled drug dose to capture trough drug levels.
Neither approach is perfect, as matching patients by timepoint and study day are only attempts to gather samples matched by trough drug level.

% "A LOR visit should be scheduled any time after week 14 if the LOR criteria are met (page 11) and the patient’s gastroenterologist has escalated steroid, immunomodulatory or anti-TNF therapy. Escalation of anti-TNF therapy includes an increase in the dose, or a shortening of the interval between doses."
A further complication is the inclusion of loss of response samples in analyses.
One treatment option after loss of response is dose escalation, which may raise trough drug levels for all subsequent visits for those patients.
Howeve, since the \gls{PANTS} protocol allows for loss of response visits that coincide with major visits to be labelled as a major visit,
there is no guarantee that simply excluding samples labelled as loss of response would resolve this.
The best solution may to explicitly model measured serum drug levels as a covariate, where like cell proportions, it would likely act as a mediator of some associations with response.
% Overall 319/840 missing corresponding drug levels.
I did not do this as data missingness would reduce the sample size by about 40\% in this study.
Finding a suitable normalisation of drug level for use in pooled drug analyses would also be challenging.
Infliximab and adalimumab have differing pharmokinetics: infliximab has higher peak concentrations, higher peak-trough ratios, and shorter half-life.
The same serum concentrations of infliximab and adalimumab also have different biological effects due to differing therapeutic windows
\autocite{tracey2008TumorNecrosisFactor,lichtenstein2013ComprehensiveReviewAntitumor} \url{https://onlinelibrary.wiley.com/doi/full/10.1111/apt.15643}.

Finally, the effects of differential drop out in responders and non-responders has not been explored.
% https://rpsychologist.com/lmm-slope-missingness
% Why linear mixed-effects models are probably not the solution to your missing data problems
% https://stats.stackexchange.com/questions/354279/can-a-linear-mixed-model-handle-missing-data-that-is-not-missing-at-random
% As you say, linear mixed-effects (LME) models are assuming that any missing data are missing at random (MAR). A way to account for suspected violations of the MAR assumption of LME models is to use joint models. Under this framework, the longitudinal response and the survival probability are associated by jointly estimating a single likelihood function. Using a joint model, you can get an estimate of how much a possible difference in survival is affecting the longitudinal response and how much the fluctuations in the longitudinal response are associated with survival.
The three main mechanisms of missing data are:
\gls{MCAR}, probability of data being missing is independent of both observed and missing data;
\gls{MAR}, probability of data being missing conditional on observed data is independent of missing data;
and \gls{MNAR}, probability of data being missing depends on missing data \url{https://doi.org/10.1016/B978-0-12-801342-7.00014-9}.
Even conditional on response status, it's more likely that expression data from more extreme non-responders, so the likely mechanism here is \gls{MNAR},
and the linear mixed models used in this study may be biased.
If it is indeed the most extreme non-responders dropping out, the estimation of responder versus non-responder effects may be conservative.
Note there is no sidestepping a \gls{MNAR} mechanism by analysing only the complete cases, 
since they will differ systematically from the sample as a whole \url{https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3016756/}.

% TODO: add a whilst we were not able to X in the intro, the field has advanced by Y, future Z needed
In conclusion,
it remains unclear whether there are any robust single-gene markers for anti-\gls{TNF} response in baseline whole blood of \gls{CD} patients.
Baseline module associations were observed, but there was unexpected heterogenenity between infliximab and adalimumab patients,
so it remains to be seen if such associations will be replicated in separate cohorts.
Large upcoming datasets with drug response phenotypes such as the 1000IBD project\autocite{imhann20191000IBDProjectMultiomics} will be invaluable for attempted replication of the associations found in \gls{PANTS}.
Expression differences between responders and non-responders were more distinct at timepoints after the induction period.
I found type I interferon genes went against the general trend of greater transcriptomic change from baseline in responders,
being more upregulated post-treatment in non-responders.
Given the type I interferon expression in blood has also been associated with anti-\gls{TNF} response in \gls{RA} patients,
there may be an opportunity to consider the shared biology of anti-\gls{TNF} response in \gls{IBD} and \gls{RA}.
% and other \glspl{IMID} treated with anti-\gls{TNF} drugs.
Much work has been done generating and validating signatures for anti-\gls{TNF} response in \gls{RA} \url{https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0033199}; but not much work on validating \gls{RA} signatures in \gls{IBD} cohorts and vice versa.

This chapter has been purely descriptive.
I have deliberately avoided the term \enquote{signature} in my own results, 
as I have not yet had a chance to assess their predictive capability.
I also did not find strong evidence for strong \gls{reQTL} effects over time in whole blood, 
so was unable to form hypotheses on the genetic mechanisms influencing anti-\gls{TNF} response.
However, the presence of \glspl{eQTL} for most genes and the presence of strong differences in expression post-induction may allow testing for causal mechanisms where genotype affects drug response via expression.
Strategies for moving onto both prediction and causal inference will be discussed in \autoref{chap:discussion}.

% TODO: resolution of sets
% testing specific hypotheses/mechs of antiTNF action
% in this ch, nto possible

\end{outline}
