%
% Chapter 1
%
% Unifies results chapters under the framework of human genetics.

% TODO: look through dloads folder for relevant papers for each section
% TODO: first year report

\chapter{Introduction}

\begin{outline}

\section{Structure and diversity of the human genome}

\1 The human genome is almost three billion \glspl{bp} in length, 
containing 20000-25000 protein-coding genes \autocite{theencodeprojectconsortium2012IntegratedEncyclopediaDNA,1000genomesprojectconsortium2015GlobalReferenceHuman} that span 1-3\% of its length, 
with the remainder being non-coding.
% Some of the the remainder is likely dedicated to regulatory activities \autocite{theencodeprojectconsortium2012IntegratedEncyclopediaDNA}.
Each diploid human cell contains two copies of the genome; 46 chromosomes comprised of 23 maternal-parental pairs: 22 pairs of homologous autosomes and one pair of sex chromosomes.
%
\1 Variation in the genome between individuals in a population exists in the form of \glspl{SNP}, short indels, and structural variants---the vast majority of common variants ($\text{\gls{MAF}} > 1-5\%$) are \glspl{SNP} and short indels ($> 99.9\%$) \autocite{1000genomesprojectconsortium2015GlobalReferenceHuman}.
% https://www.ncbi.nlm.nih.gov/dbvar/content/overview/
% Structural variation (SV) is generally defined as a region of DNA
% approximately 1 kb and larger in size and can include inversions and
% balanced translocations or genomic imbalances (insertions and deletions),
% commonly referred to as copy number variants (CNVs).
On average, a pair of human genomes differs by one \gls{SNP} per 1000-2000 \gls{bp} \autocite{theinternationalsnpmapworkinggroup2001MapHumanGenome}.
Each version of a variant is called an allele; an individual has a maternal and parental allele at each variant.
\todo{consider moving awkward defs to margin notes, in the style of nature reviews}

% The Law of Dominance: An organism with alternate forms of a gene will express the form that is dominant.
% Law of Segregation: states that a diploid organism passes a randomly selected allele for a trait to its offspring, such that the offspring receives one allele from each parent.
% Law of independent assortment: states that genes do not influence each other with regard to the sorting of alleles into gametes; every possible combination of alleles for every gene is equally likely to occur.
\1 The many variants in a population are inherited in a smaller number of haplotypes: 
contiguous stretches of the genome passed through generations via meiotic segregation.
The fundamental sources of genetic diversity are mutation and meiotic recombination, generating new alleles and breaking apart haplotypes into shorter ones over time.
\todo{LD decay just takes a really really long time, but there are evo forces at work too that maintain LD}
Variants at locations on a chromosome (loci) that are physically close are less likely to flank a recombination event, hence more likely to cosegregate on the same haplotype, referred to as genetic linkage.
Genetic linkage is one source of \gls{LD}: the non-random association of alleles at two loci, differing from expectation based on their frequencies and the law of independent assortment \autocite{slatkin2008LinkageDisequilibriumUnderstanding}.
\gls{LD} is often quantified within a population by $r^2$, the squared correlation coefficient between alleles \autocite{slatkin2008LinkageDisequilibriumUnderstanding}.

Recombination events are not distributed uniformly throughout the genome.
The genome is a mosaic of blocks delimited by recombination hotspots, 
characterised by strong \gls{LD} within blocks, and little \gls{LD} between blocks \autocite{wall2003HaplotypeBlocksLinkage,theinternationalhapmapconsortium2007SecondGenerationHuman} (\autoref{fig:intro_haplotypeBlocks}).
\todo{Heard it's good for the reader's attention span to have figures in intro. Unless it's ok to use figures from papers, I only want to spend the time making the min that are necessary though.}
The structure of correlated haplotypes reflects a population's unique evolutionary history, and can be used to trace the demography of human populations back through time \autocite{karczewski2020AnalyticTranslationalGenetics}.

\begin{figure}
    \centering
    \includegraphics[width=1.0\textwidth,page=1]{mainmatter/figures/chapter_01/fig_mockup_haplotypeBlocks_Screenshot 2020-05-21 at 17.08.33.png}
    \caption{The genomic mosaic: block-like \gls{LD} structure of the genome}
    \label{fig:intro_haplotypeBlocks}
\end{figure}

\section{Genetic association studies for complex traits}

\subsection{Principles of genetic association}

\1 Variation in human traits arises from an interplay between genetics and environment.
Traits for which genetic variation explains a non-zero fraction of phenotypic variation are heritable.
Many measurable human traits are heritable and twin studies provide upper bounds on this heritability \url{https://www.nature.com/articles/ng.3285}.
Discovering the specific genetic variants that contribute to heritability, through association of variants and phenotypes measured from the same individual, is a mainstay of the field of human genetics.
Barring somatic mutation, an individual's genome is fixed at conception, providing a causally upstream anchor.
Genetic association studies have intrinsic resistance to many back-door path effects that permeate observational studies of the causes of human phenotypes.
\todo{add something sweeping about utility here or elsewhere: e.g. insights into trait biology and clinical translational potential for disease traits, genetically support drug target identification}

\1 Under the central dogma, information flows from gene to RNA to protein to phenotype via transcription and translation, 
thus it is assumed that genetic variants at loci in the genome affect phenotype by impacting on the function or regulation of target genes.
How genetic variation contributes to any heritable trait defines it's genetic architecture: the number of genes affecting that trait; along with the allele frequencies, effect sizes, and interactions of trait-associated variants \autocite{visscher2019Fisher1918Paper}.
The number of genes defines a spectrum of traits from monogenic (where inheritance follow simple Mendelian patterns) to polygenic (where inheritance is complex).
Many architectures have been proposed for complex traits; all have in common that the number of genes that affect a complex trait is large (ranging from dozens to many thousands),
thus the average effect of each trait-associated loci is small \autocite{gibson2011RareCommonVariants,boyle2017ExpandedViewComplex} \url{https://www.pnas.org/content/106/23/9362}.

\subsection{Lessons from 15 years of genome-wide association}

\1 For decades, linkage analysis had been successfully applied to map loci affecting Mendelian traits by tracing their cosegregation with the trait through pedigrees \autocite{visscher2012FiveYearsGWAS}.
Small-scale genetic association studies were also performed, focusing on variants in or near candidate genes selected on the basis of prior biological knowledge \autocite{hirschhorn2002ComprehensiveReviewGenetic}.
These approaches were not successful for complex traits, as small effect sizes lead to low penetrance in pedigrees \autocite{visscher2012FiveYearsGWAS}
and poor power at the sample sizes typically used in early candidate gene studies \autocite{border2019NoSupportHistorical}.

\1 \Glspl{GWAS} systematically test common variants selected in a hypothesis-free manner across the genome for association with a trait (\autoref{fig:intro_architectureGWAS}).
Using large sample sizes to overcome small effects and large multiple testing burden, thousands of associations have been discovered for complex traits and disease,
many robustly replicated across populations \autocite{visscher2012FiveYearsGWAS,visscher201710YearsGWAS}.
Most genetic variance is explained by additive effects, the contribution of epistatic interactions is small \autocite{visscher2019Fisher1918Paper}, 
and pleiotropy is widespread \autocite{visscher2012FiveYearsGWAS}.
Sample sizes in the millions are increasingly commonplace, 
and discovery of new associations with increasing sample size shows no sign of plateauing \autocite{tam2019BenefitsLimitationsGenomewide}.
It is now appreciated that most heritable phenotypes are complex, and have remarkable polygenicity.
% TODO: any more takehomes from: \autocite{gallagher2018PostGWASEraAssociation,tam2019BenefitsLimitationsGenomewide}

% Genetic variants exist along a spectrum of allele frequencies and
% effect sizes. Most risk variants identified by GWAS lie within the two diagonal lines.
% Rare variants with small effect sizes are difficult to identify using GWAS, and common
% variants with large effects are unusual for common complex diseases.
% tam2019BenefitsLimitationsGenomewide
\begin{figure}
    \centering
    \includegraphics[width=1.0\textwidth,page=1]{mainmatter/figures/chapter_01/fig_mockup_architecture_Screenshot 2020-05-21 at 17.08.41.png}
    \caption{The reach of GWAS. OR vs MAF ala tam2019BenefitsLimitationsGenomewide, extended by imputation, sample size, WGS based genotypes, but may be indistinguishable from noise at the limits}
    \label{fig:intro_architectureGWAS}
\end{figure}

\subsection{From complex trait to locus}

\glspl{GWAS} rely on the tendency of common variants on the same haplotype to be in strong \gls{LD}.
As the number of haplotypes is comparatively few, 
it is possible to select a subset of tag variants such that all other known common variants are within a certain \gls{LD} threshold of that subset. 
In practice, there is enough redundancy that the number of variants measured on a modern genotyping array (in the order of \numrange[retain-unity-mantissa=false]{1e5}{1e6}) is sufficient to tag almost all common variants  \autocite{theinternationalhapmapconsortium2005HaplotypeMapHuman,barrett2006EvaluatingCoverageGenomewide}.
Associations with unmeasured variants are indirectly detected through their strong correlation with a tag variant.
Furthermore, as unrelated individuals still share short ancestral haplotypes, 
study samples can be assigned haplotypes from a panel of haplotypes derived from reference samples by matching on the directly genotyped variants.
This process of genotype imputation allows ascertainment of many more variants not directly genotyped \autocite{das2018GenotypeImputationLarge},
but helps to recover rarer variants that are poorly-tagged \autocite{visscher201710YearsGWAS}.
\todo{seems like there is some connection to be made between the tagability of common variation and the feasibility of imputation both being enabled by the relatively small number of common haplotypes compared to variants}
Modern imputation panels enable cost-effective \glspl{GWAS} including tens of millions of variants down to frequencies of \textapprox{0.01\%} \url{https://www.biorxiv.org/content/10.1101/563866v1}.

Testing such large numbers of variants incurs a massive multiple testing burden, but acknowledging the correlation between variants due to \gls{LD},
there are only the equivalent of \textapprox{10^6} independent tests in the European genome, regardless of the number of tests actually performed \autocite{peer2008EstimationMultipleTesting}.
The field has thus converged on a fixed discovery threshold of $0.05 / 10^6 = \num{5e-8}$ for genome-wide significance in European populations \autocite{jannot2015108HasEmerged}, akin\footnote{
    The Bonferroni procedure makes no assumptions about the dependence structure of the \pvalues{}, and is conservative (i.e. controls the \gls{FWER} at a stricter level than the chosen $\alpha$) even for independent tests. In fact it is always conservative unless the \pvalues{} have strong negative correlations \autocite{goeman2014MultipleHypothesisTesting}.
}
to controlling the family-wise type I error rate at using the Bonferroni correction.

% \todo{CUT: There is a limit, the strategy of association is fundamentally unsuited to rare variants.}
%
%     Rare Variants Association Analysis in Large-Scale Sequencing Studies at the Single Locus Level
%     https://doi.org/10.1371/journal.pcbi.1004993
%     Despite being extremely successful for common variants in GWA studies [43–46], procedures based on false-positive control are often underpowered in NGS studies involving rare variants (as illustrated in Fig 1).
%     In NGS studies with rare variants, the Signals region often degenerates due to extremely low MAF and high dimensionality.
%
% GWAS arrays don't tag rare variants very well:
%
% https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3830981/
% When the allele frequencies of two loci are very different, the r2 will never
% be very large. To see this, let's assume pA<<PB and 1-pA≈1. Then the maximum
% r2≈pA/pB*(1-pB), which is <<1 since PA is very small compared to pB. This
% indicates that causal rare variants are mostly likely to be missed in GWAS for
% single marker tests, since GWAS chips are designed to include predominantly
% common variants (e.g. MAF>0.05).
%
% Therefore, GWASs are by design powered to detect association with causal variants that are relatively common in the population.
% \autocite{visscher2012FiveYearsGWAS}

% Also see:
% 2007: Wellcome Trust Case Control Consortium landmark GWAS
% 2009: Well-known paper: Finding the missing heritability of complex diseases https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2831613/
% 2018: Revisiting the Missing Heritability of Complex Diseases, Ten Years On https://www.genome.gov/event-calendar/Revisiting-Missing-Heritability-of-Complex-Diseases-Ten-Years-On
%
% NOTE: a second approach to heritability estimation is linkage disequilibrium (LD) score regression.
%
% \todo{CUT: Missing heritability}
%
% \1 Missing heritability refers to the observation that SNP-based heritability estimates from \gls{GWAS} fall short of additive (narrow-sense) heritability estimates from traditional quantitative methods such as twin studies.
% Perhaps unsurprisingly, it has been hypothesised that the remaining heritability lies in variants that can not be assessed by \gls{GWAS} due to rarity or small effect.
% A classic example is the heritability of human height, estimated at 80\% by twin studies \autocite{maher2008PersonalGenomesCase},
% where considering only significant associations from \gls{GWAS} explains 5\% \autocite{maher2008PersonalGenomesCase}.
% Consideration of all common variants using mixed models (\software{GCTA}) increases the estimate to 45\% \autocite{yang2010CommonSNPsExplain},
% Recent work (\software{GREML-WGS}) suggests the full estimate of 80\% might be recoverable by also including rare and poorly tagged variants measured by \gls{WGS} \autocite{wainschtein2019RecoveryTraitHeritability}.

% \todo{CUT: WGS/WES migitates the coverage bias of GWAS towards known variation.}
%
% \1 approx 10-fold increase in variants per tech upgrade
% \1 WES/WGS is a trade-off between sample size and genomic coverage
%     \2 Allows tdiscovery and association with rare and novel variation, including structural variants.
%     \2 "In addition, other genome-wide scans, such as WES and WGS studies, allow testing for a burden of rare variants across shared functional units (e.g., genes) in a way that is not accessible to GWASs."
% \1 WES (covers about 40Mbp of the genome)
%     \2 covers more of the genome than GWAS
%     \2 but lower n, so lower power to do single variant associations
%     \2 needs 50x: variable coverage due to pulldown
% \1 WGS
%     \2 there is a tradeoff between variant capture (n needed to observe variant) and sequencing depth (gives confidence to call variants)
%     \2 20x ok to call 90\% of singletons
%     \2 rare variants, including in nc regions
%         \3 current discovery biases, finding higher effect size vars first
%         \3 burden tests (e.g. SAIGE)
%             \4 review \url{https://www.nature.com/articles/s41576-019-0177-4}
%     \2 also gets structural variants

\subsection{From locus to causal variant}

\1 By design, a significantly-associated variant from a \gls{GWAS} needs not be a variant that causally affects the trait, and may only tag a causal variant.
    \2 Fine-mapping is the process of determining which of the many correlated variants at a \gls{GWAS} locus are causal.
    % For sparse enough problems, most other Bayesian methods provide posterior probabilities on "models" by enumerating (CAVIAR), schochastic search (FINEMAP) or sampling (BIMBAM) from all possible combinations of variables.
    % https://stephenslab.github.io/susie-paper/manuscript_results/motivating_example.html
    \2 State-of-the-art methods (e.g. PAINTOR, CAVIARBF, FINEMAP \url{https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6050137/}, SuSiE) provide Bayesian posterior probabilities that associated variants are causal, and some methods can consider the presence of multiple causal variants at the same locus \autocite{schaid2018GenomewideAssociationsCandidate}.
    \2 Even if a single causal variant cannot be assigned, a credible set can
    \2 Power: to separate causal and tag variants depends on \gls{LD} and sample size \autocite{visscher201710YearsGWAS}. \url{https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6050137/}
    \2 Resolution: Naturally, these methods assign probabilities assuming the causal variant is in the set of variants observed.
    \2 The causal variant must either be genotyped or confidently imputed. Denser genotyping e.g. by WGS, and larger imputation panels will help.

\subsection{From causal variant to target gene via expression}

\1 Unlike for Mendelian traits where most causal variants are coding \url{https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4573249/}, 
over 90\% of \gls{GWAS} loci fall in non-coding regions of the genome \autocite{gallagher2018PostGWASEraAssociation},
and often too far from the nearest gene to be in \gls{LD} \url{https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5291268/}.
Thus even if the causal variant at a locus is fine-mapped, 
it may not be obvious how to find the target genes through which that variant affects the trait.
% \2 For two-thirds of \gls{GWAS}-associated complex trait loci where target genes have been assigned, the implicated gene is not the nearest gene \autocite{visscher201710YearsGWAS}.
%
% https://twitter.com/Eric_Fauman/status/1198595609013489670
% I tweeted this diagram 2 weeks ago and it's applicable here too. The sentence
% in the paper is true if restated to be specific to eQTLs, but is demonstrably
% not true for pQTLs. Metabolite QTLs are also most likely to be caused by the
% closest gene: https://ncbi.nlm.nih.gov/pmc/articles/PMC6326795/

% TODO: read cano-gamez2020GWASFunctionUsing
\1 Rather than directly impacting the coding sequence of a gene, 
many non-coding GWAS loci are thought to affect traits by affecting the regulation of target genes \autocite{gallagher2018PostGWASEraAssociation}.
\gls{GWAS} loci are enriched in regulatory elements annotated by functional genomics studies, such as
    % Transposase-Accessible Chromatin followed by sequencing (ATAC-seq) is a simple protocol for detection of open chromatin.
    regions of open chromatin, 
    % DNase-seq
    DNase I hypersensitive sites,
    splice sites, UTRs,
    % ChIP-seq combines chromatin immunoprecipitation (ChIP) with massively parallel DNA sequencing to identify the binding sites of DNA-associated proteins.
    histone binding sites, 
    \gls{TF} binding motifs,
    and enhancers \autocite{trynka2015DisentanglingEffectsColocalizing,gallagher2018PostGWASEraAssociation} \url{https://genome.cshlp.org/content/22/9/1748.full}.
For complex diseases, enrichment is observed in disease-relevant tissues \autocite{visscher201710YearsGWAS}.
As these regulatory elements not only have \textit{cis}, but also \textit{trans} regulatory effects on gene expression,
these enrichments put forth expression as an important molecular phenotype linking non-coding \gls{GWAS} variants to their associated traits (\autoref{fig:intro_sysBio}).

% \1 The subset of heritable traits that are not only complex but continuous are called quantitative traits, and genetic associations for those traits are called \glspl{QTL}.
% \1 <basics of eQTLs> \autocite{westra2014GenomeFunctionStudying,albert2015RoleRegulatoryVariation,vandiedonck2017GeneticAssociationMolecular}
\1 Studies of the genetic architecture of quantitative molecular phenotypes have further reinforced this hypothesis.
    \2 Molecular phenotypes like expression are heritable complex traits \autocite{gaffney2013GlobalPropertiesFunctional}
    \2 Expression can be assayed by e.g. array or RNAseq
    \2 The variants associated with expression are called \glspl{eQTL}.
    % \2 The per-variant effect sizes on molecular phenotypes can be large \autocite{visscher201710YearsGWAS}.
    \2 eQTLs can also be cis- or trans- to their target gene \autocite{albert2015RoleRegulatoryVariation}.
    \2 Their effect size declines with distance to the TSS, so most eQTLs detected are \textit{cis}, and within 1Mb \autocite{vandiedonck2017GeneticAssociationMolecular}

\1 GWAS variants are enriched for eQTLs \url{https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1000888}
    \2 So GWAS loci that are also eQTL naturally prioritise target genes.
    \2 Is it a narrow view to assume that the effect of GWAS loci on complex traits not only act through a target gene, but are specifically mediated by eQTL effects?
    \2 Over many complex traits, a median of 11\% heritability could be explained by mediation of GWAS loci by common (MAF > 0.01) cis-eQTL, 
    and this proportion does not include \textit{trans} or post-transcriptional effects.

\1 With increasing sample size, most genes (60-80\%) have a detectable eQTL \autocite{vandiedonck2017GeneticAssociationMolecular}.
Assuming that a locus on the genome is associated with both a complex trait and an \gls{eQTL},
how can we separate the scenario where one variant affects both trait and expression (pleiotropy),
from coincidental overlap between distinct causal variants that may possibly in \gls{LD}?
Bayesian probabilistic colocalisation methods (e.g. eCAVIAR, Sherlock, coloc \autocite{wallace2020ElicitingPriorsRelaxing}) 
address this by estimating the posterior probability that the same causal variant is associated with both phenotypes.
\todo{add uses other vars}
distinguishing pleiotropy from linkage, 
but not vertical pleiotropy (mediation) from horizontal pleiotropy (independent effects on trait and expression) \autocite{hemani2018EvaluatingPotentialRole}.
As colocalisation of a \gls{GWAS} loci with \glspl{eQTL} is is necessary but not sufficient for mediation, 
it should be supported by complementary lines of evidence from other methods that integrate intermediate phenotypes (TWAS, MR, mediation analysis etc.) \autocite{hemani2018EvaluatingPotentialRole}
to help untangle the multiplex of possible causal pathways from variant to trait.
%
% Other types of methods:
% Summary: https://sashagusev.github.io/2017-10/twas-vulnerabilities.html
% transcriptome association (e.g. TWAS)
%    https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6342197/
%    As TWAS methods were originally proposed as tests for association between
%    local genetically regulated component of expression and disease with no
%    causality guarantees [Gamazon, et al. 2015; Gusev, et al. 2016; Mancuso,
%    et al. 2017a; Mancuso, et al. 2017b; Zhu, et al. 2016], it remains unclear
%    whether and when TWAS can be interpreted as valid tests of causality.
%
%     https://www.nature.com/articles/s41576-018-0020-3
%     Transcriptome-wide association studies (TWAS) integrating GWAS and eQTLs
%     data have been proposed to unravel gene–trait associations7,9,10. However,
%     although these studies aim to identify genes whose (genetically predicted)
%     expression is significantly associated to complex traits, they do not aim
%     to estimate the strength of the causal effect and are unable to distinguish
%     causation from horizontal pleiotropy (i.e., when a genetic variant
%     influences multiple phenotypes independently).
%
%     https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5986723/
%     Like other transcriptome-wide association studies (TWASs),32 PrediXcan can be
%     considered a weighted burden test, where each variant in a gene set is weighted
%     by its additive allelic effect on expression.
%
% Mendelian randomisation (e.g. MR-Egger, SMR) under certain assumptions
%     hemani2018EvaluatingPotentialRole
%     Of prime focus among the many limitations to MR is the unprovable assumption
%     that apparent pleiotropic associations are mediated by the exposure (i.e.
%     reflect vertical pleiotropy), and do not arise due to SNPs influencing the two
%     traits through independent pathways (‘horizontal pleiotropy’)
%
% Mediation
%     hemani2018EvaluatingPotentialRole
%     Genetic mediation-based analyses (37–40) are more liable to problems of
%     confounding and measurement error than MR (41–43), but could potentially
%     separate between vertical and horizontal pleiotropy in some scenarios.
%     Use genetic colocalization to eliminate possibility distinct causal variants
%     (25,30,31); if instruments are available for the outcome then test the reverse
%     causal effect (110); if not use MR Steiger (43); use genetic mediation-based
%     analysis (40,111) to try to separate horizontal and vertical pleiotropy
%
%     millstein2009DisentanglingMolecularRelationships
%     Causal effect estimates often
%     considered in 'Mendelian randomization' approaches [11], can be confounded by
%     pleiotropic effects and reverse causation [12], thus, these approaches are not
%     generally considered for problems such as reconstructing transcript regulatory
%     pathways, in which pleiotropy is common and there may be little a priori
%     information on the structure of the causal relationship between traits.

\begin{figure}
    \centering
    \includegraphics[width=1.0\textwidth,page=1]{mainmatter/figures/chapter_01/fig_mockup_systemsBio_Screenshot 2020-05-21 at 17.08.47.png}
    \caption{Mediation of genetic effect to phenotype, through the biological system}
    \label{fig:intro_sysBio}
\end{figure}

\section{Genetic effects on expression: environment is key}

\1 The effects of \glspl{eQTL} (and molecular \glspl{QTL} in general) are incredibly context-dependent \autocite{albert2015RoleRegulatoryVariation,vandiedonck2017GeneticAssociationMolecular}.
    \2 This represents genotype-environment interactions at those eQTL.
    \2 A non-exhaustive list of environments that \glspl{eQTL} have been found to interact with:
        \3 sex, age \url{https://academic.oup.com/hmg/article/23/7/1947/655184}
        \3 ancestry \autocite{dejager2015ImmVarProjectInsights,nedelec2016GeneticAncestryNatural,quach2017LivingAdaptiveWorld}
        \3 tissue \autocite{nica2011ArchitectureGeneRegulatory,aguet2017GeneticEffectsGene}
        \3 cell type composition in bulk samples \autocite{westra2015CellSpecificEQTL,zhernakova2017IdentificationContextdependentExpression,glastonbury2019CellTypeHeterogeneityAdipose,kim-hellmuth2019CellTypeSpecific}
        \3 individual cell type \autocite{dimas2009CommonRegulatoryVariation,dejager2015ImmVarProjectInsights,peters2016InsightGenotypePhenotypeAssociations,chen2016GeneticDriversEpigenetic,kim-hellmuth2019CellTypeSpecific}
        \3 disease status \autocite{peters2016InsightGenotypePhenotypeAssociations},
        \3 and experimental stimulation (see \autoref{subsec:intro_reQTL}).

\1 Given the effect of an eQTL can be starkly different between environments, it is difficult to determine the appropriate \gls{eQTL} dataset to use for target gene prioritisation at GWAS loci.
\2 It has already been shown that use of cell-type specific eQTLs increases coloc rates with GWAS hits \autocite{kim-hellmuth2019CellTypeSpecific} \url{https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4498151/} \url{https://www.biorxiv.org/content/10.1101/2020.01.15.907436v1}
    \2 Successful colocalisation of GWAS loci with coloc may prioritise not only the target gene, but the specific environments most relevant to a trait.
    % TODO:  coloc then finemap: also can help fine mapping, as large fx = smaller n required = able to do denser genotypeing by e.g. wgs, so smaller credible sets at much lower sample sizes https://www.biorxiv.org/content/10.1101/2020.01.15.907436v1

\1 What molecular mechanisms might facilitate genotype-environment interactions at \glspl{eQTL}?
    \2 \autocite{ackermann2013ImpactNaturalGenetic}: defines static, conditional, dynamic eQTLs
    \2 \textcite{fu2012UnravelingRegulatoryMechanisms}: proposes TF-based mechanisms for cis-eQTL (here, define mag, damp, flip) (\autoref{fig:intro_reQTLmechs})
    \2 \textcite{gaffney2013GlobalPropertiesFunctional,rotival2019CharacterisingGeneticBasis}: suggests info on more regulatory layers will help break down transcriptional and post-transcriptional

\begin{figure}
    \centering
    \includegraphics[width=1.0\textwidth,page=1]{mainmatter/figures/chapter_01/fig_mockup_reQTLs_Screenshot 2020-05-21 at 17.08.49.png}
    \caption{eqtl mech models: magnify, dampen, flip}
    \label{fig:intro_reQTLmechs}
\end{figure}

\subsection{Immune \glsfmtlongpl{reQTL}}
\label{subsec:intro_reQTL}

\1 A important subclass of context-dependent \gls{eQTL} are \gls{reQTL}, where the interacting environment is experimental stimulation \autocite{vandiedonck2017GeneticAssociationMolecular,huang2019GeneticsGeneExpression}.
Most \gls{reQTL} studies to date have been conducted on immune cells \textit{in vitro}, 
not only because the immune system is specialised for responding to environmental exposures,
but due to the abundance of immune cells easily accessible in peripheral blood,
and amenable to separation (e.g. FACS) and stimulation.
    \2 \textit{In vitro}, potential interacting variables such as cell type, and the nature, length, and intensity of stimulation can be precisely controlled.

\1 A seminal early study was conducted by \autocite{barreiro2012DecipheringGeneticArchitecture}, where eQTLs were mapped separately in monocyte-derived dendritic cells before and after 18h infection with \textit{Mycobacterium tuberculosis}.
    \2 reQTLs were detected for 198 genes, 102 specific to the uninfected state, and 96 specific to the infected state. 

    \2 Since then, \textit{in vitro} immune reQTL studies have been conducted for a variety of 
    cell types
        (e.g. primary CD14+ monocytes \autocite{fairfax2014InnateImmuneActivity}) 
    and stimulations 
    (IFN$\gamma$ and LPS \autocite{fairfax2014InnateImmuneActivity}).

    \todo{list a few from \autocite{fairfax2014InnateImmuneActivity} until \autocite{alasoo2019GeneticEffectsPromoter}}
    % 	Found reQTLs? So what? Gather take home messages and inspiration for post reQTL analyses
        % •	reQTL effect characteristation
        % o	reQTLs can show reversal of effect between conditions (Fairfax et al. 2014)
        % •	reQTLs vs expression
        % o	enrichment in DE genes vs eQTLs and non eQTLs (Barreiro et al., 2012)
        % o	context-specific eQTL are identified because of both treatment-induced regulatory effects and treatment-inducing gene expression to detectable levels (Fairfax et al. 2014)
        % •	trans effects
        % o	reQTLs develop trans-effects on stimulation (Fairfax et al. 2014)
        % •	reQTLs in pathways
        % o	Fairfax et al. 2014: reQTLs frequently intersected established canonical pathways of monocyte signaling
        % •	reQTLs enrichment in relevant GWAS hits
        % o	Barreiro et al., 2012
        % o	Fairfax et al. 2014
        % •	genomic feature enrichment of reQTLs in certain feature classes
        % o	UTRs (Fu et al., 2012)
        % •	Overlap of reQTL genes with DE genes
        % o	Franco et al. 2014
        % •	Mediation analysis of eQTL -> Ab response (Franco et al. 2014)
        % •	Colocalisation with GWAS (Kim-Hellmuth et al. 2017)
        % •	motifs enriched at reQTL binding sites e.g. STATs, IRFs (Caliskan et al. 2015; Davenport et al. 2018)
        % 	Gather reQTL datasets for coloc
        % •	Fairfax 2014 stim monocytes
        % •	CEDAR range of resting cells
        % •	Schmiedel_2018 stim CD4/8s
        % •	Alasoo 2018 stim macro
        % •	… and whole blood meta control

\1 A complementary approach is \textit{in vivo} \gls{reQTL} mapping
    \2 There are numerous pros to \textit{in vivo} stimulation.
        \3 the innumerable interactions in the immune system that are absent \textit{in vitro}
        \3 ability to get whole organism phenotypes
        \3 ability to get repeated measures: can reason about change in expression over time
    \2 Major disadvantages: 
        the choice of stimulation must be ethical \textit{in vivo}, 
        and many environmental factors (e.g. diet, lifestyle, immune exposures) cannot be controlled, leading to greater experimental noise (?), and more complex interpretations.

    \2 There are few published \textit{in vivo} \gls{reQTL} studies.
        \3 \autocite{franco2013IntegrativeGenomicAnalysis}: seasonal \gls{TIV}, whole blood, antigen processing and intracellular trafficking genes, attempted mediation for Ab titres, but underpowered
        % In addition to DIABLO and TRAPPC4 identified above, several genes identified in the epistatic analysis help regulate apoptosis, including FADD,22 ITK23 and REG3A.24
        \3 \autocite{lareau2016InteractionQuantitativeTrait}: fold-change expression after inactivated vaccinia vaccine, focus was on pairwise epistatic interactions, apoptosis pathways
        \3 \autocite{davenport2018DiscoveringVivoCytokineeQTL}: whole blood, IFN status and anti-IL6 drug exposure, reQTL driven by ISRE and IRF4 motifs

\1 <why care about immune reQTLs>
    \2 Exposes differences in regulatory architecture between conditions, but does not automatically reveal the mechanisms behind those differences 

    \2 Immune \textit{in vitro} \gls{reQTL} have been shown to be enriched more so than non-\gls{reQTL} among GWAS loci for immune-related phenotypes such as
    susceptibility to infectious \autocite{barreiro2012DecipheringGeneticArchitecture,manry2017DecipheringGeneticControl}
    and immune-mediated diseases \autocite{manry2017DecipheringGeneticControl,kim-hellmuth2017GeneticRegulatoryEffects}.

    \2 Not yet clear whether \textit{in vivo} reQTL have any utility on top of \textit{in vitro} reQTL for interpreting GWAS loci: not that many studies, and complex interpretations.

    \2 Nevertheless, as the number of cell types systems and stimulations both \textit{in vitro} and \textit{in vivo} increases, the number of known reQTLs continues to grow.

\section{Immunity is a complex trait}

\1 Heritability of immune phenotypes is not only restricted to the expression phenotypes discussed above.
\todo{a little out of order, but since most reQTL studies are immune, I went context-specific -> reQTL -> immune rather than context-specific -> immune -> reQTL}
    \2 Studies of interindividual variation in the healthy immune system shows many aspects of the immune system are heritable and complex.
    \2 Immune parameters are influenced by age, sex, seasonality, and chronic infection \autocite{brodin2015VariationHumanImmune,liston2016ShapingVariationHuman,brodin2017HumanImmuneSystem,patin2018NaturalVariationParameters,liston2018OriginsDiversityHuman} \url{https://www.nature.com/articles/ncomms8000},
    but most individuals have a healthy baseline immune state that is individual-specific, 
    and relatively stable over time \autocite{liston2016ShapingVariationHuman,brodin2017HumanImmuneSystem,lakshmikanth2020HumanImmuneSystem}.
    \todo{stable, yet varies by age?}
    \2 Overall estimates of the heritability of many immune parameters, such as cell composition and serum protein levels, lies between 20-40\% \autocite{liston2016ShapingVariationHuman,brodin2017HumanImmuneSystem,patin2018NaturalVariationParameters,liston2018OriginsDiversityHuman}
    \2 Genetic regulation is more important for the innate immune system than the adaptive immune system \autocite{patin2018NaturalVariationParameters}.

\1 A central goal of systems immunology is to establish causal relationships between the many parts and levels of the immune system
    \2 Natural genetic variation represents small scale perturbation that is causally anchored \autocite{tsang2015UtilizingPopulationVariation,villani2018SystemsImmunologyLearning}
    \2 But as discussed in the context section above, specific effects may not be apparent in the baseline state, stimulation is required
    \2 Studies of natural infection are complicated by e.g. determining exposure.
    \2 As in the immune in vivo reQTL studies, vaccines and drugs used as controlled immune perturbations to study the activated immune system
    \2 reQTL may have utility for interpretation of these immune-related complex traits too, not just IMIDs/infectious disease discussed in reQTL section above

\subsection{Immune response to vaccination is a complex trait}

\1 Vaccination has enormous impact on global health \autocite{greenwood2014ContributionVaccinationGlobal}
    \2 <quick vaccine bio>
        \3 Vaccines stimulate the immune system with pathogen-derived antigens to induce effector responses (primarily antigen-specific antibodies) and immunological memory against the pathogen itself.
        \3 These effector responses are then be rapidly reactivated in cases of future exposure to the pathogen, mediating long-term protection.
        \3 <...>
    \2 A vaccine that is highly efficacious in one human population may have significantly lower efficacy in other populations.
    Particularly challenging populations for vaccination include the infants and elderly, pregnant, immuno comprimised patients, ethnically-diverse populations, and developing countries.
        \3 <1 example statistic on vaccine efficiacy differences e.g. rotavirus>
        \3 e.g. \url{https://www.sciencedirect.com/science/article/pii/S1473309918304900}
    \2 Traditional vaccine dev is empirical (classical "isolate, inactivate, inject" paradigm), often successful vaccine dev does not offer insights into the mechanisms of efficacy 
    \2 The immunological mechanisms that underpin a specific vaccine's success or failure in a given individual are often poorly understood.
    % <For the majority of licensed vaccines, there is a lack of understanding regarding the molecular mechanisms that underpin this variation in host immune response.>

\1 A sub-discipline of systems immunology is systems vaccinology.
    \2 Systems vaccinology is the application of -omics technologies to provide a systems-level characterisation of the human immune system after vaccine-perturbation.
    % \2 These systems vaccinology studies often consider longitudinal measurements of the transcriptomic, cellular, cytokine, and antibody immune responses following vaccination.
    % \2 Measurements are taken at multiple molecular levels (e.g. genome, transcriptome, proteome), and molecular signatures that correlate with and predict vaccine-induced immunity are identified.
    \2 Systems vaccinology has been successfully applied to a variety of licensed vaccines [yellow fever, influenza], and also to vaccine candidates against [HIV, malaria], resulting in the identification of early transcriptomic signatures that predict vaccine-induced antibody responses.
    \todo{define what a signature is}
        \3 <add more to list of what vaccines have been studied, pull out of sysvacc\_review\_docx>
    \2 Sysvacc informs more mechanism-based and cost-effective design (rational paradigm), and the move towards personalised vaccinology.
    \2 Sysvacc has revealed many influences on vaccine response (age, sex, dose, adjuvants, expression signatures, microbiome, strain etc.)
    \2 Studies of impact of host genetics is underrepresented \autocite{linnik2016ImpactHostGenetic}

\1 Like for other complex traits, from twin studies it's known that vaccine Ab responses are heritable.
    \2 Moving out of the candidate gene era (e.g. \url{https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3570049/}) into GWAS.
    \2 \autocite{oconnor2013CharacterizingVaccineResponses} has heritability estimatates
    \2 Many loci have been implicated by GWAS e.g. HLA \autocite{oconnor2013CharacterizingVaccineResponses,mooney2013SystemsImmunogeneticsVaccines,mentzer2015SearchingHumanGenetic,linnik2016ImpactHostGenetic,scepanovic2018HumanGeneticVariants,dhakal2019HostFactorsImpact}
    \todo{find best GWAS ref, probably mooney2013SystemsImmunogeneticsVaccines, then prune and reassign these citations}
    \2 Overall, systems vacc studies that include genetics are nowhere near as mature compared to the trait to gene pipeline described in above e.g. for immune-mediated disease

    \subsection{Immune response to (biologic therapies for immune-mediated diseases) is a complex trait}
\todo{WIP, need to do more reading up on PANTS background.}

\todo{Not yet sure if I need to go as wide as X=biologics for IMIDs, or only as narrow as anti-TNFs for CD}
\1 There is variation in response success to X for immune-mediated disease
\1 Many subphenotypes may be complex: immunogenicity, primary response, loss of response, remission rate, adverse fx
\1 Existing work on expression response to X 
\1 Existing work on genetic factors for X 

\section{Thesis overview}

\1 <By chapter context-content-conclusion overview.>
    \2 <ch 2: systems vaccinology study of Pandemrix>
        \3 meta analysis
    \2 <ch 3: in vivo reQTL study of Pandemrix>
        \3 Gap: relatively few studies have assessed the impact of human genetic variation on responses.
    \2 <ch 4: systems immunology and reQTL study of response to anti-TNF treatment in CD>
    \2 <discussion: limitations, future outlook>

\end{outline}

