%
% Chapter 5
% Discussion
%

\chapter{Discussion}
\label{ch:discussion}

% TODO: autoref to cref
% TODO: check each section provides some solutions
\begin{outline}

\1 In this thesis, ... <a summary of the thesis projects>
    \2 Chapters 2 and 4 focus on describing the transcriptomic response of circulating immune cells to perturbation with uncontrolled longitudinal designs
        \3 looking at how expression changes over time, and differences in transcriptomic response associated with phenotypic definitions of response
        \3 associations are identified at a gene and gene set level
    \2 Chapter 3 (and the last part of chapter 4) focus on finding common genetic variants associated with transcriptomic response to perturbation
        \3 reQTLs are identified at lead eQTL variants for each gene
        \3 this is a first foray into defining causal molecular mechanisms that explain individual variation in response to these perturbations
    \2 Each chapter discusses the results and limitations of each study, but being very similar studies in design and analysis, they also suffer from a core set of limitations.
    \2 This chapter discusses these, and considerations for design and analysis of similar longitudinal studies in the future to better understand the biology of immune response to drugs and vaccines.

\section{Increasing the power to detect associations}

\1 It has been challenging to find robust single gene-level associations between expression and response, especially at baseline.
    \2 The biological signal may be small compared to other sources of variation: measurement platform (ch2); drug, response definition (ch4)
    \2 The obvious thing to do is increase sample size.
        \3 Under a fixed resource constraints, longitudinal studies tradeoff more samples over time for fewer individuals
        \3 Multiomic designs further tradeoff against richness/number of layers sampled
    \2 there are distinct advantages to these designs
        \3 e.g. repeated measures allows modelling of within patient covariance between timepoints with mixed model DGE (ch2 and ch4)
        \3 e.g. spline analysis and clustering in ch4 using multiple timepoints to separate responders and nonresponders
        \3 e.g. ability to test for association between layers (eQTLs), to explore molecular mechanisms
    \2 but especially if the intent is to consider the small effect of genetic variants, a lesson learnt in the field of complex diseases from years of underpowered candidate gene studies, is that even a single complex phenotype requires large samples
        % \3 systems vaccinology studies have always had these rich datasets, but rarely considered genetics, limited by sample size
        % \3 it is also challenging to collect sufficient sample size in a pharmacogenomic context \url{https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3003940/}

\1 with rich datasets, be careful about subgroup analyses, subgrouping quickly attenuates your effective sample size
    \2 e.g. looking for timepoint, responder, cell type interactions
    \2 be wary of rules of thumb for sample size,         
        \3 e.g. in ch3, the intention was to map eQTLs at each timepoint, look for timepoint x G interaction, but it became apparent that cell proportion interactions were necessary for interpretation
    \2 prespecify interactions of interest at the power analysis stage in design of new experiments
    \2 for existing datasets, a workaround like the two stage strategy in ch3, assuming that interactions are only interesting at significant main effects

\1 Rather than increasing the sample size, can change the unit of analysis from single gene to multiple genes
    \2 use information from coordinated changes in multiple genes to detect subtle effects
    \2 use prior biological knowledge to define set of genes
    \2 this is the basis for gene set analyses used throughout this thesis
    \2 tradeoff between better interpretability but worse resolution

\1 <reproducibility of gene set vs single gene associations> 
    \2 reports about any single gene associations e.g. SIGLEC10 in ch4 of this thesis, should be treated with caution pending replication
    \2 while the rank of a single gene may vary from sample to sample due to sampling noise, a summary measure computed from multiple genes should be more robust
    \2 it is hoped that the replicability of gene sets will be greater

\section{Responder analysis}

\1 <how well these models correspond to reality>
    \2 What does it mean to be a responder?
    \2 this has been discussed on this in the field of personalised medicine in the context of RCTs, but similar issues pertain to response in observational studies. \autocite{senn2018StatisticalPitfallsPersonalized}
    \2 Is being a responder is a stable personal characteristic of an individual over time, or would the same individual vary in response status over time?
    % Anchorman: The Legend of Ron Burgundy (2004) - Paul Rudd ...
    % Brian Fantana : They've done studies, you know. 60% of the time, it works every time. [cheesy grin]. Ron Burgundy : That doesn't make sense ..
    \2 Observing between individual variation in response for a single measure is not sufficient to imply response is personal.
    \2 e.g. a hypothetical agent with a 60\% response rate in a group of individuals: \enquote{60\% of the time, it works every time}
    \2 what does this mean? working 60\% of the time on 100\% of individuals, 100\% of the time on 60\% of individuals, or anything in between would be compatible
    \2 cannot distinguish with a single measure, yet the assumption is often made at the study design stage that it is the 2nd option
    \2 1st option means no associations to detect

\1 Need to establish how consistent response is over time
    \2 to determine within-individual variation, needs replication at the level of the individual, i.e. need to perturb and measure the same person more than once \autocite{senn2016MasteringVariationVariance}
        \3 in ch4, repeated anti-TNF dosing showed expression differences between responders and NR, at least, were stable over time
        \3 although probably partially due to selection of non-remission by w54
    \2 note for the specific case of genetic factors, the field of complex trait genetics has long had a solution for this: twin studies are analogous to measuring the same individual twice
        \3 For phenotypes like vaccine response in ch2 that can be, twin studies have already demonstrated heritability of certain response phenos like influenza vaccine Ab titres \autocite{brodin2015VariationHumanImmune}
        \3 In pharmocogenetics, twin studies are less common \url{https://www.sciencedirect.com/science/article/abs/pii/S0928098719300326} and have not been done for response to anti-TNFs (the ch4 setting)
        \3 repeated crossover trials can be a feasible alternative to twin studies \url{https://www.researchgate.net/profile/Laszlo_Endrenyi/publication/13553486_Hypothesis_Comparisons_of_inter-_and_intra-individual_variations_can_substitute_for_twin_studies_in_drug_research/links/59e97c2e458515c36370e6a7/Hypothesis-Comparisons-of-inter-and-intra-individual-variations-can-substitute-for-twin-studies-in-drug-research.pdf} \autocite{senn2016MasteringVariationVariance}

\1 <How a responder to a perturbation was modelled in this thesis>
    \2 As discussed in \autoref{subsec:hird_dge_TRI} in ch2, a binary definition of response based on dichotomisation is inefficient and assumes a sharp change in biological relevance at the threshold
    \2 In ch2, used \gls{TRI}, a continuous change score residualised on the baseline, which gets around the dichotomisation and dependence of change score on baseline
        \3 two-stage approach here, so df is wrong
    \2 In multiPANTS, \autoref{multiPANTS:PR_definition} in ch4 sets out the binary definition of response, which is complex and based on many inputs
        \3 having response dichotomies based on multiple inputs can lead to large discontinuities and non-monotonicity in response probabilities under small changes in input \autocite{senn2005DichotomaniaObsessiveCompulsive} 
    \2 neither approach perfect, but the driving force was pragmatism
        \3 for DGE, the model traditionally requires the response variable to be gene expression, and a predictor variable to be response status
        \todo{might be worth subbing predictor for independent and response with dependent in all chapters, but then that gets confused with statistical independence...}
        \3 thus I needed an approach that generates a single predictor variable to represent response status
        \3 In ch2, needed to combine pre-post titre data from two assays
        \3 In ch4, the clinical algorithm already combines many measures into one.
    \2 having complicated response definitions can hinder replication by adding to interstudy variation e.g. the failure to replicate the TREM1 association in ch4
    \2 it may be better to directly with the constituent phenotypes: ch2 (log titre levels), ch4 (CRP level, HBI score), but then the fundamental modelling approach for both DGE/eQTL would need to change to not be restricted to a single response x variable 
    \2 also applies to continuous measures of response, as what is being considered is that it only makes sense to talk about response if the response measure is tightly correlated within an individual over time

% \1 bias in non-randomised studies
    % \2 In RCTs, the treatment effect would be defined against the effect in the control group
    % \2 In prospective cohort studies, the treatment effect is determined between two subgroups within the cohort

\section{Challenges in the interpretation of bulk expression data}

\1 NGS data is compositional, only quantifies relative abundance \todo{move and merge this paragraph into ch2 section on composition bias in RNAseq, it's out of place here}
    % https://academic.oup.com/gigascience/article/8/9/giz107/5572529
    % This property emerges from the sequencer itself; the sequencer, by design, can only sequence a fixed number of nucleotide fragments. 
    % Consequently, the final number of fragments sequenced is constrained to an arbitrary limit so that doubling the input material does not double the total number of counts.
    % This constraint also means that an increase in the presence of any 1 nucleotide fragment necessarily decreases the observed abundance of all other transcripts [9], and applies to bulk and single-cell sequencing data alike.
    % It is especially problematic when comparing cells that produce more total RNA than their comparator (e.g., high–c-Myc cells, which up-regulate 90% of all transcripts without commensurate down-regulation [10]).
    % However, even if a sequencer could directly sequence every RNA molecule within a cell, the cells themselves are compositional because of the volume and energy constraints that limit RNA synthesis, as evidenced by the observation that smaller cells of a single type contain proportionally less total messenger RNA (mRNA) [11].
    % [..]
    % First, statistical models that assume independence between features are flawed because of the mutual dependency between components [15]. Second, distances between samples are misleading and erratically sensitive to the arbitrary inclusion or exclusion of components [16]. Third, components can appear definitively correlated even when they are statistically independent [17].
    \2 raw abundance estimates are often called \enquote{counts}, but actually dimensionless proportions, interpretable only compared to other genes in the same sample (composition)
    % https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6084572/
    % Compositional data measure each sample as a composition, a vector of non-zero positive values (i.e. components) carrying relative information (Aitchison, 1986).
    % Compositional data have two unique properties. First, the total sum of all component values (i.e. the library size) is an artifact of the sampling procedure (van den Boogaart and Tolosana-Delgado, 2008). Second, the difference between component values is only meaningful proportionally [e.g. the difference between 100 and 200 counts carries the same information as the difference between 1000 and 2000 counts (van den Boogaart and Tolosana-Delgado, 2008)].
    \2 a fixed number of reads is sequenced: if a gene's counts go up, another will go down
    % https://rnajournal.cshlp.org/content/early/2020/04/13/rna.074922.120.full.pdf
    \2 naive normalisations such as CPM, TPM, RPKM are still rates (per transcriptome), and cannot be compared across samples, as they still depend on composition
    \2 normalisation such as TMM tries to account for composition by normalising on non DE genes, to allow for valid DGE analysis
    \2 getting absolute measures is difficult without spike-ins
    \2 for valid DGE, need to decide on the units e.g. mRNA/cell? mRNA/transcriptome? \autocite{soneson2013ComparisonMethodsDifferential}

\1 bulk data is a mixture of cell types with different expression patterns
    \2 one of the largest sources of variation in bulk expression data is cell proportions 
    \2 the more cell type-specific a gene's expression, the more it is affected by cell proportions \autocite{farahbod2020UntanglingEffectsCellular}
        \3 perfectly cell type-specific genes are often called marker genes, and are used in deconvolution methods to estimate cell proportions from bulk data when they are not directly measured
    \2 both variation from in true variation in proportions, and sampling noise
    \2 in this thesis, estimated by:
        \3 In HIRD ch 3, xcell deconvolution from expression data
        \3 In PANTS ch 4, deconvolution from methylation data
    \2 these are fit as covariates in models
        \3 could act as precision variables (for sampling noise in non cell type-specific genes),
        \3 but also mediators for cell type-specific genes, where the interpretation of DGE effects changes (perturbation -> cell prop -> expression)
    \2 but coarse grained (just the main cell types) adjustment is not always sufficient, you'll still find associations that are proxies for rare cell types \url{https://www.biorxiv.org/content/10.1101/2020.05.28.120600v1}
        \3 Recommend the use of e.g. PEER, as in ch3/4 eQTL analysis, some of those factors are likely to be rarer cell types

\1 <eQTL analysis>
    \2 analogous to DGE where independent variable of interest is now genotype
    \2 as discussed in \autoref{subsec:hird_reQTL_methods_cellTypeInteraction}, it is model misspecification to not have the genotype x cell proportion interaction if the effect of genotype changes depending on cell proportions
        \3 in fact, it is popular to use genotype x cell proportion (or a proxy of) interactions to call cell type-specific eQTLs e.g. \autocite{westra2015CellSpecificEQTL,kim-hellmuth2020CellTypeSpecific}
        \3 but you cannot distinguish between correlated cell types this way

\1 <in vivo reQTL analysis>
    \2 here the cell props are causally affected by the perturbation (active recruitment, differentiation, proliferation of immune cells)
    \todo{would an in vitro stimulation of a mixture of cells be immune to this? differentiation and proliferation could still occur}
    \2 consider the contrived case, where vaccine perturbation causes active increase of a rare cell type that is near absent at baseline, but is a greatly increased proportion of the bulk mixture after perturbation.
    \2 further assume this is true for all individuals (no variation in cell proportions).
    \2 an eQTL specific to this cell type will show up as a reQTL in the post-perturbation timepoint because expression of that cell type contributes more to the bulk mixture.
        \3 this will happen even if you adjust for proportion of the cell type, since there is no variation between individuals, it will only offset the regression line
        \3 this does not require the gene to be upregulated on average, as the effect of interest is G -> delta E, not delta E.
    \2 if the aim of a reQTL study is to identify variants that causally affect transcriptomic response to perturbation, 
        such that if we changed the genotype of an individual (observe the counterfactual), the transcriptomic response would be different.
        \todo{assumes eQTL is finemapped}
    \2 does this achieve that aim? 
        yes, as the change in expression after perturbation differs when you change someone's genotype,
        as there is no difference between genotypes at baseline,
        but there is after perturbation.
        \todo{I cannot figure out what sort of conditioning would need to be done to block this sort of effect even if you had perfect cell proportion measurements}
    \2 for genotype to explain more variance in bulk expression after perturbation, there are multiple compatible mechanisms
        \3 a cell type with a cell type-specific eQTL increases in proportion (recruitment) 
        \3 a gene with an eQTL increases in expression (activation or recruitment depending on if you condition by cell proportions)
        \3 the effect of a cell type-specific eQTL increases within that cell type (activation)
    \2 one could imagine crude ways to falsify each hypothetical model e.g. interaction/coloc with each cell type (ch3), but always at risk of missing a cell type
    \2 the challenge is distinguishing these scenarios in bulk in vivo data to form mechanistic hypotheses for expression response

\1 future studies should consider alternatives to bulk before they can fully take advantage of benefits of in vivo design
    % http://bayes.cs.ucla.edu/PRIMER/ch3-preview.pdf
    % The difference between intervening on a variable and conditioning on that variable should,hopefully, be obvious.
    % When we intervene on a variable in a model, we fix its value.
    % Wechangethe system, and the values of other variables often change as a result.
    % When we condition on avariable, we change nothing; we merely narrow our focus to the subset of cases in which thevariable takes the value we are interested in.
    % What changes, then, is our perception about theworld, not the world itself.
    % \2 control is not intervention
    \2 rather than conditioning on cell proportion, effectively looking at the subset of data with same cell proportions, just control them
    \2 If you want a mechanistic interpretation, need to control
    \2 single-cell RNAseq has the advantage of getting expression and composition simultaneously, and can be comparable in cost to bulk on FACS sorted cells \url{https://arxiv.org/ftp/arxiv/papers/1909/1909.12550.pdf}
        \3 but lower coverage of the transcriptome due to drop out, and smaller sample sizes
    \2 do DGE/map eQTLs within each cell type 
        \3 combines advantage of in vivo stimulation with the controlled cell composition of in vitro studies
    \2 can also be used to annotate bulk reQTLs such as those found in ch3 e.g. find the likely cell types for bulk reQTL effects by looking at eGene expression in single cell data \autocite{devries2020IntegratingGWASBulk}

% more n allows more phenotypes
%
%         disease specific biobanks e.g. ibd bioresource/predicct
%
%    Our GWASfor 54 functionally relevantphenotypes oftheadaptiveimmune system in 489 healthy individuals identifieseight genome-wide significant associations explain-ing 6%–20% of variance.
%         https://www.cell.com/cell-reports/pdf/S2211-1247(18)31493-1.pdf
%
%     PheWAS\autocite{verma2017CurrentScopeChallenges}
%         https://onlinelibrary.wiley.com/doi/full/10.1111/imm.12195
%         PheWAS has the advantage of identifying genetic variants with pleiotropic properties.
%
% and Many more conditions
    % e.g. 250 condition ASE % https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5131815/
    % e.g. StructLMM
    %     Identifies eQTLs with GxE, where the number of environments in E is large (modelled as a random effect)

\section{From association to prediction}

\1 I have now discussed both the independent and dependent variables used in descriptive models of expression vs response
    \2 so far, always put expression as dependent, and response status as independent, and estimated the effect size of response status on expression
    \2 did this even when response is measured after expression (baseline expression)
        \3 this is the assumption that response is a stable individual characteristic
        \3 also assumes it is measured without error, otherwise would stray into the realm of error in variables models
    \2 but might also want to consider if given expression, you could predict response (probability) as a dependent variable
    \2 this is also directly relevant to clinical goal of predicting patient response from baseline expression
    
\1 a large p >> n prediction problem
    \2 given the few gene-level associations with response is it futile to try?
    \2 \textcite{efron2020PredictionEstimationAttribution} provides an encouraging counterexample on n=102 (52 prostate cancer, 50 controls), p=6033 genes
        \3 random forest to predict cancer status from gene expression: 2\% test set error
        \3 removed the 348 genes with positive importance scores, these are all the genes that were ever involved in a splitting rule
        \3 rerun on resulting 102 x 5685 matrix, still a similar error rate
        \3 repeat that process, removing 364 genes. rerun on resulting 102 x 5321 matrix, still a similar error rate
    \2 performance of such purely predictive models is dominated by combining many weak predictors
    \2 pure prediction is an easier problem than the traditional approach trying to attribute statistical significance to a few strong predictors
    \2 this may be why there is sometimes little overlap in the gene signatures from different studies on the same trait

% What sample size?
    % https://www.bmj.com/content/368/bmj.m441
    % Calculating the sample size required for developing a clinical prediction model

\1 Need to consider what it takes for a gene expression prediction model to be clinically useful?
    \2 Useful to look at examples from cancer, one of the earliest fields where the approach generated interest
    \2 Despite the early gene expression signatures being reported in the early 2000s, only a handful in use \autocite{chibon2013CancerGeneExpression} \url{https://clincancerres.aacrjournals.org/content/21/21/4743}
    \2 many hurdles: need to be accurate enough to base changes in therapy on, but incrementally more accurate than existing clinical markers, and cost effective. \url{https://doi.org/10.1093/annonc/mdw307}
    \2 measuring a whole transcriptome is likely not cost effective. what are potential methods that might be?
    \2 for predicting vaccine response from transcriptomic data, popular methods are:
        \3 DAMIP e.g. \autocite{querec2009SystemsBiologyApproach}, which gives rulesets composed of small sets of genes
        \3 elastic net
        % https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2810828/
        % Sparse partial least squares regression for simultaneous dimension reduction and variable selection
        % https://pubmed.ncbi.nlm.nih.gov/20361856/
        % Sparse Partial Least Squares Classification for High Dimensional Data
        \3 sparse partial least squares e.g. \autocite{tsang2014GlobalAnalysesHuman} 
    \2 Note the sparsity assumption for these methods (most genes are not predictive) is in opposition to the idea of combining many weak predictors to maximise predictive power, but will be less costly to translate to clinical practice

\section{From association to causality}

\1 uncovering \enquote{mechanism}
    \2 Knowing the molecular mechanisms of response are crucial for conceiving of possible interventions.
        \3 e.g. ch4 baseline SIGLEC10 expression is associated with anti-TNF response, but would changing baseline SIGLEC10 affect response?
    \2 In observational data, a causally upstream anchor is required: genetic variation
    \2 the regression analyses in this thesis:
        \3 DGE describes expression response association with phenotypic response (such as Ab titre) 
        \3 reQTLs describe genotype association with expression response
    \2 Need a model that encodes causal assumptions integrating multiple data layers: genotype -> gene expression -> phenotypic response
    \2 there are several families of complementary methods that integrate multiple data layers

\1 <Mendelian randomisation>
    \2 <explain the analogy to RCTs: segregation should be independent of env factors>
    \2 is a type of IV analysis that uses genetic instruments: can use eQTLs as instrumental variables (IV) to estimate causal effect between expression (exposure) and phenotype (outcome)
    \2 three assumptions that define a valid IV \autocite{daveysmith2014MendelianRandomizationGenetic,hemani2018EvaluatingPotentialRole,neumeyer2020StrengtheningCausalInference}
        \3 IV1 SNP is associated with expression i.e. an eQTL (no need to be fine mapped \autocite{daveysmith2014MendelianRandomizationGenetic,burgess2018InferringCausalRelationships})
        \3 IV2 SNP is not associated with confounders of the expression-phenotype association 
        \3 IV3 SNP has no association with phenotype except through expression 
    \2 combined, these assumptions mean that expression is a complete mediator of the causal effect of the SNP on phenotype (a vertical pleiotropy is assumed)
    \2 two-sample MR is were G -> E and E -> pheno are estimated in two non-overlapping samples \autocite{hemani2018EvaluatingPotentialRole,neumeyer2020StrengtheningCausalInference}, and can come from large existing QTL datasets
    % \2 Also avoids weak-instrument bias, where G -> E is estimated in a small sample
        \3 a related family of methods, TWAS \url{https://www.nature.com/articles/ng.3506}, tests association of genetically predicted expression to phenotypes, have their methodological basis in two-sample MR \url{https://link.springer.com/article/10.1007/s40484-020-0207-4}

\1 assumption violations will lead to bias in estimating causal effect of E on pheno
    \2 IV1 is satisfied by definition of an eQTL
    \2 IV2 violations are not likely, as genetic variants are unlikely to be associated with environmental confounders \todo{citation needed, try daveysmith2020MendelLawsMendelian}
    \2 IV3 violation is most likely

\1 <coloc>
    \2 IV3 violated by linkage: if the instrument eQTL is actually not causal, but simply correlated with another variant that causally affects phenotype
        \3 use colocalisation methods to test whether the same variants are associated with E and phenotype \autocite{hemani2018EvaluatingPotentialRole}

\1 <mediation analysis>
    \2 IV3 violated by horizontal pleiotropy: instrument eQTL causes expression and phenotype independently (e.g. by also being an eQTL for another gene whose expression causally affects pheno), such that expression and phenotype are not causally related
    \2 use mediation analysis e.g. (CIT\autocite{millstein2009DisentanglingMolecularRelationships}, findr\autocite{wang2017EfficientAccurateCausal}) that model comparison to distinguish between models of horizontal from vertical pleiotropy
        \3 \autocite{millstein2009DisentanglingMolecularRelationships} used by \textcite{franco2013IntegrativeGenomicAnalysis} in a vaccine response reQTL setting, although they concluded they were underpowered
    \2 however, requires individual level data, and more susceptible to measurement error than MR \autocite{hemani2018EvaluatingPotentialRole}

\1 IV3 violated by reverse causation: assumed expression -> response, but the causal direction might actually be the reverse: G -> phenotype -> expression
    \2 if suspected, recommendations to distinguish \autocite{daveysmith2014MendelianRandomizationGenetic,hemani2018EvaluatingPotentialRole,neumeyer2020StrengtheningCausalInference} \url{https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5036871/}
        \3 perform MR in the reverse direction if there are instruments for the phenotype (bi-directional MR)
        \3 extensions that test the directionality based on variance explained (MR Steiger) \url{https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1007081}
        \3 incorporate biological knowledge, e.g. if eQTL variant is in a TF motif for a gene, more likely G -> E than G -> pheno -> E

% NOTE: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7221461/
% has the nicest MR diagrams to copy.

% Mediation can exist in the absence of a 'significant' total effect G -> pheno
% http://davidakenny.net/cm/mediate.htm
% In the opinion of most though not all analysts, Step 1 is not required. (See the Power section below why the test of c can be low power, even if paths a and b are non-trivial.)

% Canalisation/compensation and population stratification affects MR with genetic instruments also.
% https://www.nature.com/articles/s41576-018-0020-3
% https://spiral.imperial.ac.uk/bitstream/10044/1/23560/2/SIM%202015_Accepted%20version.pdf
    % The most likely cause of violation of the third IV assumption is pleiotropy,  which occurswhen a genetic variant influences multiple intermediate phenotypes that separately affectthe outcome of interest.
    % The assumption can be violated through other mechanisms includ-ing canalization and population stratification.
    % Developmental canalization occurs when agenetic variant expressed during fetal development or post-natal growth stimulates compen-satory processes that protect against the effect of that variant on the outcome in adulthood[1], while population stratification occurs when ancestral sub-populations with different al-lele frequency and outcome distributions confound theG−Yassociation [1, 5, 6].

% Leland
% \autocite{taylor2019IntegrativeAnalysisGene}
% MR is a statistical framework that uses a genetic association (“the
    % instrument”) for one trait (“the exposure”) to test for a causal
    % influence on another trait (“the outcome”).  An MR result, in which the
    % exposure instrument is associated with the outcome, can arise under four
    % distinct models (10, 11): (i) there is no causal relationship, but a SNV
    % that influences the outcome is in linkage disequilibrium (LD) with a SNV
    % that influences the exposure; (ii) the exposure causally influences the
    % outcome; (iii) the outcome causally influences the exposure (a reverse
    % causal relationship); or (iv) the exposure and outcome are not causally
    % related but share a SNV that influences both the exposure and the outcome
    % independently (horizontal pleiotropy).

\section{Triangulation}

\1 Triangulation is the use of methods with different assumptions, biases, and limitations that address the same question \autocite{munafo2018RobustResearchNeeds}
    \2 discussed above is one example of triangulation, using complementary methods MR, colocalisation, mediation analysis to tease apart causal mechanisms from G -> E -> pheno
        % To distinguish among these models, we use four complementary tests (defined
        % here; Fig. S18): an MR test (consistent with all models), a
        % colocalization test (distinguishing model i from models ii–iv; Note S1),
        % the MR Steiger test (distinguishing model ii from model iii), and the
        % causal inference test (CIT; distinguishing among models ii–iv).
        \3 e.g. \textcite{taylor2019IntegrativeAnalysisGene}: an example combining MR, coloc (HEIDI), MR Steiger, CIT, to disentangle causal effect of SNPs on DNA Me and expression
        % TODO specific distinguishes
        \3 \autocite{zheng2019PhenomewideMendelianRandomization} Predicted associations between proteins and phenotypes may indicate four explanations: causality, reverse causality, confounding by LD between the leading SNPs for proteins and phenotypes or horizontal pleiotropy (Supplementary Fig. 3). Given these alternative explanations, we conducted a set of sensitivity analyses to evaluate whether each MR association reflected a causal effect of protein on phenotype: tests of reverse causality using bidirectional MR22 and MR Steiger filtering;23,24 heterogeneity analyses for proteins with multiple instruments25; and colocalization analyses26 to investigate whether the genetic associations with both protein and phenotype shared the same causal variant (Fig. 1). 
    \2 another example, how G -> E effect changes in response to perturbation
        \3 combine reQTL mapping (between individuals), dynamic ASE (within heterozygous individuals), (and maybe variance QTLs)
        % Allele-specific expression changes dynamically during T cell activation in HLA and other autoimmune loci
        \3 dynamic ASE example for multiple T cell activation states \url{https://www.nature.com/articles/s41588-020-0579-4}
    \1 another example, multiomics integration
        \2 different layers have non-redundant readouts
        \2 e.g. pQTLs
        % https://www.nature.com/articles/s41576-020-0258-4
        % Although protein and mRNA levels typically show reasonable correlation, we describe how transcriptomics and proteomics provide useful non-redundant readouts.
        \2 e.g. foreshadowing of eQTL by caQTLs 
        % https://www.nature.com/articles/s41588-018-0046-7
        \2 network approaches, MOFA
\1 many limitations discussed cannot be solved by increasing n
\1 triangulation will be key in moving from a descriptive to mechanistic understanding of immune response to perturbation

\end{outline}

%
% Misc
%
% Network analysis
    % https://bmcgenomics.biomedcentral.com/articles/10.1186/1471-2164-13-356
    % Beyond differential expression: the quest for causal mutations and effector molecules
    % In trying to understand why a phenotype changes, one should not merely think “which gene(s) are the most differentially expressed”, but rather “which gene(s) are the most differentially connected.”
    % This insight introduces us to the field of network science.
%
% Large resources 
    % e.g. dejager2015ImmVarProjectInsights, zalocusky201810000Immunomes
%
% Longitudinal
%     Finally, note that a longitudinal design can provide measurements of gene expression at many timepoints
%     Causal inference with a time-varying exposure is even harder.

