%
% Chapter 5
% Discussion
%

\chapter{Discussion}

\begin{outline}

summary of chapters

    baseline prediction might require much larger sample sizes

    Tie ch 2 to 3 using baseline predictors?
    A response eqtl is not always a response eqtl

    Limitations, and the perfect study.

suffer from core set of limitations 
    dichot
        addressed in ch2
    bulk data
        addressed in ch3
    no statistical integration
        simple overlap will not work

        e.g. other papers that talk about 'integrating'
        % franco,
        % https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0180824
        % devries2020IntegratingGWASBulk

reQTL on bulk in vivo has huge challenges

    composition explains a huge amount of var in bulk farahbod2020UntanglingEffectsCellular

    some are implicitly accounted for in peer
        but interactions with geno are not guaranteed to be

    effect is EXARCERBATED by in vivo
        not just sample var, but active recuitment changes in cell prop e.g. recruitment
            why should this be considered a con of in vivo?

    unclear what benefit would outweigh this

need to recognise 
    bulk and sorted eQTL differ in interp
        and correction using cell prop covariates does not unify them

    also, it's not the same as intervertion to set cell counts e.g. FACS

        control is not intervention
            % http://bayes.cs.ucla.edu/PRIMER/ch3-preview.pdf
            The difference between intervening on a variable and conditioning on that
            variable should,hopefully, be obvious. When we intervene on a variable in a
            model, we fix its value. Wechangethe system, and the values of other variables
            often change as a result. When we condition on avariable, we change nothing; we
            merely narrow our focus to the subset of cases in which thevariable takes the
            value we are interested in. What changes, then, is our perception about
            theworld, not the world itself.

must no longer be divorced from rigourous statistical theory
    statistical considerations from small scale studies
        Two disciplines of statistical genomics : Neither can be neglected

    move away from dichotomania
    % https://discourse.datamethods.org/t/responder-analysis-loser-x-4/1262
    % Responder analysis is a 4-fold loser. It fails to give the needed clinical
    % interpretation, has poor statistical properties, increases the cost of studies
    % because binary outcomes require significantly higher sample sizes, and raises
    % ethical issues because more patients are required to be randomized than would

            rely of stability of 'responder', something intrinsic
            % "60% of the time, it works every time"
                \url{https://discourse.datamethods.org/t/responder-analysis-loser-x-4/1262}
                senn2016MasteringVariationVariance
                senn2018StatisticalPitfallsPersonalized

        and to Cost-effectiveness and clinical implementation
            if you can identify NRs, what are you going to do about it?

        in ch 3, used TRI
        in ch 4, 
            May be more powerful to directly with the constituent phenotypes directly (CRP level, HBI score, whether there was escalation in steroid treatment, or exit for treatment failure)

            predictive claims

    in chapter 3, i discussed in brief
        change is not ancova
        stratified is not interaction
        yet many studies out there the that use one or the other to answer the same biological reQTL questions

    also beware the effects of normalisation

    beware the effects of model misspec
        missing interaction terms

esp when considering replication of reQTL
    all above must line up

Prediction is not inference (the rift between philosophies, see 2 x 2 schools papers)
    % Calculating the sample size required for developing a clinical prediction model
    two chapters of mine
        have touched on suggesting to build a predictive model
    but have always treated the R as indep in DGE
    kinda strange: response is a organism pheno should be downstream of E

        Is response , the main predictor
        A property measured at w14
        Is actually A time invariant property of the individual?

        for modelling convenicnece
        cant make causal claims anyway
            without a control, we cannot observe the counterfactual of what if an individual was a non-responder

        correlates of protection

        no error
            assuming otherwise would stray into the realm of error in variables models
        
            we are effectively assuming that R is an intrinsic thing, and is determined without error
    
    convert the language e.g. logFC, to prediction e.g. CV error

    % beyond dge
    % TODO: https://bmcgenomics.biomedcentral.com/articles/10.1186/1471-2164-13-356

    potential methods

        Sparse partial least squares regression for simultaneous dimension reduction and variable selection
        % https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2810828/

         Sparse Partial Least Squares Classification for High Dimensional Data 
        % https://pubmed.ncbi.nlm.nih.gov/20361856/


    sysvacc: need genetics to move beyond prediction

    gene signatures, rise and fall
        expression as a biomarker

        rise and fall paper: 15 years experience from cancer: list challenges to clinical implementation from Discussion

        % https://www.annalsofoncology.org/article/S0923-7534(19)36531-7/fulltext
        Table 1Evidence-based criteria for a prognostic gene signature in the path from the laboratory to clinical practice

The design of more longitudinal cohorts in the future

    more n
        just because n sufficient for eqtl at 1 condition
        underpowered when looking for interactions (implied for reQTL, even more problematic for e.g. further cell count interatcions)
        similar to stratified analysis problem in clinical trials

    ofc if avoid bulk, no prob with cell counts:
        Even if cell type interaction/proxy gene approach
        Cannot distinguish between correlated cell types

    more chance for in vivo
        but how to take advantage of it

    systems immunology/vacc still needed: generate mol phenotypes, in the right context,
    but make sure to include genotypes
    Moving out of correlation land

    all the change score nonsense

    gone are the days of GWAS marginals (just a screening approach)
        under time pressure, or convention
        throwing in covariates

    as     complex disease genetics moves 
    computation not limits correct stat


Extending the sample size

    longitudinal studies are smallish
    e.g. IBD bioresource
        TWAS: predicted gene expression, then associate with phenotypes

    will everything be gwas associated as n continues to increase
    sensitive to the smallest differences in case/control
    
    equiavlence testing \url{https://doi.org/10.1177/1948550617697177}
        but what is really smallest effect size of interest?

as more n, allows
Finer and finer context
    in the intro: gwas to function pipeline
    now, the future

    More timepoints
    Allele-specific expression changes dynamically during T cell activation in HLA and other autoimmune loci
    % https://www.nature.com/articles/s41588-020-0579-4

    More conditions
    e.g. 250 condition ASE % https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5131815/
    e.g. StructLMM 
        Identifies eQTLs with GxE, where the number of environments in E is large (modelled as a random effect)

        vqtls: e.g. % doi: https://doi.org/10.1101/2020.07.28.225730 

    as datasets and conditions get larger, proportion of eGenes is going to be 100pc, then the question is what are the most relevant ones
    % We observed that cis-eQTLs can be detected for 88% of the studied genes in vosa2018UnravelingPolygenicArchitecture
    % In contrast, trans-eQTLs (detected for 37% of 10,317 studied trait-associated variants) were more informative.

    Era of single cell. (ultamite context)
        1st
        Single-cell RNA sequencing identifies celltype-specific cis-eQTLs and co-expression QTLs
        https://www.nature.com/articles/s41588-018-0089-9

        "Single-cell eQTLGen Consortium: a personalized understanding of disease"
        https://arxiv.org/abs/1909.12550

        Optimal design of single-cell RNA sequencing experiments for cell-type-specific eQTL analysis
        https://www.biorxiv.org/content/biorxiv/early/2019/09/12/766972.full.pdf

        Single-cell genomic approaches for developing the next generation of immunotherapies Ido Yofe, Rony Dahan and Ido Amit

        % Lähnemann, D., Köster, J., Szczurek, E., McCarthy, D. J., Hicks, S. C.,
        % Robinson, M. D., Vallejos, C. A., Campbell, K. R., Beerenwinkel, N., Mahfouz,
        % A., Pinello, L., Skums, P., Stamatakis, A., Attolini, C. S.-O., Aparicio, S.,
        % Baaijens, J., Balvert, M., Barbanson, B. de, Cappuccio, A., … Schönhuth, A.
        % (2020). Eleven grand challenges in single-cell data science. Genome Biology,
        % 21(1). https://doi.org/10.1186/s13059-020-1926-6

        reQTL detection: bulk, sorted, sc
        current sc will only detect highly expressed genes

        reqtl followup using single cell cell type specific expression
        % https://journals.plos.org/plospathogens/article?id=10.1371/journal.ppat.1008408

        devries2020IntegratingGWASBulk
        Subsequently, scRNA-seq data was used to pinpoint the potential cell type in which the response QTL effects manifest themselves
            but really they only looked at cell type specific expression, not sc eqtl mapping

and allows
more phenotypes
        
        disease specific biobanks e.g. ibd bioresource/predicct

    vaccines:
        the cellular response, beyond ab titires cao2016SystemsImmunologyAntibody

    % Our GWASfor 54 functionally relevantphenotypes oftheadaptiveimmune system in 489 healthy individuals identifieseight genome-wide significant associations explain-ing 6%–20% of variance.
    % https://www.cell.com/cell-reports/pdf/S2211-1247(18)31493-1.pdf

    PheWAS\autocite{verma2017CurrentScopeChallenges}
    % https://onlinelibrary.wiley.com/doi/full/10.1111/imm.12195
    PheWAS has the advantage of identifying genetic variants with pleiotropic properties.

Translational directions
\1 Why care?

    \2 polygenic scores, prs: marker for diagnosis

    % first
    % 13. Wray, N.R., Goddard, M.E., and Visscher, P.M. (2007). Prediction
    % of individual genetic risk to disease from genome-wide
    % association studies. Genome Res. 17, 1520–1528.

        \3 use in the clinic
            % doi: http://dx.doi.org/10.1101/19013086
            \4 e.g. polygenic background can modify penetrance 

        \3 but challenges from:

            \4 ancestry effects
            % https://twitter.com/skathire/status/1202554709107712001?s=09
            \4 need expanding into global populations, global biobanks e.g. Gains from Africa H3Africa, japanese biobanks

            % Variable prediction accuracy of polygenic scores within an ancestry group
            % https://elifesciences.org/articles/48376
            \4 non-ancestry effects

    \2 pathway analysis: "the great hairball gambit"

    \2 pathway prs
        \3 challenge is variant to gene assignment/mapping
            \4 e.g. restrictions to fine mapped eQTLs

    \2 Understand mech. of causal genes: molecular pathogenesis
    % word mechanism thrown around
    % Mechanism is important because of possibility for intervention. Here, causal is important

% \subsection{From target gene to candidate drug}
%
% https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1008489
% Are drug targets with genetic support twice as likely to be approved? Revised
% estimates of the impact of genetic support for drug mechanisms on the
% probability of drug approval
    \2 how to drug a complex disease with no single 'candidate gene'?
% GWASing and fine-mapping complex diseases like IBD turns out a large number of common causal variants with small-effect sizes.
% - Is polygenicity a population or individual property? i.e. are most individual IBD cases driven solely by a distribution of small-effects, or do most patients also have 1 or more large-effect rare variants that point out priority targets for their own personalised treatment?
% - Do many of these common causal variants e.g. converge to hit on the same pathways?
% - Otherwise, what is the use of these target discovery pipelines that output ranked lists of target genes? Could a drug designed to modulate a single protein target be expected to work for a large number of patients?

    % e.g. schizo is usually polygenic, future drug development could benefit from taking a multi-target approach \autocite{visscher201710YearsGWAS}

        \3 e.g. of successful GWAS -> drug target
            \4 drug targets with genetic support are more likely

        \3 building allelic series

sample size
    complex traits and disease already moved out of candidate gene era
    vaccine and drug response traits lagging due to sample size requirements

unification
    immunology and vaccine dev: deep phenotyping, small cohorts achieved -> larger cohorts
    human genetics and gwas: large cohorts achieved -> deeper phenotyping

    more intermediates:
        The proteome
        % https://pubmed.ncbi.nlm.nih.gov/32709985/
        % Although protein and mRNA levels typically show reasonable correlation, we describe how transcriptomics and proteomics provide useful non-redundant readouts.

    MOFA
    multiomics
        % https://twitter.com/Eric_Fauman/status/1266128178692718598?s=19
        Pqtls more accurate?

    already lots of expression data

    combining systems immunology studies of genetic arch of immune parameters
            e.g. dejager2015ImmVarProjectInsights, zalocusky201810000Immunomes
        giving gmore intermediate phenotypes
        layering of evidence (triangulation)

    with GWAS of immune phenotypes

    getting at causality

\end{outline}

