%
% Chapter 1
%
% Unifies results chapters under the framework of human genetics.

% TODO: look through dloads folder for relevant papers for each section

\chapter{Introduction}

\begin{outline}

\section{Structure and diversity of the human genome}

\1 The human genome is about three billion \glspl{bp} in length, containing 20000-25000 protein-coding genes \autocite{1000genomesprojectconsortium2015GlobalReferenceHuman,theencodeprojectconsortium2012IntegratedEncyclopediaDNA} that span 1-3\% of its length.
The majority of the remainder is likely dedicated to regulatory activities \autocite{theencodeprojectconsortium2012IntegratedEncyclopediaDNA}.
A diploid human cell contains two copies of the genome; 46 chromosomes comprised of 23 maternal-parental pairs: 22 pairs of homologous autosomes and one pair of sex chromosomes.

\1 Variation in the genome exists in the form of \glspl{SNP}, short indels, and structural variants.
% https://www.ncbi.nlm.nih.gov/dbvar/content/overview/
% Structural variation (SV) is generally defined as a region of DNA
% approximately 1 kb and larger in size and can include inversions and
% balanced translocations or genomic imbalances (insertions and deletions),
% commonly referred to as copy number variants (CNVs).
The vast majority ($> \percentage{99.9}$) of variants are \glspl{SNP} and short indels \autocite{1000genomesprojectconsortium2015GlobalReferenceHuman}.
On average, a pair of human genomes differs by one \gls{SNP} per 1000-2000 \gls{bp} \autocite{theinternationalsnpmapworkinggroup2001MapHumanGenome}.

\todo{work in definition of haplotype somewhere, is it specifically from parents, or from any ancestor?}
The fundamental sources of genetic diversity are mutation and meiotic recombination, generating new alleles and haplotypes over time.
Variants at physically close loci on a chromosome are less likely to flank a recombination event, hence more likely to cosegregate, referred to as linkage.
% The Law of Dominance: An organism with alternate forms of a gene will express the form that is dominant.
% Law of Segregation: states that a diploid organism passes a randomly selected allele for a trait to its offspring, such that the offspring receives one allele from each parent.
% Law of independent assortment: states that genes do not influence each other with regard to the sorting of alleles into gametes; every possible combination of alleles for every gene is equally likely to occur.
Non-random association of alleles at two loci, more often than expected from the law of independent assortment, is known as \gls{LD} \autocite{slatkin2008LinkageDisequilibriumUnderstanding}.
\gls{LD} is often quantified within a population by $r^2$, the squared correlation coefficient between the binary presence/absence vectors for each allele over population individuals \autocite{slatkin2008LinkageDisequilibriumUnderstanding}.
Recombination events are not distributed uniformly throughout the genome.
Over time, the result is a genome structured in a mosaic of blocks delimited by recombination hotspots, each block containing haplotypes in strong \gls{LD} \autocite{wall2003HaplotypeBlocksLinkage,theinternationalhapmapconsortium2007SecondGenerationHuman}.
\todo{not exactly sure if having LD automatically results in haplotype blocks over time, there are evo forces at work too}
This correlation structure reflects population-specific evolutionary history, and can be used to trace the demography of human populations back through time \autocite{karczewski2020AnalyticTranslationalGenetics}.

\section{Genetic association studies for complex traits}

\1 Variation in human traits arises from an interplay between genetics and environment.
Traits for which genetic variation explains a non-zero fraction of phenotypic variation are heritable.
    \2 Twin studies provide upper bounds on heritability \url{https://www.nature.com/articles/nrg3243}
\1 Discovering the specific variants that contribute through association of genetic variants and phenotypes is a mainstay of the field of human genetics.
Barring somatic mutation, an individual's genome is fixed at conception, providing a causally upstream anchor.
Genetic association studies have intrinsic resistance to backdoor path effects that plague observational studies of phenotypes.
\todo{something sweeping about utility: mechanism and translational utility}
    % karczewski2020AnalyticTranslationalGenetics
    % In humans, analysis of large-scale natural variation data can provide valuable insights into human
    % disease and gene function. Joint analysis of genotype and phenotype data is especially powerful,
    % but even in the absence of phenotype data, patterns of variation data across the genome can inform
    % population history, natural selection, and the biological function of genes.
    \2 What are the benefits of leveraging natural genetic variation? Analogies to RCTs.

\1 Under the central dogma, information flows from gene to RNA to protein to phenotype via transcription and translation, thus it is assumed that genetic variation affects phenotype through it's impact on the function or regulation of genes.
How genetic variation contributes to any heritable trait defines it's genetic architecture: the number of genes affecting that trait, and the frequencies, effect sizes, and interactions between trait-associated variants \autocite{visscher2019Fisher1918Paper}.
The number of genes defines a spectrum of traits from monogenic (where inheritance may follow a Mendelian pattern) to polygenic (where inheritance is complex).
Many architectures have been proposed for complex traits; all have in common that the number of genes that affect a complex trait is very large (ranging from dozens to many thousands),
and the average effect of each locus is relatively small compared to loci associated with monogenic traits \autocite{gibson2011RareCommonVariants,boyle2017ExpandedViewComplex}.
\todo{distinguish genes and loci and variants, consider adding margin notes}

\1 [Principles of GWAS]
For decades, linkage analysis was successfully used to map genetic variants affecting Mendelian traits by tracing their cosegregation through pedigrees.
They were not successful for complex traits, as small effect size on disease risk leads to low penetrance \autocite{visscher2012FiveYearsGWAS}.
% \1 Prior to the advent of \glspl{GWAS}, genetic association studies were performed for variants in or near candidate genes selected on the basis of prior knowledge \autocite{hirschhorn2002ComprehensiveReviewGenetic}.
%     \2 Reviews of these early studies already noted issues with replication \autocite{hirschhorn2002ComprehensiveReviewGenetic},
%     \2 \glspl{GWAS} has revealed the vast majority of humans traits are complex, and that many such studies suffered from poor power at the samples sizes
%     \2 \autocite{border2019NoSupportHistorical} TODO: read this to find out exactly why they failed
\glspl{GWAS} systematically tests large numbers of variants selected in an unbiased manner across the genome for association with a trait, taking advantage of the block-like \gls{LD} structure of the genome.
    \2 The number of variants on a modern genotyping array is in the order of $10^5 - 10^6$, carefully selected to sufficiently tag (be in strong \gls{LD} with) known common variants \autocite{theinternationalhapmapconsortium2005HaplotypeMapHuman}.
    \2 Genotype imputation then allows ascertainment of variants not directly genotyped \autocite{das2018GenotypeImputationLarge}.
    As unrelated individuals still share ancestral haplotypes, study samples can be matched to a panel of haplotypes derived from reference samples via the directly genotyped variants, then unmeasured variants can then be imputed.
    Given a sufficient sample size, modern imputation panels now allow cost-effective genetic association at hundreds of millions of variants with frequencies as rare as 0.01\% \url{https://www.biorxiv.org/content/10.1101/563866v1}. 
    \todo{seems like there is some connection to be made between the selection of tag snps and the feasibility of imputation}
    \2 Testing such large numbers of variants incurs a massive multiple testing burden, but again, correlation between variants due to \gls{LD} means there are only the equivalent of $\approx 10^6$ tests that can be performed in the European genome \autocite{peer2008EstimationMultipleTesting}.
    \2 The field has thus converged on a discovery threshold of $0.05 / \num{10e6} = \num{5e-8}$ for genome-wide significance in European populations \autocite{jannot2015108HasEmerged},
    akin to controlling the family-wise type I error rate at $\alpha = 0.05$ for one million tests using the Bonferroni correction\footnote{
        The Bonferroni procedure makes no assumptions about the dependence structure of the \pvalues, and is always conservative (i.e. controls the \gls{FWER} at a stricter level than the chosen $\alpha$) unless the \pvalues have strong negative correlations \autocite{goeman2014MultipleHypothesisTesting}.
    }.

% Also see:
% 2007: Wellcome Trust Case Control Consortium landmark GWAS
% 2009: Well-known paper: Finding the missing heritability of complex diseases https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2831613/
% 2018: Revisiting the Missing Heritability of Complex Diseases, Ten Years On https://www.genome.gov/event-calendar/Revisiting-Missing-Heritability-of-Complex-Diseases-Ten-Years-On
%
% NOTE: a second approach to heritability estimation is linkage disequilibrium (LD) score regression.
%
% \1 Missing heritability refers to the observation that SNP-based heritability estimates from \gls{GWAS} fall short of additive (narrow-sense) heritability estimates from traditional quantitative methods such as twin studies.
% Perhaps unsurprisingly, it has been hypothesised that the remaining heritability lies in variants that can not be assessed by \gls{GWAS} due to rarity or small effect.
% A classic example is the heritability of human height, estimated at 80\% by twin studies \autocite{maher2008PersonalGenomesCase},
% where considering only significant associations from \gls{GWAS} explains 5\% \autocite{maher2008PersonalGenomesCase}.
% Consideration of all common variants using mixed models (\software{GCTA}) increases the estimate to 45\% \autocite{yang2010CommonSNPsExplain},
% Recent work (\software{GREML-WGS}) suggests the full estimate of 80\% might be recoverable by also including rare and poorly tagged variants measured by \gls{WGS} \autocite{wainschtein2019RecoveryTraitHeritability}.

% \1 WES/WGS is a trade-off between sample size and genomic coverage
%     \2 Allows discovery and association with rare and novel variation, including structural variants.
% \1 WES (covers about 40Mbp of the genome)
%     \2 covers more of the genome than GWAS
%     \2 but lower n, so lower power to do single variant associations
%     \2 needs 50x: variable coverage due to pulldown
% \1 WGS
%     \2 there is a tradeoff between variant capture (n needed to observe variant) and sequencing depth (gives confidence to call variants)
%     \2 20x ok to call 90\% of singletons
%     \2 rare variants, including in nc regions
%         \3 current discovery biases, finding higher effect size vars first
%         \3 burden tests (e.g. SAIGE)
%             \4 review \url{https://www.nature.com/articles/s41576-019-0177-4}
%     \2 also gets structural variants

% Read:
%     5 years of GWAS: most common traits are polygenic
%     10 years of GWAS:
% TODO: then outline only at this point, or risk wasting time
\1 [Finding the causal variant]
    \2

Operationally, in this
review what we mean by ‘‘causal variant’’ is an
(unknown) variant that has a direct or indirect functional
effect on disease risk, rather than a variant
that is associated with disease risk through LD,

    \2 tagging enables, but also a curse
    \2 tag snps: causal snps may not be directly typed, may need to be imputed

    \2 as sample sizes get larger, and provided that sequencing or imputation can more exhaustively identify all of the candidate SNPs on the haplotype, rare recombination events will pile up, helping to make the causal SNP stand out above the passenger SNPs that usually travel on its haplotype [Huang 2017].

\section{Interpretation of genetic associations with molecular phenotypes}

QTLs

\1 [locus to gene mapping problem]
    \2 Even if the causal variant is known
    \2 nc snps regulate
        \3 Genome-wide association studies have successfully identified genetic variants associated with immune-mediated disease, the majority of which are non-coding[10 Years of GWAS Discovery].

        % many prioritisation methods possible https://www.nature.com/articles/ng.3359
        e.g.
            https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5062579/
            We then used Sherlock, a tool that integrates GWAS and expression quantitative trait locus (eQTL) data, to identify weak GWAS signals that are also supported by eQTL data. 
    \2 enriched in reg

\1 using ifntermediate/endophenotypes
    \2 endophenotypes paper \autocite{goldman2007DeconstructionVulnerabilityComplex}
    % and used when final phenos are complicated https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4961231/

\1 Gene expression is a fundamental intermediate phenotype
    % dogma
        \2 measure by array, rnaseq 
    \2 theory is that genetic variants manifest their effects through these phenotypes, central dogma based

    % For example, Gamazon et al.31proposed agene-based association method called
    % PrediXcan that testspredicted gene expression through which genetic
    % variationaffects a phenotype. In this method, an external referencedataset
    % that has jointly measured genotypes and tissue-spe-cific gene expression is
    % used to identify a set of variants thatmodulate transcript abundance of a
    % gene. Like other tran-scriptome-wide association studies
    % (TWASs),32PrediXcancan be considered a weighted burden test, where
    % eachvariant in a gene set is weighted by its additive allelic effecton
    % expression.
    % 31.Gamazon, E.R., Wheeler, H.E., Shah, K.P., Mozaffari,
    % S.V.,Aquino-Michaels, K., Carroll, R.J., Eyler, A.E., Denny, J.C.,
    % Nic-olae, D.L., Cox, N.J., Im, H.K.; and GTEx Consortium (2015).A
    % gene-based association method for mapping traits usingreference
    % transcriptome data. Nat. Genet.47, 1091–1098.
    % 32.Gusev, A., Ko, A., Shi, H.,
    % Bhatia, G., Chung, W., Penninx,B.W.J.H., Jansen, R., de Geus, E.J.C.,
    % Boomsma, D.I., Wright,F.A., et al. (2016). Integrative approaches for
    % large-scale tran-scriptome-wide association studies. Nat. Genet.48,
    % 245–252.

    GTEx

    \1 eqtl reviwe: albert2015RoleRegulatoryVariationa
    % and
    % Imprialou, M., Petretto, E., & Bottolo, L. (2017). Expression QTLs Mapping and Analysis: A Bayesian Perspective. In K. Schughart & R. W. Williams (Eds.), Systems Genetics (Vol. 1488, pp. 189–215). Springer New York. https://doi.org/10.1007/978-1-4939-6427-7_8

    recent review
    Vandiedonck
    % Altogether, these studies revealed important recurrent features of the
    % thousands of eQTLs found in different tissues. Thus, almost 60%‐80% of
    % genes have at least one eQTL,64 mostly cis‐acting usually within 1 Mb of
    % the transcription start site (TSS).89 Distant eQTLs or even eQTLs lying on
    % a different chromosome are less frequent, although a bias due to a lack of
    % statistical power to detect them cannot be excluded.90 A general rule is
    % that the more distant the eQTLs, the more cell‐ and context‐specific and
    % the lower their effect sizes.

\1 coloc methods
    % (that photo on all the coloc methods that all attempt to solve the problem)

    \2 coloc
        \3 Under the assumption that the mechanism by which non-coding associations affect disease risk is through their effect on gene expression, a successful way to link associations to their target gene is by statistical colocalisation with eQTL datasets, to determine if the GWAS and eQTL signal share the same causal variant[Co-localization of Conditional eQTL and GWAS Signatures in Schizophrenia].

        % https://www.nature.com/articles/s41588-019-0385-z
    \2 TWAS
    % When a TWAS gene also colocalizes with your trait of interest you have a 19% chance you've found the correct causal gene.
    % https://twitter.com/Eric_Fauman/status/1220885592193499137?s=09
    % _tmp_TWAS_faq

    \2 MR
    https://www.nature.com/articles/s41576-018-0020-3
    Using genetic data to strengthen causal inference in observational research
    Transcriptome-wide association studies (TWAS) integrating GWAS and eQTLs data have been proposed to unravel gene–trait associations7,9,10. However, although these studies aim to identify genes whose (genetically predicted) expression is significantly associated to complex traits, they do not aim to estimate the strength of the causal effect and are unable to distinguish causation from horizontal pleiotropy (i.e., when a genetic variant influences multiple phenotypes independently). For this reason, we rather chose to apply a Mendelian randomization (MR) approach to estimate the causal effect of gene expression on complex traits.

    \2 a transcriptional risk score (TRS)

    \2 causal inference

\1 for eqtls, closest gene is often not the best candidate
        % https://twitter.com/Eric_Fauman/status/1198595609013489670/photo/1
    \2 annotation of nc var is functional genomics
        \3 e.g. gtex, ENCODE
        % http://www.cureffi.org/2019/12/04/prnp-gwas-other-traits/
        % Another solution is functional genomics — you ask whether the GWAS hit SNPs are also known, from other datasets, to have functional effects on a gene. See, as one example, a recent study of drug targets for immune-related traits [Fang 2019].

    \2 Drug target prioritisation for disease traits
    % higher success rate
    % https://doi.org/10.1371/journal.pgen.1008489

    % karczewski2020AnalyticTranslationalGenetics
    % Genetic studies inform therapeutic discoveries and development


\section{The effects of genetic variation on expression: context is key}


\1 inthe dreaded GxE interaction
    \2 "In genetics, context matters"
    % https://www.nature.com/articles/s41588-019-0515-7

    \2 for both gwas, and molQTLs, context is key

\1 Architecture varies e.g. across cell type and tissues
% Common regulatory variation impacts gene expression in a cell type–dependent manner
% The architecture of gene regulatory variation across multiple human tissues: The MuTHER study.
\1 emph all these are just interactions with diff things
    \2 tissue
    \2 cell type
    \2 interaction between cells in vivo
    \2 stimulation conditions

    % emph that reveals differences in regulation structure or importance between conditions
review: condition/Cell-type specific methods
    refere to 2019-11-19 Cell-count specific eQTL mapping papers

    % https://academic.oup.com/hmg/article/23/7/1947/655184/
\1 QTLs can interact with sex and age

\1 types of context specific QTL
    \2 ackerman conditional vs dynamic

\1 why care about reQTLs
Expose differences in regulation

% 	Found reQTLs? So what? Gather take home messages and inspiration for post reQTL analyses
% •	reQTL effect characteristation
% o	reQTLs can show reversal of effect between conditions (Fairfax et al. 2014)
% •	reQTLs vs expression
% o	enrichment in DE genes vs eQTLs and non eQTLs (Barreiro et al., 2012)
% o	context-specific eQTL are identified because of both treatment-induced regulatory effects and treatment-inducing gene expression to detectable levels (Fairfax et al. 2014)
% •	trans effects
% o	reQTLs develop trans-effects on stimulation (Fairfax et al. 2014)
% •	reQTLs in pathways
% o	Fairfax et al. 2014: reQTLs frequently intersected established canonical pathways of monocyte signaling
% •	reQTLs enrichment in relevant GWAS hits
% o	Barreiro et al., 2012
% o	Fairfax et al. 2014
% •	genomic feature enrichment of reQTLs in certain feature classes
% o	UTRs (Fu et al., 2012)
% •	Overlap of reQTL genes with DE genes
% o	Franco et al. 2014
% •	Mediation analysis of eQTL -> Ab response (Franco et al. 2014)
% •	Colocalisation with GWAS (Kim-Hellmuth et al. 2017)
% •	motifs enriched at reQTL binding sites e.g. STATs, IRFs (Caliskan et al. 2015; Davenport et al. 2018)
% 	Gather reQTL datasets for coloc
% •	Fairfax 2014 stim monocytes
% •	CEDAR range of resting cells
% •	Schmiedel_2018 stim CD4/8s
% •	Alasoo 2018 stim macro
% •	… and whole blood meta control


\1 Mechanisms of reQTLs
What molecular mechanisms might allow for interaction between \Gls{eQTL} and different environmental conditions?
% Fu, J., Wolfs, M. G. M., Deelen, P., Westra, H.-J., Fehrmann, R. S. N., te Meerman, G. J., … Franke, L. (2012). Unraveling the Regulatory Mechanisms Underlying Tissue-Dependent Genetic Variation of Gene Expression. PLoS Genetics, 8(1), e1002431. https://doi.org/10.1371/journal.pgen.1002431
Four categories of tissue-dependent cis-eQTL effects, and proposed two molecular models.
% o	reQTL review
% 	Prev paragraph:
% •	Molecular models (Fu 2012)
% •	Conceptual models (Ackermann 2014)


% Rotival, M. (2019). Characterising the genetic basis of immune response variation to identify causal mechanisms underlying disease susceptibility. HLA, 94(3), 275?284. https://doi.org/10.1111/tan.13598

% notice that reQTLs and DE can be decoupled.
% Maranville, J. C., Luca, F., Richards, A. L., Wen, X., Witonsky, D. B., Baxter, S., Stephens, M., & Di Rienzo, A. (2011). Interactions between Glucocorticoid Treatment and Cis-Regulatory Polymorphisms Contribute to Cellular Response Phenotypes. PLoS Genetics, 7(7), e1002162. https://doi.org/10.1371/journal.pgen.1002162

coloc of immune mediated traits is enhanced by context-specific eQTLs



Longitudinal studies
    what are we actually being told?
    differences in regulation over time

    genetic effects on change in expression

    both appear as an interaction

\section{Immunity is a complex trait}

Is it even plausible that genetic var is important?
    Brodin: most env paper.

Immune-mediated diseases
    Heritability of immune parameters and immune-mediated diseases
        ranges from 

% The innate immune response 
% The adaptive immune response can be either humoral or cell-mediated

Genetic factors affecting the healthy immune system

    Why study health?
    Factors affecting the healthy immune system.

    In healthy populations, $\approx$50\% variation in immune system driven by non-genetic factors, $\approx$30–40\% variation is driven by genetic variation (Liston and Goris 2018).

    "Such systems  immunology studies in  healthy individuals have  revealed that human immune  systems are incredibly  variable among individuals,  but very stable within  individuals over time (11),  and most of this variation is  attributed to non-heritable  factors (12)."

Genetic factors affecting immune response to challenge

    Given the genetic control of the healthy immune system, one can hypothesise that immune response to challenge may also be influenced by genetic factors.

    The need for controlled immune challenge in trials.
    Studies of natural infection are complicated.
    clinical trials as an opportunity: 
        Vaccines and drugs used as controlled immune challenge.

Need for systems immunology
    generate mol phenotypes 
    in the right context

\subsection{Context-specific immune response eQTLs in vitro}

% nice figure from human adaptation perspective
% https://www.researchgate.net/publication/315705480_Living_in_an_adaptive_world_Genomic_dissection_of_the_genus_Homo_and_its_immune_response

reqtls in immunity
% kim-h 2017
% An increasingly popular approach to identifying genetic factorsinfluencing
% the interindividual variation in immune response is tomap expression
% quantitative trait loci (eQTLs)—variants thatassociate  with  gene
% expression—and  to  identify  so-calledresponse  eQTLs  (reQTLs)  where  the
% eQTL  effect  differsbetween immune stimuli. Such genetic variants can impact
% thetranscriptional response to infection, and also represent geneticeffects
% that are modified by the infectious environment viagene-by-environment
% interactions. We and other groups havepreviously published reQTL studies of
% stimulated immune cellsand demonstrated that the effects of a genetic variant
% ongene expression are highly context-specific and informativefor disease6–11.

The majority of response eQTL mapping experiments to date have been conducted \textit{in vitro}, where one can precisely adjust both the length and intensity of stimulation.
Environmental variables including cell type composition or tissue type that are expected to interact with the eQTL effect and may affect the interaction effect with stimulation can be controlled.
The choice of experiment system and stimulation can also be hypothesis-driven, for example, if certain tissues are expected to be more relevant for a specific disease. 
\todo{add more pros for in vitro reQTLs here, and find citations}.

One of the first studies to perform \gls{reQTL} mapping for an immune stimulation was \autocite{barreiro2012DecipheringGeneticArchitecture}, where eQTLs were mapped separately in monocyte-derived dendritic cells before and after 18h infection with \textit{Mycobacterium tuberculosis}.
reQTLs were detected for 198 genes, 102 specific to the uninfected state, and 96 specific to the infected state. 
These reQTLs were enriched for GWAS SNPs associated with host susceptibility to tuberculosis; this was not observed for eQTLs that were not reQTLs.

% read teh classic
% Common Genetic Variants ModulatePathogen-Sensing Responsesin Human Dendritic Cells

Since then, \textit{in vitro} immune reQTL studies have been conducted for a variety of experimental systems (e.g. primary CD14+ monocytes\autocite{fairfax2014InnateImmuneActivity}) and stimulations (IFN$\gamma$ and LPS\autocite{fairfax2014GeneticsGeneExpression}).

Take home messages:
- reQTLs develop trans-effects on stimulation \autocite{fairfax2014InnateImmuneActivity}
Overall, as the number of experimental systems and stimulations increases, large number of eQTLs are only detected.

moost recent are very high thrgouhpug
% https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5131815/

\subsection{\textit{in vivo} response QTL mapping}

less popular
A complementary approach.

in vivo pros
    choice of context 
    whole organism phenotypes
    more likely to be repeated measures
        why is this good

        blocking: deliberate aliasing with higher order interactions

Review of in vivo mapping.
What we learn on top of in vitro
(Franco et al., 2013)
% TB https://www.nature.com/articles/srep16882
% Franco
% Lareau smallpox apoptosis
% Caliskan Rhinovirus
% Davenport

Large cohorts: 
% vQTLs possible: https://advances.sciencemag.org/content/5/8/eaaw3538/tab-pdf

\subsection{Immune response to vaccination is a complex trait}

Vaccination has enormous impact on global health [10.1098/rstb.2013.0433].

Vaccines stimulate the immune system with pathogen-derived antigens to induce effector responses (primarily antigen-specific antibodies) and immunological memory against the pathogen itself.
These effector responses are then be rapidly reactivated in cases of future exposure to the pathogen, mediating long-term protection.

Systems vaccinology: from empirical to rational vaccinology

    % Convert sysvacc lit review to text

    History of vaccine dev
    [summary of low-throughput immunology e.g. animal models]  

    - Vaccination coverage in vunerable populations is below optimal

    However, a vaccine that is highly efficacious in one human population may have significantly lower efficacy in other populations.
    [1 statistic on vaccine efficiacy differences e.g. rotavirus]
    % Vaccination of special populations: Protecting the vulnerable.
    Particularly challenging populations for vaccination include the infants and elderly, pregnant, immuno comprimised patients, ethnically-diverse populations, and developing countries.
    For the majority of licensed vaccines, there is a lack of understanding regarding the molecular mechanisms that underpin this variation in host immune response.
    Immunological mechanisms that underpin a specific vaccine's success or failure in a given individual are often poorly understood[Immunological mechanisms of vaccination]. 

    rational vacc, where the key is sys vacc

    Review of systems vaccinology (pull out of self\_viva\_copypasta)
    These systems vaccinology studies often consider longitudinal measurements of the transcriptomic, cellular, cytokine, and antibody immune responses following vaccination[Vaccinology in the era of high-throughput biology.].

    Systems vaccinology is the application of -omics technologies to provide a systems-level characterisation of the human immune system after vaccine-perturbation.
    Measurements are taken at multiple molecular levels (e.g. genome, transcriptome, proteome), and molecular signatures that correlate with and predict vaccine-induced immunity are identified [http://dx.doi.org/10.1098/rstb.2014.0146].
    \todo{define what a signature is}
    Systems vaccinology has been successfully applied to a variety of licensed vaccines [yellow fever, influenza], and also to vaccine candidates against [HIV, malaria], resulting in the identification of early transcriptomic signatures that predict vaccine-induced antibody responses.

    % Merge in first year report intro

    Cotugno
    - dna meth: DNA methylation [52, 53, 54] events

    How to use sysvacc to inform better design (A systems framework for vaccine design Mooney2013), and how to move towards personalised vaccinology (https://doi.org/10.1016/j.vaccine.2017.07.062).

    Overview, including pathogen-side factors

Genetic factors affecting vaccine response

    % linnik2016ImpactHostGenetic
    % Although immunobiological research (green cluster) shows close proximity to genetic studies (blue cluster), vaccine research and clinical studies (red cluster) are almost separated from genetic studies. In contrast, cancer research (yellow cluster) highly overlaps with both genetic studies and immunobiological research.

    Read this \url{Vaccine. 2018 August 28; 36(36): 5350–5357. doi:10.1016/j.vaccine.2017.07.062.}
    Search for "variation in vaccine response genetics GA Poland" in google scholar

    % https://www.thelancet.com/journals/laninf/article/PIIS1473-3099(19)30121-5/fulltext

    % See protocol paper for references

    measles
    % https://www.sciencedaily.com/releases/2011/09/110922134546.htm

    Relatively few studies have assessed the impact of human genetic variation on responses[Franco, Lareau 2016].
    % Franco to be touched on in chapter 3

    This is despite evidence from genome-wide association studies suggesting such genetic variation influences immune response to vaccines and susceptibility to disease[Systems immunogenetics of vaccines.].
    %     https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3570049/
    %     https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5548390/
    %     https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2610683/

    Results from vaccine-related twin studies e.g. in "TWIN STUDIES ON GENETIC VARIATIONS IN RESISTANCE TO TUBERCULOSIS", and (Defective T Memory Cell Differentiation after Varicella Zoster Vaccination in Older Individuals)

    Nice summary table for gwas 
    Review paper on GWAS for vaccines mooney2013SystemsImmunogeneticsVaccines

    Genetics of adverse events e.g. \url{https://www.ncbi.nlm.nih.gov/pubmed/18454680}

\subsection{Immune response to biologic therapies is a complex trait}

Expression response to biologics

Genetic factors affecting biologic responses

e.g. PANTS immunogenicity
% touched on in chapter 3

\section{Thesis overview}

By chapter CCC overview.

\end{outline}

