%
% Preamble
%
% https://tex.stackexchange.com/questions/553/what-packages-do-people-load-by-default-in-latex

% The package defines commands that switch according to the prevailing ‘draft’ or ‘final’ options; each command takes two arguments, the first for the ‘true’, the second for the ‘false’ case.
\usepackage{ifdraft}

%
% Typesetting
%

% https://tex.stackexchange.com/questions/44694/fontenc-vs-inputenc
%
% Use modern font encoding
% fontenc is oriented to output, that is, what fonts to use for printing characters.
\usepackage[T1]{fontenc}
% inputenc allows the user to input accented characters directly from the keyboard
\usepackage[utf8]{inputenc}
% provide additional commands e.g. \texteuro; fixes some errors of the type "Command * unavailable in encoding T1."
\usepackage{textcomp}

% Culturally-determined typographical (and other) rules for a wide range of languages
% e.g. Correct hyphenation. Specify patterns using e.g. \hyphenation{hy-phen-a-tion MATLAB}
% NOTE: english is specified globally as a document class option
\usepackage{babel}

% Typographic optimisation
% It doesn't work with latex, you have to use pdflatex instead.
\usepackage{microtype}

% Better Computer Modern font
\usepackage{lmodern}

% Extension package to amsmath
\usepackage{mathtools}
% Extended symbol collection
\usepackage{amssymb}
% For theorems
\usepackage{amsthm}

% Makes the standard macros for Greek letters in mathematical mode also available in text mode
\usepackage{alphabeta}

% Context-sensitive quotes, provides \enquote
%
% When using babel or polyglossia with biblatex, loading csquotes is recommended
% to ensure that quoted texts are typeset according to the rules of your main
% language.
\usepackage{csquotes}

% Format numbers using e.g. \numprint{2.742647826672E-01}
\usepackage{numprint}

% Format si units
%
\usepackage{siunitx}
\sisetup{round-mode=places, round-precision=2, range-phrase=--, range-units=single}
% Declare units
\DeclareSIUnit\bp{bp}
\DeclareSIUnit\year{yr}
\DeclareSIUnit\month{mo}

% Define line spacing
% Setspace switches off double-spacing at places where even the most die-hard
% official would doubt its utility (footnotes, figure captions, and so on); it’s
% very difficult to do this consistently if you’re manipulating \baselinestretch
% yourself.
\usepackage{setspace}
\onehalfspacing

% 1 space after punctuation
\frenchspacing

% Allow modifications to enumerate lists
\usepackage{enumitem}

% Defines an outline environment, which allows outline-style indented lists with freely mixed levels up to four levels deep.
\usepackage{outlines}

%
% Document structure and layout
%

% Automatically adds the bibliography and/or the index and/or the contents, etc.,
% to the Table of Contents listing.
\usepackage[nottoc]{tocbibind}

% Increase the value of tocdepth and secnumdepth. The tocdepth value determines
% to which level the sectioning commands are printed in the ToC (they are always
% included in the .toc file but ignored otherwise). The secnumdepth value
% determines up to what level the sectioning titles are numbered.
\setcounter{tocdepth}{3}
\setcounter{secnumdepth}{3}

% Extra control of appendices
% \usepackage[toc]{appendix}

% The package provides an easy and flexible user interface to customize page layout and margins.
% NOTE: screws with todonotes
% https://www.jswilsonandson.co.uk/thesis-binding/
% 3cm on the binding edge and 2cm on the three remaining edges.
% Please ensure that fold-outs and diagrams produced have reasonable margins and that page numbers, headers and footers are away from the edges.
%
\usepackage[\ifoptionfinal{}{showframe},includehead,top=20mm,bottom=20mm,inner=30mm,outer=20mm]{geometry}

%
% Extensive control of page headers and footers
%
% NOTE: also consider the titlesec, titleps, and titletoc trilogy.
% \usepackage{fancyhdr}
% \pagestyle{fancy}
% Clear defaults, then define elements for fancy
% \fancyhf{}
% \fancyhead[LE]{\textbf{\thepage}}
% \fancyhead[RE]{\nouppercase{\leftmark}}
% \fancyhead[LO]{\nouppercase{\rightmark}}
% \fancyhead[RO]{\textbf{\thepage}}
% Redefine plain to remove all but the page numbers
% \fancypagestyle{plain}{
%     \fancyhf{}
%     \fancyhead[LE]{\textbf{\thepage}}
%     \fancyhead[RO]{\textbf{\thepage}}
%     \renewcommand{\headrulewidth}{0pt}
% }
%
\usepackage{titleps}
\renewpagestyle{plain}{
    \sethead%
        [\textbf{\thepage}][][]% even
        {}{}{\textbf{\thepage}}% odd
}
\newpagestyle{pagestylemainmatter}{
    \setheadrule{0.4pt}%
    \sethead%
        [\textbf{\thepage}][][\chaptertitle]% even
        {\thesection\space\sectiontitle}{}{\textbf{\thepage}}% odd
}
\pagestyle{pagestylemainmatter}

% Fancy chapters
% \usepackage[Lenny]{fncychap}

%
% Figures and tables
%

% The package builds upon the graphics package, providing a key-value interface for optional arguments to the \includegraphics command.
\usepackage{graphicx}
% The caption package provides many ways to customise the captions in floating environments like figure and table, and cooperates with many other packages.
\usepackage[font={footnotesize},labelfont=bf]{caption}
% The package provides a means of using facilities analagous to those of the caption package, when writing captions for subfigures and the like.
\usepackage[font={footnotesize}]{subcaption}
% Changes the filename algorithm to check for known graphics extensions
\usepackage{grffile} 
% Using \usepackage{flafter} avoids figures floating before they were defined
\usepackage{flafter}

% tabu provides a single environment: tabu designed to make all kind of tabulars provided that they do not split across pages.
% \usepackage{tabu}

% The package defines an environment tabularx, an extension of tabular which has an additional column designator, X, which creates a paragraph-like column whose width automatically expands so that the declared width of the environment is filled.
% \usepackage{tabularx}

% Nicer table rules
\usepackage{booktabs}

% Pretty-print code
\usepackage{listings}

% Define box for scaling tables
\usepackage{adjustbox}

% Stop floats passing into other sections
\usepackage[section]{placeins}

%
% Misc
%

% The package starts from the basic facilities of the color package, and provides easy driver-independent access to several kinds of color tints, shades, tones, and mixes of arbitrary colors
\usepackage{xcolor}

% TO-DO command and list
\usepackage[
    obeyFinal,
    textsize=scriptsize,
    backgroundcolor={orange!50!white}
]{todonotes}
% \presetkeys{todonotes}{inline}{}
%
% Widen margins for todonotes + geometry combo
\ifoptionfinal{}{\paperwidth=\dimexpr \paperwidth + 6cm \relax}
\ifoptionfinal{}{\oddsidemargin=\dimexpr \oddsidemargin + 3cm \relax}
\ifoptionfinal{}{\evensidemargin=\dimexpr \evensidemargin + 3cm \relax}
% \ifoptionfinal{}{\marginparwidth=\dimexpr \marginparwidth + 3cm \relax}

% Allow dummy text
\usepackage{lipsum}

% This package provides commands for formatting dates, times and time zones
\usepackage{datetime2}

% Use symbols for footnotes
\usepackage[symbol,perpage]{footmisc}

% This package provides the command \marginnote that may be used instead of \marginpar at almost every place where \marginpar cannot be used, e.g., inside floats, footnotes, or in frames made with the framed package. 
\usepackage{marginnote}

% The epigraph package can be used to typeset a relevant quotation or saying as an epigraph, usually just after a sectional heading.
\usepackage{epigraph}

%
% Bibliography
%
% https://tex.stackexchange.com/questions/25701/bibtex-vs-biber-and-biblatex-vs-natbib
%
% bibtex and biber are external programs that process bibliography information
% and act (roughly) as the interface between your .bib file and your LaTeX
% document.
% natbib and biblatex are LaTeX packages that format citations and
% bibliographies; natbib works only with bibtex, while biblatex (at the
% moment) works with both bibtex and biber.)

% https://leesjohn.wordpress.com/2013/04/22/a-latex-bibliography-style-i-like-nature-style-in-biblatex/
\usepackage[
    style=nature,
	maxcitenames=1,
	maxbibnames=6,
    giveninits=true,
	uniquename=false,
	sorting=none,
	date=year,
	url=false,
	doi=true,
	isbn=false,
	eprint=false,
	texencoding=utf8,
	bibencoding=utf8,
    % https://tex.stackexchange.com/questions/58576/autocite-footnote/58592#58592
    % [...] the higher-level command \autocite; biblatex will automatically translate it
    % into the lower-level citation command most appropriate for the style family
    % (e.g., \parencite for authoryear, \footcite for authortitle).
    autocite=plain,
	backend=biber
]{biblatex}
% 
% NOTE: this is a path relative to main.tex
\addbibresource{backmatter/zotero_My-Library_phd_Better-BibLaTeX.bib}

% Allow URL breaks at any alphanumerical character
% Must load after biblatex for biblatex to use xurl.
\usepackage{xurl}

%
% Navigation
%

% Clickable links
% NOTE: hyperref should be loaded relatively late, but should be loaded before glossaries-extra
\usepackage{hyperref}
%
% Redefine contextual label in front of the reference for \autoref
% Package babel supports many languages, therefore you have to put the redefinition into \extrasenglish.
% \addto\extrasenglish{\renewcommand\figureautorefname{Fig.}}
%
% Note label conventions: https://en.wikibooks.org/wiki/LaTeX/Labels_and_Cross-referencing#Sections
% ch: 	chapter
% sec: 	section
% subsec: 	subsection
% fig: 	figure
% tab: 	table
% eq: 	equation
% lst: 	code listing
% itm: 	enumerated list item
% alg: 	algorithm
% app: 	appendix subsection

% Nice colors for clickable links
% \usepackage{bookmark}
\hypersetup{
	colorlinks,
	linkcolor={red!70!black},
	citecolor={blue!70!black},
	urlcolor={blue!60!black}
}

% The package enhances LaTeX's cross-referencing features, allowing the format of references to be determined automatically according to the type of reference.
% NOTE: must be loaded after hyperref
\usepackage[nameinlink,capitalise]{cleveref}
% Customise cross-ref names for figures
\crefname{figure}{Fig.}{Fig.}
\Crefname{figure}{Fig.}{Fig.}

%
% Abbreviations
%
% NOTE: there are certain packages that must be loaded before glossaries, if
% they are required: hyperref, babel, polyglossia, inputenc and fontenc.
%
% NOTE: https://www.dickimaw-books.com/faq.php?action=view&category=glossaries#pdfbookmark
% I get the message "Token not allowed in a PDF string (PDFDocEncoding)"
% As mentioned in Using Glossary Terms Without Links, you can't use
% non-expandable commands in PDF bookmarks. The command gets ignored (hyperref
% tells you this in the "removing `\Glsentrytext'" part of the warning) so you
% end up with just the expandable part (the label) in the bookmark.
% NOTE: If you want \glsglossarymark to use \MakeUppercase in the header, use the ucmark option described below.
\usepackage[
    abbreviations,
    ucmark,
    % nohypertypes={abbreviations}
]{glossaries-extra}
%
% Set first use of abbreviation to appear expanded.
\setabbreviationstyle{long-short}
% use TeX to sort (slow)
\makenoidxglossaries
%
% Reset expansions per chapter
\preto\chapter{\glsresetall}
%
% Abbreviation definitions
%
% \newabbreviation[options]{label}{short}{long}
\newabbreviation[longplural=genome-wide association studies]{GWAS}{GWAS}{genome-wide association study}
\newabbreviation[longplural=transcriptome-wide association studies]{TWAS}{TWAS}{transcriptome-wide association study}
\newabbreviation[longplural=quantitative trait loci]{QTL}{QTL}{quantitative trait locus}
\newabbreviation[longplural=expression quantitative trait loci]{eQTL}{eQTL}{expression quantitative trait locus}
\newabbreviation[longplural=response expression quantitative trait loci]{reQTL}{reQTL}{response expression quantitative trait locus}
\newabbreviation[longplural=molecular quantitative trait loci]{molQTL}{molQTL}{molecular expression quantitative trait locus}
\newabbreviation[]{HIRD}{HIRD}{Human Immune Response Dynamics}
\newabbreviation[]{TRI}{TRI}{titre response index}
\newabbreviation[]{RNAseq}{RNA-seq}{RNA-sequencing}
\newabbreviation[]{scRNAseq}{scRNA-seq}{single-cell RNA-sequencing}
\newabbreviation[]{DGE}{DGE}{differential gene expression}
\newabbreviation[]{HAI}{HAI}{haemagglutination inhibition}
\newabbreviation[]{MN}{MN}{microneutralisation}
\newabbreviation[]{PBMC}{PBMC}{peripheral blood mononuclear cell}
\newabbreviation[]{PCA}{PCA}{principal component analysis}
\newabbreviation[]{PC}{PC}{principal component}
\newabbreviation[longplural=minor allele frequencies]{MAF}{MAF}{minor allele frequency}
\newabbreviation[]{LD}{LD}{linkage disequilibrium}
\newabbreviation[]{CPM}{CPM}{counts per million}
\newabbreviation[]{TPM}{TPM}{transcripts per million}
\newabbreviation[]{TMM}{TMM}{trimmed mean of M-values}
\newabbreviation[]{ML}{ML}{maximum likelihood}
\newabbreviation[]{REML}{REML}{restricted maximum likelihood}
\newabbreviation[]{lfsr}{lfsr}{local false sign rate}
\newabbreviation[]{DC}{DC}{dendritic cell}
\newabbreviation[]{FC}{FC}{fold change}
\newabbreviation[]{FDR}{FDR}{false discovery rate}
\newabbreviation[]{FACS}{FACS}{fluorescence-activated cell sorting}
\newabbreviation[]{CyTOF}{CyTOF}{cytometry by time-of-flight}
\newabbreviation[]{HA}{HA}{haemagglutinin}
\newabbreviation[]{NA}{NA}{neuraminidase}
\newabbreviation[]{IIV}{IIV}{inactivated influenza vaccine}
\newabbreviation[]{LAIV}{LAIV}{live attenuated influenza vaccine}
\newabbreviation[]{TIV}{TIV}{trivalent inactivated influenza vaccine}
\newabbreviation[]{BH}{BH}{Benjamini-Hochberg}
\newabbreviation[]{BTM}{BTM}{blood transcription module}
\newabbreviation[]{NK}{NK}{natural killer}
\newabbreviation[]{LMM}{LMM}{linear mixed model}
\newabbreviation[]{LOCO}{LOCO}{leave-one-chromosome-out}
\newabbreviation[]{MANOVA}{MANOVA}{multivariate analysis of variance}
\newabbreviation[]{INT}{INT}{inverse normal transformation}
\newabbreviation[]{AC}{AC}{allele count}
\newabbreviation[]{TSS}{TSS}{transcription start site}
\newabbreviation[]{LRT}{LRT}{likelihood ratio test}
\newabbreviation[]{ASE}{ASE}{allele-specific expression}
\newabbreviation[]{TF}{TF}{transcription factor}
\newabbreviation[]{HLA}{HLA}{human leukocyte antigen}
\newabbreviation[]{PVE}{PVE}{proportion of variance explained}
\newabbreviation[]{CIT}{CIT}{causal inference test}
\newabbreviation[]{SNP}{SNP}{single nucleotide polymorphism}
\newabbreviation[]{FWER}{FWER}{family-wise error rate}
\newabbreviation[]{WGS}{WGS}{whole-genome sequencing}
\newabbreviation[]{WES}{WES}{whole-exome sequencing}
\newabbreviation[]{OVB}{OVB}{omitted-variable bias}
\newabbreviation[]{PANTS}{PANTS}{Personalised Anti-TNF Therapy in Crohn's Disease}
\newabbreviation[]{PNR}{PNR}{primary non-response}
\newabbreviation[]{LOR}{LOR}{loss of response}
\newabbreviation[]{IBD}{IBD}{inflammatory bowel disease}
\newabbreviation[]{CD}{CD}{Crohn's disease}
\newabbreviation[]{UC}{UC}{ulcerative colitis}
\newabbreviation[]{IMID}{IMID}{immune-mediated inflammatory disease}
\newabbreviation[]{TNF}{TNF}{tumour necrosis factor}
\newabbreviation[]{CRP}{CRP}{C-reactive protein}
\newabbreviation[]{HBI}{HBI}{Harvey Bradshaw index}
\newabbreviation[]{DAMP}{DAMP}{damage-associated molecular pattern}
\newabbreviation[]{PAMP}{PAMP}{pathogen-associated molecular pattern}
\newabbreviation[]{TLR}{TLR}{Toll-like receptor}
\newabbreviation[]{PRR}{PRR}{pattern recognition receptor}
\newabbreviation[]{MCAR}{MCAR}{missing completely at random}
\newabbreviation[]{MAR}{MAR}{missing at random}
\newabbreviation[]{MNAR}{MNAR}{missing not at random}
\newabbreviation[]{T1D}{T1D}{type 1 diabetes}
\newabbreviation[]{T2D}{T2D}{type 1 diabetes}
\newabbreviation[]{SLE}{SLE}{systemic lupus erythematosus}
\newabbreviation[]{RA}{RA}{rheumatoid arthritis}
\newabbreviation[]{MS}{MS}{multiple sclerosis}
\newabbreviation[]{BMI}{BMI}{body mass index}
\newabbreviation[longplural=degrees of freedom,shortplural=df]{df}{df}{degree of freedom}
\newabbreviation[]{HWE}{HWE}{Hardy-Weinberg equilibrium}
\newabbreviation[]{mRNA}{mRNA}{messenger RNA}
\newabbreviation[]{ncRNA}{ncRNA}{non-coding RNA}
\newabbreviation[]{UTR}{UTR}{untranslated region}
\newabbreviation[]{HSC}{HSC}{hematopoietic stem cell}
\newabbreviation[]{APC}{APC}{antigen-presenting cell}
\newabbreviation[]{MHC}{MHC}{major histocompatibility complex}
\newabbreviation[]{WHO}{WHO}{World Health Organization}
\newabbreviation[]{BCR}{BCR}{B cell receptor}
\newabbreviation[]{TCR}{TCR}{T cell receptor}
\newabbreviation[]{ELISA}{ELISA}{enzyme-linked immunosorbent assay}
\newabbreviation[]{ELISPOT}{ELISPOT}{enzyme-linked immune absorbent spot}
\newabbreviation[]{ASC}{ASC}{antibody-secreting cell}
\newabbreviation[]{RBC}{RBC}{red blood cell}
\newabbreviation[]{AUC}{AUC}{area under the curve}
\newabbreviation[]{CDR}{CDR}{complementarity-determining region}
\newabbreviation[]{MR}{MR}{Mendelian randomisation}
\newabbreviation[]{IV}{IV}{instrumental variable}
% \newabbreviation[]{bp}{bp}{base pair}
% \newabbreviation[]{}{}{}
%

%
% Custom commands
%

% Format decimals as percentages with rounding
% https://tex.stackexchange.com/questions/468597/convert-decimal-to-percentage-value
\newcommand\percentage[2][round-precision = 1]{% default precision
    \SI[round-mode = places,
        scientific-notation = fixed, fixed-exponent = 0,
        output-decimal-marker={.}, #1]{#2e2}{\percent}%
}

% Format software package names
\newcommand{\software}[1]{\texttt{#1}}

% Format gene names
\newcommand{\gene}[1]{\textit{#1}}

% Format 'p-value'
% Call using \pvalue{} to get space after.
\newcommand{\pvalue}{\textit{p}-value}
\newcommand{\pvalues}{\textit{p}-values}

% Format approx tilde with correct spacing
\newcommand{\textapprox}[1]{${\sim}{#1}$}

